diff --git a/scenes/enemies/common/ai/sequences/wander_sequence/ShouldWander.gd b/scenes/enemies/common/ai/sequences/wander_sequence/ShouldWander.gd
index 46b4ad8..4a7f728 100644
--- a/scenes/enemies/common/ai/sequences/wander_sequence/ShouldWander.gd
+++ b/scenes/enemies/common/ai/sequences/wander_sequence/ShouldWander.gd
@@ -21,12 +21,24 @@ func tick(actor: Node, blackboard: Blackboard):
 				spawn_position.y * randf_range(0.85, 1.15)
 			)
 		else:
-			wander_destination = _random_point_within_radius(spawn_position, wander_radius)
+			wander_destination = _random_point_within_radius(actor, spawn_position, wander_radius)
 		blackboard.set_value("destination_global_position", wander_destination)
 	return SUCCESS
 
 
-func _random_point_within_radius(origin: Vector2, distance: float) -> Vector2:
+func _random_point_within_radius(actor: Node, origin: Vector2, distance: float) -> Vector2:
+	var open_check := RayCast2D.new()
+	actor.add_child(open_check)
+	open_check.collision_mask = J.PHYSICS_LAYER_WORLD
+	var dst: Vector2 = _make_rand_vector(origin, distance)
+	open_check.target_position = dst
+	while open_check.is_colliding():
+		dst = _make_rand_vector(origin, distance)
+	open_check.queue_free()
+	return dst
+
+
+func _make_rand_vector(origin: Vector2, distance: float) -> Vector2:
 	return Vector2(
 		float(randi_range(origin.x - distance, origin.x + distance)),
 		float(randi_range(origin.y - distance, origin.y + distance))
