diff --git a/scripts/singletons/global.gd b/scripts/singletons/global.gd
index 57c67f4..dd7ab40 100644
--- a/scripts/singletons/global.gd
+++ b/scripts/singletons/global.gd
@@ -45,8 +45,7 @@ var env: GodotEnv
 
 func _ready():
 	env = GodotEnv.new()
-	audio.parent = self
-	audio.main_instance = audio
+	
 	load_common_env_variables()
 
 
diff --git a/scripts/singletons/jdungeon.gd b/scripts/singletons/jdungeon.gd
index e26d772..61e1c1b 100644
--- a/scripts/singletons/jdungeon.gd
+++ b/scripts/singletons/jdungeon.gd
@@ -30,6 +30,9 @@ var audio: SoundManager = SoundManager.new()
 
 func _ready():
 	uuid_util = load("res://scripts/uuid/uuid.gd").new()
+	
+	audio.parent = self
+	audio.main_instance = audio
 
 
 func register_scenes():
