diff --git a/Scenes/Levels/CustomLevelMenu.tscn b/Scenes/Levels/CustomLevelMenu.tscn
index 8dcf047..5eca666 100644
--- a/Scenes/Levels/CustomLevelMenu.tscn
+++ b/Scenes/Levels/CustomLevelMenu.tscn
@@ -368,11 +368,11 @@ layout_mode = 2
 focus_mode = 0
 
 [node name="ScrollContainer" parent="BG/Border/Levels/VBoxContainer/LevelInfo/SelectedLevel/MarginContainer/HBoxContainer/HSplitContainer/LeftHalf/LevelInfo" index="0"]
-scroll_horizontal = 200
+scroll_horizontal = 166
 is_active = true
 
 [node name="ScrollContainer2" parent="BG/Border/Levels/VBoxContainer/LevelInfo/SelectedLevel/MarginContainer/HBoxContainer/HSplitContainer/LeftHalf/LevelInfo" index="1"]
-scroll_horizontal = 8
+scroll_horizontal = 41
 is_active = true
 
 [node name="Panel" type="PanelContainer" parent="BG/Border/Levels/VBoxContainer/LevelInfo"]
@@ -383,7 +383,7 @@ theme_override_styles/panel = ExtResource("8_psbt7")
 [node name="AutoScrollContainer" type="ScrollContainer" parent="BG/Border/Levels/VBoxContainer/LevelInfo/Panel"]
 layout_mode = 2
 mouse_filter = 2
-scroll_vertical = 54
+scroll_vertical = 49
 horizontal_scroll_mode = 3
 vertical_scroll_mode = 3
 script = ExtResource("24_wf6nb")
@@ -569,12 +569,12 @@ layout_mode = 2
 focus_mode = 0
 
 [node name="ScrollContainer" parent="BG/Border/Levels/VBoxContainer/LSSLevelInfo/SelectedOnlineLevel/MarginContainer/HBoxContainer/HSplitContainer/LeftHalf/LevelInfo" index="0"]
-scroll_horizontal = 30
+scroll_horizontal = 9
 is_active = true
 auto_connect_focus = false
 
 [node name="ScrollContainer2" parent="BG/Border/Levels/VBoxContainer/LSSLevelInfo/SelectedOnlineLevel/MarginContainer/HBoxContainer/HSplitContainer/LeftHalf/LevelInfo" index="1"]
-scroll_horizontal = 8
+scroll_horizontal = 41
 is_active = true
 auto_connect_focus = false
 
@@ -587,7 +587,7 @@ theme_override_styles/panel = ExtResource("8_psbt7")
 [node name="AutoScrollContainer" type="ScrollContainer" parent="BG/Border/Levels/VBoxContainer/LSSLevelInfo/Panel"]
 layout_mode = 2
 mouse_filter = 0
-scroll_vertical = 54
+scroll_vertical = 49
 horizontal_scroll_mode = 3
 script = ExtResource("24_wf6nb")
 direction = 1
diff --git a/Scripts/UI/CustomLevelMenu.gd b/Scripts/UI/CustomLevelMenu.gd
index a7eca9a..45258a2 100644
--- a/Scripts/UI/CustomLevelMenu.gd
+++ b/Scripts/UI/CustomLevelMenu.gd
@@ -17,6 +17,7 @@ func _ready() -> void:
 	Checkpoint.passed = false
 	Global.world_num = 1
 	Global.level_num = 1
+	Global.reset_values()
 	Checkpoint.sublevel_id = 0
 	Global.current_campaign = "SMB1"
 	AudioManager.stop_all_music()
