diff --git a/game/Scripts/Game/3D/Hand.gd b/game/Scripts/Game/3D/Hand.gd
index 6fbb754..afb5d22 100644
--- a/game/Scripts/Game/3D/Hand.gd
+++ b/game/Scripts/Game/3D/Hand.gd
@@ -212,11 +212,6 @@ func _srv_set_card_rotation(card: Card) -> void:
 # body: The node to try and set the front face visibility of.
 # visible: Whether the front face should be visible or not.
 func _try_set_front_face_visible(body: Node, visible: bool) -> void:
-	# We don't want to hide the front face if this is our hand!
-	if get_tree().has_network_peer():
-		if get_tree().get_network_unique_id() == owner_id():
-			return
-	
 	var valid = false
 	if body is Card:
 		valid = true
@@ -227,16 +222,26 @@ func _try_set_front_face_visible(body: Node, visible: bool) -> void:
 		_set_front_face_visible_recursive(body, visible)
 
 func _on_Area_body_entered(body: Node):
-	if body.get("over_hand") != null:
-		body.over_hand = owner_id()
+	if body.get("over_hands") == null:
+		return
 	
-	_try_set_front_face_visible(body, false)
+	var this_id = owner_id()
+	if not this_id in body.over_hands:
+		body.over_hands.append(this_id)
+	
+	var player_id = get_tree().get_network_unique_id()
+	_try_set_front_face_visible(body, player_id in body.over_hands)
 
 func _on_Area_body_exited(body: Node):
-	if body.get("over_hand") != null:
-		body.over_hand = 0
+	if body.get("over_hands") == null:
+		return
+	
+	body.over_hands.erase(owner_id())
 	
-	_try_set_front_face_visible(body, true)
+	var show_face = true
+	if not body.over_hands.empty():
+		show_face = get_tree().get_network_unique_id() in body.over_hands
+	_try_set_front_face_visible(body, show_face)
 
 func _on_card_exiting_tree(card: Card):
 	srv_remove_card(card)
diff --git a/game/Scripts/Game/3D/Room.gd b/game/Scripts/Game/3D/Room.gd
index 7c2778f..2472dc0 100644
--- a/game/Scripts/Game/3D/Room.gd
+++ b/game/Scripts/Game/3D/Room.gd
@@ -972,14 +972,14 @@ master func request_add_cards_to_nearest_hand(card_names: Array) -> void:
 			push_error("Object " + card_name + " is not a piece!")
 			continue
 
-		if piece.get("over_hand") == null:
-			push_error("Piece " + card_name + " does not have the over_hand property!")
+		if piece.get("over_hands") == null:
+			push_error("Piece " + card_name + " does not have the over_hands property!")
 			continue
 
-		if piece.over_hand > 0:
+		if not piece.over_hands.empty():
 			var piece_dist = piece.hover_offset.length()
 			if (min_dist == null) or (piece_dist < min_dist):
-				hand_id = piece.over_hand
+				hand_id = piece.over_hands[0]
 				min_dist = piece_dist
 
 	if hand_id <= 0:
@@ -2024,8 +2024,8 @@ func _physics_process(_delta):
 			if _srv_hand_setup_frames == 0:
 				for piece in _pieces.get_children():
 					if piece is Card:
-						if piece.over_hand > 0:
-							var hand_name = str(piece.over_hand)
+						if piece.over_hands.size() == 1:
+							var hand_name = str(piece.over_hands[0])
 							if _hands.has_node(hand_name):
 								var hand = _hands.get_node(hand_name)
 								var ok = hand.srv_add_card(piece)
@@ -2514,7 +2514,7 @@ remotesync func _on_undo_stack_pushed():
 func _on_CameraController_adding_cards_to_hand(cards: Array, id: int):
 	var names = []
 	for card in cards:
-		if card.get("over_hand") != null:
+		if card.get("over_hands") != null:
 			names.append(card.name)
 
 	if id > 0:
diff --git a/game/Scripts/Game/CameraController.gd b/game/Scripts/Game/CameraController.gd
index 1ae14fe..522287d 100644
--- a/game/Scripts/Game/CameraController.gd
+++ b/game/Scripts/Game/CameraController.gd
@@ -474,7 +474,7 @@ func _process(delta):
 	if hand_preview_enabled:
 		if _piece_mouse_is_over_time > hand_preview_delay:
 			if _piece_mouse_is_over is Card:
-				if _piece_mouse_is_over.over_hand == get_tree().get_network_unique_id():
+				if _piece_mouse_is_over.over_hands == [ get_tree().get_network_unique_id() ]:
 					if not _selected_pieces.has(_piece_mouse_is_over):
 						_hand_preview_rect.visible = true
 	
@@ -757,7 +757,7 @@ func _unhandled_input(event):
 	elif event.is_action_pressed("game_flip_piece"):
 		if _selected_pieces.empty():
 			if _piece_mouse_is_over != null and _piece_mouse_is_over is Card:
-				if _piece_mouse_is_over.over_hand == get_tree().get_network_unique_id():
+				if _piece_mouse_is_over.over_hands == [ get_tree().get_network_unique_id() ]:
 					if not _piece_mouse_is_over.is_collisions_on():
 						_piece_mouse_is_over.rpc_id(1, "request_flip_vertically")
 		else:
@@ -767,7 +767,7 @@ func _unhandled_input(event):
 	elif event.is_action_pressed("game_reset_piece"):
 		if _selected_pieces.empty():
 			if _piece_mouse_is_over != null and _piece_mouse_is_over is Card:
-				if _piece_mouse_is_over.over_hand == get_tree().get_network_unique_id():
+				if _piece_mouse_is_over.over_hands == [ get_tree().get_network_unique_id() ]:
 					if not _piece_mouse_is_over.is_collisions_on():
 						_piece_mouse_is_over.rpc_id(1, "request_reset_orientation")
 		else:
@@ -1459,7 +1459,7 @@ func _set_control_hint_label() -> void:
 			["game_delete_piece"])
 	
 	elif _piece_mouse_is_over != null and _piece_mouse_is_over is Card:
-		if _piece_mouse_is_over.over_hand == get_tree().get_network_unique_id():
+		if _piece_mouse_is_over.over_hands == [ get_tree().get_network_unique_id() ]:
 			if not _piece_mouse_is_over.is_collisions_on():
 				text += _set_control_hint_label_row_actions(tr("Flip card"),
 						["game_flip_piece"])
@@ -1592,9 +1592,10 @@ func _set_debug_info_label() -> void:
 		
 		# Hide the identity of a card if it is in someone else's hand.
 		if _piece_mouse_is_over is Card:
-			var hand_id = _piece_mouse_is_over.over_hand
-			if hand_id > 0 and hand_id != get_tree().get_network_unique_id():
-				entry_path = "???"
+			var hand_id_arr = _piece_mouse_is_over.over_hands
+			if not hand_id_arr.empty():
+				if not get_tree().get_network_unique_id() in hand_id_arr:
+					entry_path = "???"
 		
 		cursor_over = "Piece %s (%s)" % [piece_name, entry_path]
 	elif _hidden_area_mouse_is_over:
@@ -2031,7 +2032,7 @@ func _on_MouseGrab_gui_input(event):
 						for piece in _selected_pieces:
 							if piece is Card or (piece is Stack and piece.is_card_stack()):
 								cards.append(piece)
-								if piece.over_hand > 0:
+								if not piece.over_hands.empty():
 									adding_card_to_hand = true
 							piece.rpc_id(1, "request_stop_hovering")
 						
diff --git a/game/Scripts/Game/Pieces/Card.gd b/game/Scripts/Game/Pieces/Card.gd
index d36e700..2986a15 100644
--- a/game/Scripts/Game/Pieces/Card.gd
+++ b/game/Scripts/Game/Pieces/Card.gd
@@ -27,7 +27,7 @@ class_name Card
 # Used by hands so they know to release the card.
 signal card_exiting_tree(card)
 
-var over_hand: int = 0
+var over_hands: Array = []
 
 # Check if the card will collide with other pieces.
 # Returns: If the card will collide with other pieces.
diff --git a/game/Scripts/Game/Pieces/Stack.gd b/game/Scripts/Game/Pieces/Stack.gd
index 69c5c89..ad14617 100644
--- a/game/Scripts/Game/Pieces/Stack.gd
+++ b/game/Scripts/Game/Pieces/Stack.gd
@@ -51,7 +51,7 @@ export(Resource) var chip_remove_sounds
 export(Resource) var chip_shuffle_sounds
 
 # This should only be useful for stacks of cards.
-var over_hand: int = 0
+var over_hands: Array = []
 
 var _collision_unit_height = 0
 var _mesh_unit_height = 0
