diff --git a/scenes/enemies/TreeTrunkGuy/TreeTrunkGuy.gd b/scenes/enemies/TreeTrunkGuy/TreeTrunkGuy.gd
index c894787..91fcf50 100644
--- a/scenes/enemies/TreeTrunkGuy/TreeTrunkGuy.gd
+++ b/scenes/enemies/TreeTrunkGuy/TreeTrunkGuy.gd
@@ -3,12 +3,8 @@ extends Enemy
 signal destination_reached
 signal stuck
 
-@onready var animation_player = $AnimationPlayer
-@onready var animation_synchronizer: AnimationSynchronizerComponent = $AnimationSynchronizerComponent
-@onready var skeleton = $Skeleton
-@onready var original_scale = $Skeleton.scale
+@onready var action_synchronizer: ActionSynchronizerComponent = $ActionSynchronizerComponent
 @onready var avoidance_rays := $AvoidanceRays
-@onready var floating_text_scene = preload("res://scenes/templates/FloatingText/FloatingText.tscn")
 @onready var destination = self.global_position:
 	set(new_destination):
 		destination = new_destination
@@ -30,8 +26,6 @@ func _ready():
 	if J.is_server():
 		beehave_tree.enabled = true
 		stuck_timer.timeout.connect(_on_stuck_timer_timeout)
-		stats.got_hurt.connect(_on_got_hurt)
-		stats.died.connect(_on_died)
 
 		_add_loot()
 	else:
@@ -57,7 +51,6 @@ func _physics_process(_delta):
 				avoidance_rays.find_avoidant_velocity(stats.movement_speed) * movement_multiplier
 			)
 			move_and_slide()
-			animation_synchronizer.send_new_loop_animation("Move")
 			if get_slide_collision_count() > 0:
 				if stuck_timer.is_stopped():
 					stuck_timer.start()
@@ -67,7 +60,6 @@ func _physics_process(_delta):
 			if enroute_to_destination:
 				enroute_to_destination = false
 				velocity = Vector2.ZERO
-				animation_synchronizer.send_new_loop_animation("Idle")
 				destination_reached.emit()
 
 
@@ -76,19 +68,7 @@ func attack(target: CharacterBody2D):
 
 	if target.get("stats"):
 		target.stats.hurt(name, damage)
-
-	if animation_synchronizer:
-		# TODO: add attack animation
-		animation_synchronizer.send_new_action_animation("Hurt")
-
-
-func _on_died():
-	animation_synchronizer.send_new_action_animation("Die")
-
-
-func _on_got_hurt(_from: String, _damage: int):
-	if not stats.is_dead:
-		animation_synchronizer.send_new_action_animation("Hurt")
+		action_synchronizer.attack(target.name, damage)
 
 
 func _on_stuck_timer_timeout():
diff --git a/scenes/enemies/TreeTrunkGuy/TreeTrunkGuy.tscn b/scenes/enemies/TreeTrunkGuy/TreeTrunkGuy.tscn
index 72ef28d..24c33df 100644
--- a/scenes/enemies/TreeTrunkGuy/TreeTrunkGuy.tscn
+++ b/scenes/enemies/TreeTrunkGuy/TreeTrunkGuy.tscn
@@ -1,4 +1,4 @@
-[gd_scene load_steps=30 format=3 uid="uid://d4b7x1k03s5hr"]
+[gd_scene load_steps=31 format=3 uid="uid://d4b7x1k03s5hr"]
 
 [ext_resource type="Script" path="res://scenes/enemies/TreeTrunkGuy/TreeTrunkGuy.gd" id="1_j7ilt"]
 [ext_resource type="Texture2D" uid="uid://ddw7gwukhi221" path="res://assets/images/enemies/treetrunkguy/Treetrunkguy.png" id="2_2joj6"]
@@ -8,15 +8,16 @@
 [ext_resource type="Texture2D" uid="uid://jyq083k5md5j" path="res://assets/images/enemies/treetrunkguy/Treetrunkguy_backgroundhand.png" id="4_agvxr"]
 [ext_resource type="PackedScene" uid="uid://c8cji8g03ayyx" path="res://scripts/components/statssynchronizercomponent/StatsSynchronizerComponent.tscn" id="4_ybnvc"]
 [ext_resource type="PackedScene" uid="uid://chhiamhxxhm3v" path="res://scripts/components/interfacecomponent/InterfaceComponent.tscn" id="5_5ijne"]
+[ext_resource type="PackedScene" uid="uid://bjjk2cftt2cex" path="res://scripts/components/actionsynchronizercomponent/ActionSynchronizerComponent.tscn" id="5_75v1y"]
 [ext_resource type="Texture2D" uid="uid://bysutqfdau0dj" path="res://assets/images/enemies/treetrunkguy/Treetrunkguy_backgroundleg.png" id="5_fywxg"]
 [ext_resource type="PackedScene" uid="uid://dcomdkqapr25s" path="res://scripts/components/floatingtextcomponent/FloatingTextComponent.tscn" id="5_ghg40"]
-[ext_resource type="PackedScene" uid="uid://bo2y25xscje6x" path="res://scripts/components/animationsynchronizercomponent/AnimationSynchronizerComponent.tscn" id="5_rx0wa"]
 [ext_resource type="Texture2D" uid="uid://cxxrxuta8os57" path="res://assets/images/enemies/treetrunkguy/Treetrunkguy_foregroundhand.png" id="6_37oon"]
 [ext_resource type="PackedScene" uid="uid://dikf715gqy3x" path="res://scripts/components/enemyrespawncomponent/EnemyRespawnComponent.tscn" id="6_npcmk"]
 [ext_resource type="PackedScene" uid="uid://pateh36oi60s" path="res://scripts/components/lootcomponent/LootComponent.tscn" id="6_pi68j"]
 [ext_resource type="Texture2D" uid="uid://dht8y8635wy5o" path="res://assets/images/enemies/treetrunkguy/Treetrunkguy_foregroundleg.png" id="7_7d51h"]
 [ext_resource type="PackedScene" uid="uid://dfkcs65b1vpi3" path="res://scripts/components/updatefacecomponent/UpdateFaceComponent.tscn" id="7_so3fl"]
 [ext_resource type="Texture2D" uid="uid://b6k68irc0gh3s" path="res://assets/images/enemies/treetrunkguy/Treetrunkguy_torso.png" id="8_31yn2"]
+[ext_resource type="PackedScene" uid="uid://c0p6pnjudl1v" path="res://scripts/components/animationcomponent/AnimationComponent.tscn" id="8_vt6nj"]
 [ext_resource type="PackedScene" uid="uid://dv6a22n6h5bly" path="res://scripts/components/avoidancerayscomponent/AvoidanceRaysComponent.tscn" id="10_2fx0n"]
 [ext_resource type="Script" path="res://addons/beehave/nodes/beehave_tree.gd" id="11_8dyih"]
 [ext_resource type="PackedScene" uid="uid://clyv1sf3bn5g0" path="res://scenes/enemies/common/ai/selectors/aggressive_mob/AggressiveMob.tscn" id="12_njkv6"]
@@ -797,15 +798,19 @@ attack_power_min = 10
 attack_power_max = 15
 movement_speed = 200.0
 
+[node name="ActionSynchronizerComponent" parent="." node_paths=PackedStringArray("watcher_synchronizer") instance=ExtResource("5_75v1y")]
+watcher_synchronizer = NodePath("../WatcherSynchronizerComponent")
+
 [node name="LootComponent" parent="." node_paths=PackedStringArray("stats") instance=ExtResource("6_pi68j")]
 stats = NodePath("../StatsSynchronizerComponent")
 
 [node name="EnemyRespawnComponent" parent="." node_paths=PackedStringArray("stats") instance=ExtResource("6_npcmk")]
 stats = NodePath("../StatsSynchronizerComponent")
 
-[node name="AnimationSynchronizerComponent" parent="." node_paths=PackedStringArray("watcher_synchronizer", "animation_player") instance=ExtResource("5_rx0wa")]
-watcher_synchronizer = NodePath("../WatcherSynchronizerComponent")
+[node name="AnimationComponent" parent="." node_paths=PackedStringArray("animation_player", "stats", "action_synchronizer") instance=ExtResource("8_vt6nj")]
 animation_player = NodePath("../AnimationPlayer")
+stats = NodePath("../StatsSynchronizerComponent")
+action_synchronizer = NodePath("../ActionSynchronizerComponent")
 
 [node name="UpdateFaceComponent" parent="." node_paths=PackedStringArray("skeleton") instance=ExtResource("7_so3fl")]
 skeleton = NodePath("../Skeleton")
@@ -834,36 +839,42 @@ metadata/_edit_lock_ = true
 
 [node name="RightArm" type="Sprite2D" parent="Sprites"]
 position = Vector2(-100, -560)
+scale = Vector2(1, 1)
 texture = ExtResource("4_agvxr")
 offset = Vector2(-173, 6.50012)
 metadata/_edit_lock_ = true
 
 [node name="RightLeg" type="Sprite2D" parent="Sprites"]
 position = Vector2(-110, -260)
+scale = Vector2(1, 1)
 texture = ExtResource("5_fywxg")
 offset = Vector2(-163, -293.5)
 metadata/_edit_lock_ = true
 
 [node name="LeftLeg" type="Sprite2D" parent="Sprites"]
 position = Vector2(100, -270)
+scale = Vector2(1, 1)
 texture = ExtResource("7_7d51h")
 offset = Vector2(-373, -283.5)
 metadata/_edit_lock_ = true
 
 [node name="Body" type="Sprite2D" parent="Sprites"]
 position = Vector2(0, -270)
+scale = Vector2(1, 1)
 texture = ExtResource("8_31yn2")
 offset = Vector2(-270, -280)
 metadata/_edit_lock_ = true
 
 [node name="Axe" type="Sprite2D" parent="Sprites"]
 position = Vector2(250, -1100)
+scale = Vector2(1, 1)
 texture = ExtResource("3_jrku4")
 offset = Vector2(-523, 546.5)
 metadata/_edit_lock_ = true
 
 [node name="LeftArm" type="Sprite2D" parent="Sprites"]
 position = Vector2(160, -570)
+scale = Vector2(1, 1)
 texture = ExtResource("6_37oon")
 offset = Vector2(-433, 16.5001)
 metadata/_edit_lock_ = true
