diff --git a/scenes/npcs/milklady/Milklady.gd b/scenes/npcs/milklady/Milklady.gd
index fea46f5..b506f4a 100644
--- a/scenes/npcs/milklady/Milklady.gd
+++ b/scenes/npcs/milklady/Milklady.gd
@@ -33,9 +33,11 @@ func interact(player: Player):
 		return
 
 	player.dialogue.sync_invoke(player.peer_id, "MilkLady")
-	player.dialogue.dialogue_system_instance.dialogue_finished.connect(
-		_on_dialogue_finished.bind(player.peer_id)
-	)
+	
+	if not player.dialogue.dialogue_system_instance.dialogue_finished.is_connected(_on_dialogue_finished.bind(player.peer_id)):
+		player.dialogue.dialogue_system_instance.dialogue_finished.connect(
+			_on_dialogue_finished.bind(player.peer_id), CONNECT_ONE_SHOT
+		)
 
 
 func _on_dialogue_finished(peerID: int):
