diff --git a/scripts/typeshoot/TypeShootLevel.gd b/scripts/typeshoot/TypeShootLevel.gd
index dd2b205..df96f29 100644
--- a/scripts/typeshoot/TypeShootLevel.gd
+++ b/scripts/typeshoot/TypeShootLevel.gd
@@ -62,7 +62,8 @@ func _input(event: InputEvent) -> void:
 
 func _on_heart_change(_delta: int, total: int) -> void:
 	if total <= 0:
-		EventBus.emit_signal("player_died")
+		HeartInventoryHandle.change_hearts_on(_player, 3)
+		EventBus.emit_signal("level_exited")
 
 
 func _on_enemy_typed_out(enemy: TypeShootEnemy) -> void:
diff --git a/scripts/ui/Notes.gd b/scripts/ui/Notes.gd
index c76bb64..c7675ae 100644
--- a/scripts/ui/Notes.gd
+++ b/scripts/ui/Notes.gd
@@ -23,7 +23,7 @@ func _ready():
 	load_notes()
 
 
-func _process(delta):
+func _unhandled_key_input(event):
 	if Input.is_action_just_pressed("show_notes") or (Input.is_action_just_pressed("pause") and _hbox.visible):
 		toggle_visible()
 
