diff --git a/source/match/Navigation.gd b/source/match/Navigation.gd
index 15ff450..ded740b 100644
--- a/source/match/Navigation.gd
+++ b/source/match/Navigation.gd
@@ -1,7 +1,5 @@
 extends Node3D
 
-const NAVIGATION_FIXING_TIMER_INTERVAL_S = 0.1
-
 var _dummy_agent_rids = {
 	Constants.Match.Navigation.Domain.AIR: null,
 	Constants.Match.Navigation.Domain.TERRAIN: null,
@@ -15,7 +13,6 @@ var _static_obstacles = []
 
 
 func _ready():
-	_setup_navigation_fixing_timer()
 	await _match.ready
 	_setup_static_obstacles()
 
@@ -34,13 +31,6 @@ func rebake(map):
 	_setup_static_obstacles()
 
 
-func _setup_navigation_fixing_timer():
-	var timer = Timer.new()
-	timer.timeout.connect(_fix_navigation)
-	add_child(timer)
-	timer.start(NAVIGATION_FIXING_TIMER_INTERVAL_S)
-
-
 func _clear_static_obstacles():
 	for static_obstacle in _static_obstacles:
 		NavigationServer3D.free_rid(static_obstacle)
@@ -69,19 +59,3 @@ func _setup_static_obstacles():
 		NavigationServer3D.obstacle_set_vertices(obstacle, obstacle_vertices)
 		NavigationServer3D.obstacle_set_avoidance_enabled(obstacle, true)
 		_static_obstacles.append(obstacle)
-
-
-func _fix_navigation():
-	"""workarounds Godot bug #72202"""
-	for domain in [
-		Constants.Match.Navigation.Domain.AIR, Constants.Match.Navigation.Domain.TERRAIN
-	]:
-		if _dummy_agent_rids[domain] == null:
-			_dummy_agent_rids[domain] = NavigationServer3D.agent_create()
-			NavigationServer3D.agent_set_position(_dummy_agent_rids[domain], Vector3(-99, 0, -99))
-			NavigationServer3D.agent_set_map(
-				_dummy_agent_rids[domain], get_navigation_map_rid_by_domain(domain)
-			)
-		else:
-			NavigationServer3D.free_rid(_dummy_agent_rids[domain])
-			_dummy_agent_rids[domain] = null
