diff --git a/src/combat/combat.gd b/src/combat/combat.gd
index 08952e8..515c151 100644
--- a/src/combat/combat.gd
+++ b/src/combat/combat.gd
@@ -14,8 +14,8 @@ var _active_arena: CombatArena = null
 # Keep track of what music track was playing previously, and return to it once combat has finished.
 var _previous_music_track: AudioStream = null
 
-@onready var _combat_container: = $CenterContainer as CenterContainer
-@onready var _transition_delay_timer: = $CenterContainer/TransitionDelay as Timer
+@onready var _combat_container := $CenterContainer as CenterContainer
+@onready var _transition_delay_timer := $CenterContainer/TransitionDelay as Timer
 
 
 func _ready() -> void:
@@ -30,7 +30,7 @@ func start(arena: PackedScene) -> void:
 
 	await Transition.cover(0.2)
 
-	var new_arena: = arena.instantiate()
+	var new_arena := arena.instantiate()
 	assert(
 		new_arena != null,
 		"Failed to initiate combat. Provided 'arena' arugment is not a CombatArena."
@@ -41,6 +41,8 @@ func start(arena: PackedScene) -> void:
 
 	_active_arena.turn_queue.combat_finished.connect(
 		func on_combat_finished(is_player_victory: bool):
+			await _display_combat_results_dialog(is_player_victory)
+			
 			# Wait a short period of time and then fade the screen to black.
 			_transition_delay_timer.start()
 			await _transition_delay_timer.timeout
@@ -72,3 +74,35 @@ func start(arena: PackedScene) -> void:
 
 	# Begin the combat. The turn queue takes over from here.
 	_active_arena.start()
+
+
+## Displays a series of dialogue bubbles using Dialogic with information about the combat's outcome.
+func _display_combat_results_dialog(is_player_victory: bool):
+	var player_party_leader_name := _active_arena.turn_queue.battlers.players[0].name
+
+	var timeline_events: Array[String]
+	if is_player_victory:
+		timeline_events = _get_victory_message_events(player_party_leader_name)
+	else:
+		timeline_events = _get_loss_message_events(player_party_leader_name)
+
+	var combat_rewards_timeline: DialogicTimeline = DialogicTimeline.new()
+	combat_rewards_timeline.events = timeline_events
+	Dialogic.start_timeline(combat_rewards_timeline)
+	await Dialogic.timeline_ended
+
+
+# These two functions are placeholders for future logic for deciding combat outcomes.
+func _get_victory_message_events(leader_name: String) -> Array[String]:
+	var events: Array[String] = [
+		"%s's party won the battle!" % leader_name
+	]
+	events.append("You wanted to find some coins, but animals have no pockets to carry them.")
+	return events
+	
+
+func _get_loss_message_events(leader_name: String) -> Array[String]:
+	var events: Array[String] = [
+		"%s's party lost the battle!" % leader_name
+	]
+	return events
diff --git a/src/combat/combat_arena.gd b/src/combat/combat_arena.gd
index 6a70c38..84ae274 100644
--- a/src/combat/combat_arena.gd
+++ b/src/combat/combat_arena.gd
@@ -23,7 +23,11 @@ func _ready() -> void:
 	_ui_turn_bar.setup(combat_participant_data)
 	
 	# The UI elements will automatically fade out once one of the battler teams has lost.
-	combat_participant_data.battlers_downed.connect(_ui_turn_bar.fade_out)
+	combat_participant_data.battlers_downed.connect(
+		func _on_battlers_downed():
+			_ui_player_menus.visible = false
+			_ui_turn_bar.fade_out()
+	)
 
 
 ## Begin combat, setting up the UI before running combat logic.
diff --git a/src/main.tscn b/src/main.tscn
index cdf5744..c219477 100644
--- a/src/main.tscn
+++ b/src/main.tscn
@@ -460,7 +460,6 @@ metadata/_edit_lock_ = true
 
 [node name="RoamingEncounter" parent="Field/Map/Town/Gamepieces" instance=ExtResource("39_p2lth")]
 position = Vector2(200, 264)
-scale = Vector2(1, 1)
 script = ExtResource("33_0gswb")
 combat_arena = ExtResource("34_moieg")
 metadata/_edit_group_ = true
@@ -470,7 +469,6 @@ emote = 0
 radius = 64
 
 [node name="GhostGFX" parent="Field/Map/Town/Gamepieces/RoamingEncounter" instance=ExtResource("36_7n3c5")]
-scale = Vector2(1, 1)
 
 [node name="ConversationEncounter" parent="Field/Map/Town/Gamepieces" instance=ExtResource("11_yntrj")]
 position = Vector2(83.2, 6.39999)
@@ -596,11 +594,9 @@ pedestal_requirement = "Red"
 
 [node name="Pedestal" parent="Field/Map/House/Gamepieces/PedestalPuzzle" instance=ExtResource("11_yntrj")]
 position = Vector2(936, 72)
-scale = Vector2(1, 1)
 metadata/_edit_group_ = true
 
 [node name="Sprite2D" type="Sprite2D" parent="Field/Map/House/Gamepieces/PedestalPuzzle/Pedestal"]
-scale = Vector2(1, 1)
 texture = ExtResource("9_woa3f")
 region_enabled = true
 region_rect = Rect2(85, 51, 16, 16)
@@ -766,9 +762,42 @@ script = ExtResource("18_cqtg7")
 gameboard_properties = ExtResource("3_esa52")
 metadata/_edit_lock_ = true
 
-[node name="UI" type="CanvasLayer" parent="Field"]
+[node name="OpeningCutscene" type="Node2D" parent="Field"]
+scale = Vector2(0.2, 0.2)
+script = ExtResource("55_vsmkr")
+timeline = ExtResource("68_jcriq")
+metadata/_edit_lock_ = true
+
+[node name="Background" type="CanvasLayer" parent="Field/OpeningCutscene"]
+
+[node name="ColorRect" type="ColorRect" parent="Field/OpeningCutscene/Background"]
+visible = false
+anchors_preset = 15
+anchor_right = 1.0
+anchor_bottom = 1.0
+grow_horizontal = 2
+grow_vertical = 2
+color = Color(0, 0, 0, 1)
+
+[node name="Combat" type="CanvasLayer" parent="."]
+script = ExtResource("69_meyff")
+
+[node name="CenterContainer" type="CenterContainer" parent="Combat"]
+anchors_preset = 15
+anchor_right = 1.0
+anchor_bottom = 1.0
+grow_horizontal = 2
+grow_vertical = 2
+size_flags_horizontal = 4
+metadata/_edit_lock_ = true
+
+[node name="TransitionDelay" type="Timer" parent="Combat/CenterContainer"]
+wait_time = 0.5
+one_shot = true
+
+[node name="UI" type="CanvasLayer" parent="."]
 
-[node name="Inventory" type="HBoxContainer" parent="Field/UI"]
+[node name="Inventory" type="HBoxContainer" parent="UI"]
 anchors_preset = 12
 anchor_top = 1.0
 anchor_right = 1.0
@@ -784,7 +813,7 @@ script = ExtResource("34_tk01t")
 metadata/_edit_group_ = true
 metadata/_edit_lock_ = true
 
-[node name="DialogueLayout" type="Control" parent="Field/UI"]
+[node name="DialogueLayout" type="Control" parent="UI"]
 visible = false
 layout_mode = 3
 anchors_preset = 15
@@ -797,7 +826,7 @@ script = ExtResource("20_lk3dv")
 metadata/_edit_group_ = true
 metadata/_edit_lock_ = true
 
-[node name="BoxMargins" type="MarginContainer" parent="Field/UI/DialogueLayout"]
+[node name="BoxMargins" type="MarginContainer" parent="UI/DialogueLayout"]
 layout_mode = 1
 anchors_preset = 12
 anchor_top = 1.0
@@ -809,11 +838,11 @@ grow_vertical = 0
 theme_override_constants/margin_left = 150
 theme_override_constants/margin_right = 150
 
-[node name="BoxBack" type="ColorRect" parent="Field/UI/DialogueLayout/BoxMargins"]
+[node name="BoxBack" type="ColorRect" parent="UI/DialogueLayout/BoxMargins"]
 layout_mode = 2
 color = Color(0.14902, 0.168627, 0.266667, 1)
 
-[node name="TextMargins" type="MarginContainer" parent="Field/UI/DialogueLayout/BoxMargins"]
+[node name="TextMargins" type="MarginContainer" parent="UI/DialogueLayout/BoxMargins"]
 layout_mode = 2
 size_flags_vertical = 8
 theme_override_constants/margin_left = 32
@@ -821,7 +850,7 @@ theme_override_constants/margin_top = 64
 theme_override_constants/margin_right = 32
 theme_override_constants/margin_bottom = 32
 
-[node name="DialogueText" type="RichTextLabel" parent="Field/UI/DialogueLayout/BoxMargins/TextMargins"]
+[node name="DialogueText" type="RichTextLabel" parent="UI/DialogueLayout/BoxMargins/TextMargins"]
 clip_contents = false
 layout_mode = 2
 theme_override_fonts/normal_font = ExtResource("21_r13nn")
@@ -836,20 +865,20 @@ shortcut_keys_enabled = false
 visible_characters_behavior = 1
 script = ExtResource("20_g8d34")
 
-[node name="Speaker" type="HBoxContainer" parent="Field/UI/DialogueLayout/BoxMargins/TextMargins/DialogueText"]
+[node name="Speaker" type="HBoxContainer" parent="UI/DialogueLayout/BoxMargins/TextMargins/DialogueText"]
 layout_mode = 1
 offset_top = -128.0
 offset_right = 380.0
 offset_bottom = -8.0
 theme_override_constants/separation = 32
 
-[node name="Portrait" type="Control" parent="Field/UI/DialogueLayout/BoxMargins/TextMargins/DialogueText/Speaker"]
+[node name="Portrait" type="Control" parent="UI/DialogueLayout/BoxMargins/TextMargins/DialogueText/Speaker"]
 custom_minimum_size = Vector2(80, 80)
 layout_mode = 2
 script = ExtResource("21_f53lt")
 mode = 1
 
-[node name="NameLabel" type="Label" parent="Field/UI/DialogueLayout/BoxMargins/TextMargins/DialogueText/Speaker"]
+[node name="NameLabel" type="Label" parent="UI/DialogueLayout/BoxMargins/TextMargins/DialogueText/Speaker"]
 layout_mode = 2
 theme_override_colors/font_outline_color = Color(0.14902, 0.168627, 0.266667, 1)
 theme_override_constants/outline_size = 24
@@ -859,71 +888,38 @@ text = "Character Name"
 vertical_alignment = 1
 script = ExtResource("19_lpxfg")
 
-[node name="DialogicNode_TypeSounds" type="AudioStreamPlayer" parent="Field/UI/DialogueLayout/BoxMargins/TextMargins/DialogueText"]
+[node name="DialogicNode_TypeSounds" type="AudioStreamPlayer" parent="UI/DialogueLayout/BoxMargins/TextMargins/DialogueText"]
 script = ExtResource("53_atrmf")
 sounds = Array[AudioStream]([ExtResource("54_fh5dy"), ExtResource("55_8lvao"), ExtResource("56_prbwr"), ExtResource("57_tomnb"), ExtResource("58_br2ni")])
 
-[node name="Choices" type="VBoxContainer" parent="Field/UI/DialogueLayout/BoxMargins"]
+[node name="Choices" type="VBoxContainer" parent="UI/DialogueLayout/BoxMargins"]
 layout_mode = 2
 size_flags_horizontal = 8
 size_flags_vertical = 8
 alignment = 2
 
-[node name="Choice1" type="Button" parent="Field/UI/DialogueLayout/BoxMargins/Choices"]
+[node name="Choice1" type="Button" parent="UI/DialogueLayout/BoxMargins/Choices"]
 layout_mode = 2
 text = "Choice1"
 script = ExtResource("53_w1dgf")
 choice_index = 1
 
-[node name="Choice2" type="Button" parent="Field/UI/DialogueLayout/BoxMargins/Choices"]
+[node name="Choice2" type="Button" parent="UI/DialogueLayout/BoxMargins/Choices"]
 layout_mode = 2
 text = "Choice2"
 script = ExtResource("53_w1dgf")
 choice_index = 2
 
-[node name="Choice3" type="Button" parent="Field/UI/DialogueLayout/BoxMargins/Choices"]
+[node name="Choice3" type="Button" parent="UI/DialogueLayout/BoxMargins/Choices"]
 layout_mode = 2
 text = "Choice3"
 script = ExtResource("53_w1dgf")
 choice_index = 3
 
-[node name="Choice4" type="Button" parent="Field/UI/DialogueLayout/BoxMargins/Choices"]
+[node name="Choice4" type="Button" parent="UI/DialogueLayout/BoxMargins/Choices"]
 layout_mode = 2
 text = "Choice4"
 script = ExtResource("53_w1dgf")
 choice_index = 4
 
-[node name="OpeningCutscene" type="Node2D" parent="Field"]
-scale = Vector2(0.2, 0.2)
-script = ExtResource("55_vsmkr")
-timeline = ExtResource("68_jcriq")
-metadata/_edit_lock_ = true
-
-[node name="Background" type="CanvasLayer" parent="Field/OpeningCutscene"]
-
-[node name="ColorRect" type="ColorRect" parent="Field/OpeningCutscene/Background"]
-visible = false
-anchors_preset = 15
-anchor_right = 1.0
-anchor_bottom = 1.0
-grow_horizontal = 2
-grow_vertical = 2
-color = Color(0, 0, 0, 1)
-
-[node name="Combat" type="CanvasLayer" parent="."]
-script = ExtResource("69_meyff")
-
-[node name="CenterContainer" type="CenterContainer" parent="Combat"]
-anchors_preset = 15
-anchor_right = 1.0
-anchor_bottom = 1.0
-grow_horizontal = 2
-grow_vertical = 2
-size_flags_horizontal = 4
-metadata/_edit_lock_ = true
-
-[node name="TransitionDelay" type="Timer" parent="Combat/CenterContainer"]
-wait_time = 0.5
-one_shot = true
-
 [connection signal="area_entered" from="Field/Map/Forest/Gamepieces/GameEndTrigger/Area2D2" to="Field/Map/Forest/Gamepieces/GameEndTrigger" method="_on_area_entered"]
