diff --git a/scenes/menu.tscn b/scenes/menu.tscn
index dab7748..fd663f3 100644
--- a/scenes/menu.tscn
+++ b/scenes/menu.tscn
@@ -5,12 +5,12 @@ anchors_preset = 15
 anchor_right = 1.0
 anchor_bottom = 1.0
 
-[node name="QuitButton" type="Button" parent="."]
-text = "Quit"
-
 [node name="StartButton" type="Button" parent="."]
 text = "Start"
 
 [node name="OptionsButton" type="Button" parent="."]
 visible = false
 text = "Options"
+
+[node name="QuitButton" type="Button" parent="."]
+text = "Quit"
