diff --git a/scenes/player/Player.tscn b/scenes/player/Player.tscn
index d2f6f22..645b3b2 100644
--- a/scenes/player/Player.tscn
+++ b/scenes/player/Player.tscn
@@ -880,53 +880,49 @@ texture = ExtResource("2_dug5c")
 metadata/_edit_lock_ = true
 
 [node name="RightArm" type="Sprite2D" parent="Sprites"]
-position = Vector2(670, 746.666)
-rotation = 1.57079
+position = Vector2(150, 0)
 texture = ExtResource("4_gw0jc")
 offset = Vector2(131, 228)
 metadata/_edit_lock_ = true
 
 [node name="RightHand" type="Sprite2D" parent="Sprites"]
-position = Vector2(200, 676.667)
+position = Vector2(80, 470)
+rotation = -1.5708
 offset = Vector2(0.00012207, 800)
 metadata/_edit_lock_ = true
 
 [node name="RightLeg" type="Sprite2D" parent="Sprites"]
-position = Vector2(230, 786.667)
-rotation = 1.57079
+position = Vector2(190, 440)
 texture = ExtResource("7_jpcwx")
 offset = Vector2(91, -212)
 metadata/_edit_lock_ = true
 
 [node name="LeftLeg" type="Sprite2D" parent="Sprites"]
-position = Vector2(210, 986.667)
-rotation = 1.57079
+position = Vector2(390, 460)
 texture = ExtResource("6_ph7fa")
 offset = Vector2(-109, -232)
 metadata/_edit_lock_ = true
 
 [node name="Body" type="Sprite2D" parent="Sprites"]
-position = Vector2(280, 866.667)
-rotation = 1.57079
+position = Vector2(270, 390)
 texture = ExtResource("8_n1x5o")
 offset = Vector2(11, -162)
 metadata/_edit_lock_ = true
 
 [node name="Head" type="Sprite2D" parent="Sprites"]
-position = Vector2(840, 896.665)
-rotation = 1.57079
+position = Vector2(300, -170)
 texture = ExtResource("5_dr4ej")
 offset = Vector2(-19, 398)
 metadata/_edit_lock_ = true
 
 [node name="LeftHand" type="Sprite2D" parent="Sprites"]
-position = Vector2(190, 1016.67)
+position = Vector2(420, 480)
+rotation = -1.5708
 offset = Vector2(0.00012207, 800)
 metadata/_edit_lock_ = true
 
 [node name="LeftArm" type="Sprite2D" parent="Sprites"]
-position = Vector2(700, 1066.67)
-rotation = 1.57079
+position = Vector2(470, -30)
 texture = ExtResource("3_dw8ng")
 offset = Vector2(-189, 248)
 metadata/_edit_lock_ = true
diff --git a/scenes/player/deathpopup/DeathPopup.tscn b/scenes/player/deathpopup/DeathPopup.tscn
index 3fd8924..fa31424 100644
--- a/scenes/player/deathpopup/DeathPopup.tscn
+++ b/scenes/player/deathpopup/DeathPopup.tscn
@@ -1,6 +1,7 @@
-[gd_scene load_steps=2 format=3 uid="uid://ccq2mcrkefjnb"]
+[gd_scene load_steps=3 format=3 uid="uid://ccq2mcrkefjnb"]
 
 [ext_resource type="Script" path="res://scenes/player/deathpopup/DeathPopup.gd" id="1_augcx"]
+[ext_resource type="Theme" uid="uid://dreboohjjcn2f" path="res://assets/themes/LoginPanelTheme.tres" id="2_htxd2"]
 
 [node name="DeathPopup" type="Control"]
 visibility_layer = 524289
@@ -14,9 +15,15 @@ script = ExtResource("1_augcx")
 
 [node name="Panel" type="Panel" parent="."]
 layout_mode = 1
-anchors_preset = 15
-anchor_right = 1.0
-anchor_bottom = 1.0
+anchors_preset = 8
+anchor_left = 0.5
+anchor_top = 0.5
+anchor_right = 0.5
+anchor_bottom = 0.5
+offset_left = -500.0
+offset_top = -340.0
+offset_right = 500.0
+offset_bottom = 340.0
 grow_horizontal = 2
 grow_vertical = 2
 
@@ -29,6 +36,7 @@ offset_left = -471.0
 offset_right = 471.0
 offset_bottom = 276.0
 grow_horizontal = 2
+theme = ExtResource("2_htxd2")
 theme_override_font_sizes/font_size = 200
 text = "You died"
 horizontal_alignment = 1
@@ -47,6 +55,7 @@ offset_right = 727.5
 offset_bottom = 138.0
 grow_horizontal = 2
 grow_vertical = 2
+theme = ExtResource("2_htxd2")
 theme_override_font_sizes/font_size = 200
 text = "Respawning in:"
 horizontal_alignment = 1
@@ -64,6 +73,7 @@ offset_top = -273.0
 offset_right = 0.5
 grow_horizontal = 2
 grow_vertical = 0
+theme = ExtResource("2_htxd2")
 theme_override_font_sizes/font_size = 200
 horizontal_alignment = 1
 vertical_alignment = 1
