diff --git a/.env.example b/.env.example
index 3f083ac..dc2e34e 100644
--- a/.env.example
+++ b/.env.example
@@ -32,6 +32,7 @@ JSON_BACKEND_FILE="res://data/users.json"
 # POSTGRES_DB=jdungeon
 
 DEBUG=true
+NETWORK_PROFILING=true
 NO_TLS=false
 # Only required if DEBUG is false. This checks the version of the server with the current client
 # VERSION_FILE="res://.version.json"
diff --git a/scripts/components/networking/statssynchronizercomponent/StatsSynchronizerComponent.gd b/scripts/components/networking/statssynchronizercomponent/StatsSynchronizerComponent.gd
index ee21d3a..b2e5909 100644
--- a/scripts/components/networking/statssynchronizercomponent/StatsSynchronizerComponent.gd
+++ b/scripts/components/networking/statssynchronizercomponent/StatsSynchronizerComponent.gd
@@ -643,5 +643,6 @@ func sync_energy_recovery(timestamp: float, from: String, current_energy: int, r
 func _on_energy_regen_timer_timeout():
 	if not ready_done:
 		return
-
-	energy_recovery(target_node.get_name(), energy_regen)
+	
+	if energy != energy_max:
+		energy_recovery(target_node.get_name(), energy_regen)
diff --git a/scripts/network/rpcs/gameserver/clock.gd b/scripts/network/rpcs/gameserver/clock.gd
index 46d87ca..69a4fac 100644
--- a/scripts/network/rpcs/gameserver/clock.gd
+++ b/scripts/network/rpcs/gameserver/clock.gd
@@ -10,11 +10,20 @@ var latency: float = 0.0
 var latency_buffer = []
 var delta_latency: float = 0.0
 
+## Network metrics
+var metrics: Dictionary = {}
+
 
 func _ready():
 	# Disable physics process for the server
 	set_physics_process(not G.is_server())
 
+	if Global.env_network_profiling:
+		metrics["fetch_server_time"] = 0
+		metrics["return_server_time"] = 0
+		metrics["get_latency"] = 0
+		metrics["return_latency"] = 0
+
 
 func _physics_process(delta):
 	G.clock += delta + delta_latency
@@ -23,6 +32,9 @@ func _physics_process(delta):
 
 @rpc("call_remote", "any_peer", "reliable")
 func fetch_server_time(client_time: float):
+	if Global.env_network_profiling:
+		metrics["fetch_server_time"] += 1
+
 	if not G.is_server():
 		return
 
@@ -32,12 +44,18 @@ func fetch_server_time(client_time: float):
 
 @rpc("call_remote", "authority", "reliable")
 func return_server_time(server_time: float, client_time: float):
+	if Global.env_network_profiling:
+		metrics["return_server_time"] += 1
+
 	latency = (Time.get_unix_time_from_system() - client_time) / 2
 	G.clock = server_time + latency
 
 
 @rpc("call_remote", "any_peer", "reliable")
 func get_latency(client_time: float):
+	if Global.env_network_profiling:
+		metrics["get_latency"] += 1
+
 	if not G.is_server():
 		return
 
@@ -47,6 +65,9 @@ func get_latency(client_time: float):
 
 @rpc("call_remote", "authority", "reliable")
 func return_latency(client_time: float):
+	if Global.env_network_profiling:
+		metrics["return_latency"] += 1
+
 	latency_buffer.append((Time.get_unix_time_from_system() - client_time) / 2)
 	if latency_buffer.size() == LATENCY_BUFFER_SIZE:
 		var total_latency = 0
diff --git a/scripts/network/rpcs/gameserver/sync.gd b/scripts/network/rpcs/gameserver/sync.gd
index 777853e..02dfa79 100644
--- a/scripts/network/rpcs/gameserver/sync.gd
+++ b/scripts/network/rpcs/gameserver/sync.gd
@@ -2,8 +2,55 @@ extends Node
 
 class_name SyncRPC
 
+## Network metrics
+var metrics: Dictionary = {}
+
+
+func _ready():
+	if Global.env_network_profiling:
+		metrics["playersynchronizer_sync_pos"] = 0
+		metrics["playersynchronizer_sync_input"] = 0
+		metrics["playersynchronizer_sync_interact"] = 0
+		metrics["playersynchronizer_request_attack"] = 0
+		metrics["positionsynchronizer_sync"] = 0
+		metrics["statssynchronizer_sync_stats"] = 0
+		metrics["statssynchronizer_sync_response"] = 0
+		metrics["playersynchronizer_request_attack"] = 0
+		metrics["statssynchronizer_sync_int_change"] = 0
+		metrics["statssynchronizer_sync_float_change"] = 0
+		metrics["statssynchronizer_sync_hurt"] = 0
+		metrics["statssynchronizer_sync_heal"] = 0
+		metrics["statssynchronizer_sync_energy_recovery"] = 0
+		metrics["networkviewsynchronizer_add_player"] = 0
+		metrics["networkviewsynchronizer_remove_player"] = 0
+		metrics["networkviewsynchronizer_add_enemy"] = 0
+		metrics["networkviewsynchronizer_remove_enemy"] = 0
+		metrics["networkviewsynchronizer_add_npc"] = 0
+		metrics["networkviewsynchronizer_remove_npc"] = 0
+		metrics["networkviewsynchronizer_add_item"] = 0
+		metrics["networkviewsynchronizer_remove_item"] = 0
+		metrics["networkviewsynchronizer_sync_bodies_in_view"] = 0
+		metrics["actionsynchronizer_sync_attack"] = 0
+		metrics["actionsynchronizer_sync_skill_use"] = 0
+		metrics["skillcomponent_sync_skills"] = 0
+		metrics["skillcomponent_sync_response"] = 0
+		metrics["playersynchronizer_sync_skill_use"] = 0
+		metrics["characterclasscomponent_sync_all"] = 0
+		metrics["characterclasscomponent_sync_response"] = 0
+		metrics["characterclasscomponent_sync_class_change"] = 0
+		metrics["dialoguesynchronizer_sync_invoke_response"] = 0
+		metrics["dialoguesynchronizer_sync_dialogue_finished"] = 0
+		metrics["statuseffectcomponent_sync_effect"] = 0
+		metrics["statuseffectcomponent_sync_effect_response"] = 0
+		metrics["statuseffectcomponent_sync_all"] = 0
+		metrics["statuseffectcomponent_sync_all_response"] = 0
+
+
 @rpc("call_remote", "authority", "unreliable")
 func playersynchronizer_sync_pos(c: int, p: Vector2, v: Vector2):
+	if Global.env_network_profiling:
+		metrics["playersynchronizer_sync_pos"] += 1
+
 	if G.client_player == null:
 		return
 
@@ -13,6 +60,9 @@ func playersynchronizer_sync_pos(c: int, p: Vector2, v: Vector2):
 
 @rpc("call_remote", "any_peer", "reliable")
 func playersynchronizer_sync_input(c: int, d: Vector2, t: float):
+	if Global.env_network_profiling:
+		metrics["playersynchronizer_sync_input"] += 1
+
 	if not G.is_server():
 		return
 
@@ -34,6 +84,9 @@ func playersynchronizer_sync_input(c: int, d: Vector2, t: float):
 
 @rpc("call_remote", "any_peer", "reliable")
 func playersynchronizer_sync_interact(t: String):
+	if Global.env_network_profiling:
+		metrics["playersynchronizer_sync_interact"] += 1
+
 	if not G.is_server():
 		return
 
@@ -55,6 +108,9 @@ func playersynchronizer_sync_interact(t: String):
 
 @rpc("call_remote", "any_peer", "reliable")
 func playersynchronizer_request_attack(t: float, e: Array):
+	if Global.env_network_profiling:
+		metrics["playersynchronizer_request_attack"] += 1
+
 	if not G.is_server():
 		return
 
@@ -76,6 +132,9 @@ func playersynchronizer_request_attack(t: float, e: Array):
 
 @rpc("call_remote", "authority", "unreliable")
 func positionsynchronizer_sync(n: String, t: float, p: Vector2, v: Vector2):
+	if Global.env_network_profiling:
+		metrics["positionsynchronizer_sync"] += 1
+
 	var entity: Node = G.world.get_entity_by_name(n)
 
 	if entity == null:
@@ -91,6 +150,9 @@ func positionsynchronizer_sync(n: String, t: float, p: Vector2, v: Vector2):
 #Called by client, runs on server
 @rpc("call_remote", "any_peer", "reliable")
 func statssynchronizer_sync_stats(n: String):
+	if Global.env_network_profiling:
+		metrics["statssynchronizer_sync_stats"] += 1
+
 	if not G.is_server():
 		return
 
@@ -114,6 +176,9 @@ func statssynchronizer_sync_stats(n: String):
 
 @rpc("call_remote", "authority", "reliable")
 func statssynchronizer_sync_response(n: String, d: Dictionary):
+	if Global.env_network_profiling:
+		metrics["statssynchronizer_sync_response"] += 1
+
 	var entity: Node = G.world.get_entity_by_name(n)
 
 	if entity == null:
@@ -130,6 +195,9 @@ func statssynchronizer_sync_response(n: String, d: Dictionary):
 func statssynchronizer_sync_int_change(
 	n: String, t: float, s: StatsSynchronizerComponent.TYPE, v: int
 ):
+	if Global.env_network_profiling:
+		metrics["statssynchronizer_sync_int_change"] += 1
+
 	var entity: Node = G.world.get_entity_by_name(n)
 
 	if entity == null:
@@ -146,6 +214,9 @@ func statssynchronizer_sync_int_change(
 func statssynchronizer_sync_float_change(
 	n: String, t: float, s: StatsSynchronizerComponent.TYPE, v: float
 ):
+	if Global.env_network_profiling:
+		metrics["statssynchronizer_sync_float_change"] += 1
+
 	var entity: Node = G.world.get_entity_by_name(n)
 
 	if entity == null:
@@ -160,6 +231,9 @@ func statssynchronizer_sync_float_change(
 
 @rpc("call_remote", "authority", "reliable")
 func statssynchronizer_sync_hurt(n: String, t: float, f: String, c: int, d: int):
+	if Global.env_network_profiling:
+		metrics["statssynchronizer_sync_hurt"] += 1
+
 	var entity: Node = G.world.get_entity_by_name(n)
 
 	if entity == null:
@@ -174,6 +248,9 @@ func statssynchronizer_sync_hurt(n: String, t: float, f: String, c: int, d: int)
 
 @rpc("call_remote", "authority", "reliable")
 func statssynchronizer_sync_heal(n: String, t: float, f: String, c: int, h: int):
+	if Global.env_network_profiling:
+		metrics["statssynchronizer_sync_heal"] += 1
+
 	var entity: Node = G.world.get_entity_by_name(n)
 
 	if entity == null:
@@ -188,6 +265,9 @@ func statssynchronizer_sync_heal(n: String, t: float, f: String, c: int, h: int)
 
 @rpc("call_remote", "authority", "reliable")
 func statssynchronizer_sync_energy_recovery(n: String, t: float, f: String, e: int, r: int):
+	if Global.env_network_profiling:
+		metrics["statssynchronizer_sync_energy_recovery"] += 1
+
 	var entity: Node = G.world.get_entity_by_name(n)
 
 	if entity == null:
@@ -202,6 +282,9 @@ func statssynchronizer_sync_energy_recovery(n: String, t: float, f: String, e: i
 
 @rpc("call_remote", "authority", "reliable")
 func networkviewsynchronizer_add_player(n: String, u: String, p: Vector2):
+	if Global.env_network_profiling:
+		metrics["networkviewsynchronizer_add_player"] += 1
+
 	var entity: Node = G.world.get_entity_by_name(n)
 
 	if entity == null:
@@ -216,6 +299,9 @@ func networkviewsynchronizer_add_player(n: String, u: String, p: Vector2):
 
 @rpc("call_remote", "authority", "reliable")
 func networkviewsynchronizer_remove_player(n: String, u: String):
+	if Global.env_network_profiling:
+		metrics["networkviewsynchronizer_remove_player"] += 1
+
 	var entity: Node = G.world.get_entity_by_name(n)
 
 	if entity == null:
@@ -230,6 +316,9 @@ func networkviewsynchronizer_remove_player(n: String, u: String):
 
 @rpc("call_remote", "authority", "reliable")
 func networkviewsynchronizer_add_enemy(n: String, en: String, ec: String, p: Vector2):
+	if Global.env_network_profiling:
+		metrics["networkviewsynchronizer_add_enemy"] += 1
+
 	var entity: Node = G.world.get_entity_by_name(n)
 
 	if entity == null:
@@ -244,6 +333,9 @@ func networkviewsynchronizer_add_enemy(n: String, en: String, ec: String, p: Vec
 
 @rpc("call_remote", "authority", "reliable")
 func networkviewsynchronizer_remove_enemy(n: String, en: String):
+	if Global.env_network_profiling:
+		metrics["networkviewsynchronizer_remove_enemy"] += 1
+
 	var entity: Node = G.world.get_entity_by_name(n)
 
 	if entity == null:
@@ -258,6 +350,9 @@ func networkviewsynchronizer_remove_enemy(n: String, en: String):
 
 @rpc("call_remote", "authority", "reliable")
 func networkviewsynchronizer_add_npc(n: String, nn: String, nc: String, p: Vector2):
+	if Global.env_network_profiling:
+		metrics["networkviewsynchronizer_add_npc"] += 1
+
 	var entity: Node = G.world.get_entity_by_name(n)
 
 	if entity == null:
@@ -272,6 +367,9 @@ func networkviewsynchronizer_add_npc(n: String, nn: String, nc: String, p: Vecto
 
 @rpc("call_remote", "authority", "reliable")
 func networkviewsynchronizer_remove_npc(n: String, nn: String):
+	if Global.env_network_profiling:
+		metrics["networkviewsynchronizer_remove_npc"] += 1
+
 	var entity: Node = G.world.get_entity_by_name(n)
 
 	if entity == null:
@@ -286,6 +384,9 @@ func networkviewsynchronizer_remove_npc(n: String, nn: String):
 
 @rpc("call_remote", "authority", "reliable")
 func networkviewsynchronizer_add_item(n: String, iu: String, ic: String, p: Vector2):
+	if Global.env_network_profiling:
+		metrics["networkviewsynchronizer_add_item"] += 1
+
 	var entity: Node = G.world.get_entity_by_name(n)
 
 	if entity == null:
@@ -300,6 +401,9 @@ func networkviewsynchronizer_add_item(n: String, iu: String, ic: String, p: Vect
 
 @rpc("call_remote", "authority", "reliable")
 func networkviewsynchronizer_remove_item(n: String, iu: String):
+	if Global.env_network_profiling:
+		metrics["networkviewsynchronizer_remove_item"] += 1
+
 	var entity: Node = G.world.get_entity_by_name(n)
 
 	if entity == null:
@@ -314,6 +418,9 @@ func networkviewsynchronizer_remove_item(n: String, iu: String):
 
 @rpc("call_remote", "any_peer", "reliable")
 func networkviewsynchronizer_sync_bodies_in_view():
+	if Global.env_network_profiling:
+		metrics["networkviewsynchronizer_sync_bodies_in_view"] += 1
+
 	if not G.is_server():
 		return
 
@@ -335,6 +442,9 @@ func networkviewsynchronizer_sync_bodies_in_view():
 
 @rpc("call_remote", "authority", "reliable")
 func actionsynchronizer_sync_attack(n: String, t: float, d: Vector2):
+	if Global.env_network_profiling:
+		metrics["actionsynchronizer_sync_attack"] += 1
+
 	var entity: Node = G.world.get_entity_by_name(n)
 
 	if entity == null:
@@ -349,6 +459,9 @@ func actionsynchronizer_sync_attack(n: String, t: float, d: Vector2):
 
 @rpc("call_remote", "authority", "reliable")
 func actionsynchronizer_sync_skill_use(n: String, t: float, p: Vector2, s: String):
+	if Global.env_network_profiling:
+		metrics["actionsynchronizer_sync_skill_use"] += 1
+
 	var entity: Node = G.world.get_entity_by_name(n)
 
 	if entity == null:
@@ -364,6 +477,9 @@ func actionsynchronizer_sync_skill_use(n: String, t: float, p: Vector2, s: Strin
 #Server only
 @rpc("call_remote", "any_peer", "reliable")
 func skillcomponent_sync_skills(n: String):
+	if Global.env_network_profiling:
+		metrics["skillcomponent_sync_skills"] += 1
+
 	if not G.is_server():
 		return
 
@@ -388,6 +504,9 @@ func skillcomponent_sync_skills(n: String):
 #Client only
 @rpc("call_remote", "authority", "reliable")
 func skillcomponent_sync_response(n: String, d: Dictionary):
+	if Global.env_network_profiling:
+		metrics["skillcomponent_sync_response"] += 1
+
 	var entity: Node = G.world.get_entity_by_name(n)
 
 	if entity == null:
@@ -402,6 +521,9 @@ func skillcomponent_sync_response(n: String, d: Dictionary):
 
 @rpc("call_remote", "any_peer", "reliable")
 func playersynchronizer_sync_skill_use(p: Vector2, s: String):
+	if Global.env_network_profiling:
+		metrics["playersynchronizer_sync_skill_use"] += 1
+
 	if not G.is_server():
 		return
 
@@ -424,6 +546,9 @@ func playersynchronizer_sync_skill_use(p: Vector2, s: String):
 #Only client can make this RPC, runs on server
 @rpc("call_remote", "any_peer", "reliable")
 func characterclasscomponent_sync_all(n: String):
+	if Global.env_network_profiling:
+		metrics["characterclasscomponent_sync_all"] += 1
+
 	if not G.is_server():
 		return
 
@@ -448,6 +573,9 @@ func characterclasscomponent_sync_all(n: String):
 #Only server can make this RPC, runs on client
 @rpc("call_remote", "authority", "reliable")
 func characterclasscomponent_sync_response(n: String, d: Dictionary):
+	if Global.env_network_profiling:
+		metrics["characterclasscomponent_sync_response"] += 1
+
 	assert(not G.is_server(), "This method is only intended for client use")
 	var entity: Node = G.world.get_entity_by_name(n)
 
@@ -464,6 +592,9 @@ func characterclasscomponent_sync_response(n: String, d: Dictionary):
 #Only client can make this RPC, runs on server
 @rpc("call_remote", "any_peer", "reliable")
 func characterclasscomponent_sync_class_change(n: String, c: Array):
+	if Global.env_network_profiling:
+		metrics["characterclasscomponent_sync_class_change"] += 1
+
 	if not G.is_server():
 		return
 
@@ -497,6 +628,9 @@ func characterclasscomponent_sync_class_change(n: String, c: Array):
 #Only server can make this RPC, runs on client
 @rpc("call_remote", "authority", "reliable")
 func dialoguesynchronizer_sync_invoke_response(n: String, d: String):
+	if Global.env_network_profiling:
+		metrics["dialoguesynchronizer_sync_invoke_response"] += 1
+
 	assert(not G.is_server(), "This method is only intended for client use")
 	var entity: Node = G.world.get_entity_by_name(n)
 
@@ -513,6 +647,9 @@ func dialoguesynchronizer_sync_invoke_response(n: String, d: String):
 #Only client can make this RPC, runs on server
 @rpc("call_remote", "any_peer", "reliable")
 func dialoguesynchronizer_sync_dialogue_finished(n: String):
+	if Global.env_network_profiling:
+		metrics["dialoguesynchronizer_sync_dialogue_finished"] += 1
+
 	if not G.is_server():
 		return
 
@@ -542,6 +679,9 @@ func dialoguesynchronizer_sync_dialogue_finished(n: String):
 
 @rpc("call_remote", "any_peer", "reliable")
 func statuseffectcomponent_sync_effect(n: String, s: String):
+	if Global.env_network_profiling:
+		metrics["statuseffectcomponent_sync_effect"] += 1
+
 	assert(not G.is_server(), "This method is only intended for client use")
 	if not G.is_server():
 		return
@@ -567,6 +707,9 @@ func statuseffectcomponent_sync_effect(n: String, s: String):
 #Only server can make this RPC, runs on client
 @rpc("call_remote", "authority", "reliable")
 func statuseffectcomponent_sync_effect_response(n: String, s: String, j: Dictionary, r: bool):
+	if Global.env_network_profiling:
+		metrics["statuseffectcomponent_sync_effect_response"] += 1
+
 	assert(not G.is_server(), "This method is only intended for client use")
 	var entity: Node = G.world.get_entity_by_name(n)
 
@@ -582,6 +725,9 @@ func statuseffectcomponent_sync_effect_response(n: String, s: String, j: Diction
 
 @rpc("call_remote", "any_peer", "reliable")
 func statuseffectcomponent_sync_all(n: String):
+	if Global.env_network_profiling:
+		metrics["statuseffectcomponent_sync_all"] += 1
+
 	assert(not G.is_server(), "This method is only intended for client use")
 	if not G.is_server():
 		return
@@ -607,6 +753,9 @@ func statuseffectcomponent_sync_all(n: String):
 #Only server can make this RPC, runs on client
 @rpc("call_remote", "authority", "reliable")
 func statuseffectcomponent_sync_all_response(n: String, d: Dictionary):
+	if Global.env_network_profiling:
+		metrics["statuseffectcomponent_sync_all_response"] += 1
+
 	assert(not G.is_server(), "This method is only intended for client use")
 	var entity: Node = G.world.get_entity_by_name(n)
 
diff --git a/scripts/singletons/gameserver.gd b/scripts/singletons/gameserver.gd
index 324c540..47add40 100644
--- a/scripts/singletons/gameserver.gd
+++ b/scripts/singletons/gameserver.gd
@@ -216,7 +216,7 @@ func _on_client_disconnected():
 
 
 func _on_clock_sync_timer_timeout():
-	if multiplayer.multiplayer_peer.get_connection_status() != MultiplayerPeer.CONNECTION_CONNECTED:
+	if multiplayer.multiplayer_peer.get_connection_status() == MultiplayerPeer.CONNECTION_CONNECTED:
 		clock_rpc.get_latency.rpc_id(1, Time.get_unix_time_from_system())
 
 
diff --git a/scripts/singletons/global.gd b/scripts/singletons/global.gd
index 16a7ca9..94f2af6 100644
--- a/scripts/singletons/global.gd
+++ b/scripts/singletons/global.gd
@@ -3,6 +3,7 @@ extends Node
 signal debug_mode_toggled(newStats: bool)
 
 var env_debug: bool = false
+var env_network_profiling: bool = false
 var env_audio_mute: bool = false
 var env_version_file: String = ""
 var env_starter_server: String = ""
@@ -164,6 +165,10 @@ func load_common_env_variables() -> bool:
 
 	GodotLogger.info("DEBUG=[%s]" % str(env_debug))
 
+	env_network_profiling = env.get_value("NETWORK_PROFILING") == "true"
+
+	GodotLogger.info("NETWORK_PROFILING=[%s]" % str(env_network_profiling))
+
 	env_run_as_gateway = env.get_value("RUN_AS_GATEWAY") == "true"
 	env_run_as_server = env.get_value("RUN_AS_SERVER") == "true"
 	env_run_as_client = env.get_value("RUN_AS_CLIENT") == "true"
