diff --git a/Scenes/Levels/Credits.tscn b/Scenes/Levels/Credits.tscn
index dbffe81..0311266 100644
--- a/Scenes/Levels/Credits.tscn
+++ b/Scenes/Levels/Credits.tscn
@@ -1,7 +1,8 @@
-[gd_scene load_steps=81 format=3 uid="uid://bgeh6ex0lbqfq"]
+[gd_scene load_steps=82 format=3 uid="uid://bgeh6ex0lbqfq"]
 
 [ext_resource type="Script" uid="uid://b3stoe2kp1nv1" path="res://Scripts/UI/Credits.gd" id="1_db8c5"]
 [ext_resource type="AudioStream" uid="uid://cdhsj02wqdt7y" path="res://Assets/Audio/BGM/Credits.mp3" id="1_k57dj"]
+[ext_resource type="Script" uid="uid://dmtw1jesw1vl7" path="res://Scripts/Classes/Components/PackStreamPlayer.gd" id="3_suvme"]
 [ext_resource type="Texture2D" uid="uid://bmvf1l15447r" path="res://Assets/Sprites/UI/Staff.png" id="4_g2gtw"]
 [ext_resource type="Texture2D" uid="uid://vqtqnrt5rco1" path="res://Assets/Sprites/UI/CreditsDioramas/1.png" id="4_j3afk"]
 [ext_resource type="Texture2D" uid="uid://dvo3gs4wg68hl" path="res://Assets/Sprites/UI/CreditsDioramas/2.png" id="6_s34bh"]
@@ -292,6 +293,8 @@ color = Color(0, 0, 0, 1)
 [node name="Music" type="AudioStreamPlayer" parent="."]
 stream = ExtResource("1_k57dj")
 bus = &"Music"
+script = ExtResource("3_suvme")
+metadata/_custom_type_script = "uid://dmtw1jesw1vl7"
 
 [node name="Labels" type="Node" parent="."]
 
