diff --git a/Scripts/Parts/Checkpoint.gd b/Scripts/Parts/Checkpoint.gd
index 2a6d854..be42506 100644
--- a/Scripts/Parts/Checkpoint.gd
+++ b/Scripts/Parts/Checkpoint.gd
@@ -34,6 +34,10 @@ func _ready() -> void:
 		hide()
 		if Settings.file.difficulty.checkpoint_style != 0:
 			queue_free()
+			return
+	elif Settings.file.difficulty.checkpoint_style == 0:
+		queue_free()
+		return
 	if passed and PipeArea.exiting_pipe_id == -1 and Global.current_game_mode != Global.GameMode.LEVEL_EDITOR and Level.vine_return_level == "" and passed_checkpoints[passed_checkpoints.size() - 1] == id:
 		for i in nodes_to_delete:
 			i.queue_free()
