diff --git a/scenes/player/statdisplay/StatDisplay.gd b/scenes/player/statdisplay/StatDisplay.gd
index 539ecd4..8ee0a1e 100644
--- a/scenes/player/statdisplay/StatDisplay.gd
+++ b/scenes/player/statdisplay/StatDisplay.gd
@@ -16,6 +16,7 @@ extends Control
 			if not stats.stats_changed.is_connected(_on_stats_changed):
 				stats.stats_changed.connect(_on_stats_changed)
 				
+			await ready
 			renew_values()
 
 @export var accept_input: bool = true
diff --git a/scripts/components/statssynchronizercomponent/StatsSynchronizerComponent.gd b/scripts/components/statssynchronizercomponent/StatsSynchronizerComponent.gd
index 99c73be..77e3b55 100644
--- a/scripts/components/statssynchronizercomponent/StatsSynchronizerComponent.gd
+++ b/scripts/components/statssynchronizercomponent/StatsSynchronizerComponent.gd
@@ -340,6 +340,7 @@ func kill():
 func load_defaults():
 	hp_max = _default_hp_max
 	energy_max = _default_energy_max
+	energy_regen = _default_energy_regen
 	attack_power_min = _default_attack_power_min
 	attack_power_max = _default_attack_power_max
 	defense = _default_defense
diff --git a/scripts/network/rpcs/gameserver/sync.gd b/scripts/network/rpcs/gameserver/sync.gd
index 2ac55a0..d17f35c 100644
--- a/scripts/network/rpcs/gameserver/sync.gd
+++ b/scripts/network/rpcs/gameserver/sync.gd
@@ -339,6 +339,7 @@ func actionsynchronizer_sync_skill_use(n: String, t: float, p: Vector2, s: Strin
 		entity.component_list["action_synchronizer"].sync_skill_use(t, p, s)
 
 
+#Server only
 @rpc("call_remote", "any_peer", "reliable")
 func skillcomponent_sync_skills(n: String):
 	if not G.is_server():
@@ -362,6 +363,7 @@ func skillcomponent_sync_skills(n: String):
 		entity.component_list["skill_component"].sync_skills()
 
 
+#Client only
 @rpc("call_remote", "authority", "reliable")
 func skillcomponent_sync_response(n: String, d: Dictionary):
 	var entity: Node = G.world.get_entity_by_name(n)
@@ -396,6 +398,8 @@ func playersynchronizer_sync_skill_use(p: Vector2, s: String):
 	if user.player.component_list.has("player_synchronizer"):
 		user.player.component_list["player_synchronizer"].sync_skill_use(p, s)
 
+
+#Server only
 @rpc("call_remote", "any_peer", "reliable")
 func characterclasscomponent_sync_all(n: String):
 	if not G.is_server():
@@ -418,6 +422,7 @@ func characterclasscomponent_sync_all(n: String):
 	if entity.component_list.has("class_component"):
 		entity.component_list["class_component"].sync_all(id)
 	
+#Client only	
 @rpc("call_remote", "authority", "reliable")
 func characterclasscomponent_sync_response(n: String, d: Dictionary):
 	var entity: Node = G.world.get_entity_by_name(n)
