diff --git a/Scripts/Classes/Blocks/BrickBlock.gd b/Scripts/Classes/Blocks/BrickBlock.gd
index 1ef429e..c09882c 100644
--- a/Scripts/Classes/Blocks/BrickBlock.gd
+++ b/Scripts/Classes/Blocks/BrickBlock.gd
@@ -13,6 +13,7 @@ func check_brick_empty() -> void:
 func on_block_hit(player: Player) -> void:
 	if player.power_state.hitbox_size == "Big":
 		if item == null:
+			self.add_collision_exception_with(player) # Don't bonk with physics, will be handled after block is broken
 			await get_tree().physics_frame
 			destroy()
 			Global.score += 50
