diff --git a/Scripts/Classes/Components/ResourceSetterNew.gd b/Scripts/Classes/Components/ResourceSetterNew.gd
index 3c746e6..19d33f1 100644
--- a/Scripts/Classes/Components/ResourceSetterNew.gd
+++ b/Scripts/Classes/Components/ResourceSetterNew.gd
@@ -190,11 +190,12 @@ func apply_properties(properties := {}) -> void:
 					obj.set(p, properties[i])
 					continue
 
+
+
 func get_variation_json(json := {}) -> Dictionary:
 	var level_theme = Global.level_theme
 	if force_properties.has("Theme"):
 		level_theme = force_properties.Theme
-	
 	for i in json.keys().filter(func(key): return key.contains("config:")):
 		get_config_file(current_resource_pack)
 		if config_to_use != {}:
diff --git a/Scripts/Classes/Singletons/Global.gd b/Scripts/Classes/Singletons/Global.gd
index 7515562..4f9fbb5 100644
--- a/Scripts/Classes/Singletons/Global.gd
+++ b/Scripts/Classes/Singletons/Global.gd
@@ -464,7 +464,7 @@ func check_completionist_achievement() -> void:
 	if achievements.count("0") == 1:
 		unlock_achievement(AchievementID.COMPLETIONIST)
 
-const FONT = preload("uid://cd221873lbtj1")
+const FONT = preload("res://Assets/Sprites/UI/Font.fnt")
 
 func sanitize_string(string := "") -> String:
 	string = string.to_upper()
diff --git a/Scripts/Parts/FontUpdater.gd b/Scripts/Parts/FontUpdater.gd
index 297ef3d..1494275 100644
--- a/Scripts/Parts/FontUpdater.gd
+++ b/Scripts/Parts/FontUpdater.gd
@@ -6,8 +6,8 @@ var score_font: FontFile = null
 var ga_font: Resource = null
 var jp_font: Resource = null
 
-var FONT_MAIN = preload("uid://bl7sbw4nx3l1t")
-var SCORE_FONT = preload("uid://cflgloiossd8a")
+var FONT_MAIN = preload("res://Resources/ThemedResources/FontMain.tres")
+var SCORE_FONT = preload("res://Resources/ThemedResources/ScoreFont.tres")
 
 
 static var current_font: Font = null
diff --git a/Scripts/UI/TextValidator.gd b/Scripts/UI/TextValidator.gd
index db362e1..5987f3b 100644
--- a/Scripts/UI/TextValidator.gd
+++ b/Scripts/UI/TextValidator.gd
@@ -4,7 +4,7 @@ extends Node
 const valid_chars := "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789-*!.^/+:,'()?_;<> \n"
 
 @export var node_to_validate: Control = null
-const FONT = preload("uid://cd221873lbtj1")
+const FONT = preload("res://Assets/Sprites/UI/Font.fnt")
 signal text_validated(new_text: String)
 
 func validate_text() -> void:
