diff --git a/scripts/classes/Boost.gd b/scripts/classes/Boost.gd
index be64c37..d2b2d84 100644
--- a/scripts/classes/Boost.gd
+++ b/scripts/classes/Boost.gd
@@ -15,33 +15,33 @@ var stat_boost_modifier_dict: Dictionary
 
 var hp_max: int = 0:
 	set(val):
-		stat_boost_dict["hp_max"] = val
+		set_stat_boost("hp_max", val)
 	get:
-		return stat_boost_dict.get("hp_max", 0 as int)
+		return get_stat_boost("hp_max")
 
 var hp: int = 0:
 	set(val):
-		stat_boost_dict["hp"] = val
+		set_stat_boost("hp", val)
 	get:
-		return stat_boost_dict.get("hp", 0 as int)
+		return get_stat_boost("hp")
 
 var attack_power_min: int = 0:
 	set(val):
-		stat_boost_dict["attack_power_min"] = val
+		set_stat_boost("attack_power_min", val)
 	get:
-		return stat_boost_dict.get("attack_power_min", 0 as int)
+		return get_stat_boost("attack_power_min")
 
 var attack_power_max: int = 0:
 	set(val):
-		stat_boost_dict["attack_power_max"] = val
+		set_stat_boost("attack_power_max", val)
 	get:
-		return stat_boost_dict.get("attack_power_max", 0 as int)
+		return get_stat_boost("attack_power_max")
 
 var defense: int = 0:
 	set(val):
-		stat_boost_dict["defense"] = val
+		set_stat_boost("defense", val)
 	get:
-		return stat_boost_dict.get("defense", 0 as int)
+		return get_stat_boost("defense")
 
 
 func combine_boost(boost: Boost):
@@ -54,10 +54,10 @@ func combine_boost(boost: Boost):
 				)
 			)
 
-	for stat: String in stat_boost_dict:
+	for stat: String in boost.stat_boost_dict:
 		add_stat_boost(stat, boost.get_stat_boost(stat))
 
-	for stat: String in stat_boost_modifier_dict:
+	for stat: String in boost.stat_boost_modifier_dict:
 		add_stat_boost_modifier(stat, boost.get_stat_boost_modifier(stat))
 
 
diff --git a/scripts/components/networking/statssynchronizercomponent/StatsSynchronizerComponent.gd b/scripts/components/networking/statssynchronizercomponent/StatsSynchronizerComponent.gd
index 2375d80..908587e 100644
--- a/scripts/components/networking/statssynchronizercomponent/StatsSynchronizerComponent.gd
+++ b/scripts/components/networking/statssynchronizercomponent/StatsSynchronizerComponent.gd
@@ -463,7 +463,10 @@ func reload_boosts():
 		for boost: Boost in active_boosts:
 			# Store the bonuses of the boost added to the flat amount, as well as the modifiers
 			flat_values[stat] += boost.get_stat_boost(stat)
-			modifier_values[stat] *= boost.get_stat_boost_modifier(stat)
+			
+			# Ignore modifiers of 1.
+			if boost.get_stat_boost_modifier(stat) != 1:
+				modifier_values[stat] *= boost.get_stat_boost_modifier(stat)
 
 		# Set the stat to the flat value multiplied by the modifier
 		set(stat, flat_values[stat] * modifier_values[stat])
