diff --git a/scenes/DDR/Level.gd b/scenes/DDR/Level.gd
index 1b6c98d..f6bbd67 100644
--- a/scenes/DDR/Level.gd
+++ b/scenes/DDR/Level.gd
@@ -20,10 +20,7 @@ func _process(_delta):
 	emit_signal("set_tick", time)
 
 func _on_Music_finished():
-	print("Music Finished")
-	#EventBus.emit_signal("ui_visibility_changed", {"visible": true})
 	EventBus.emit_signal("change_scene", { "scene": "res://scenes/Main.tscn" })
-	#EventBus.emit_signal("bgm_changed", {"playing": true})
 
 func _on_note_hit(score):
 	total_score += score
diff --git a/scenes/EndPortal.gd b/scenes/EndPortal.gd
index 90dc6e3..ab5ed15 100644
--- a/scenes/EndPortal.gd
+++ b/scenes/EndPortal.gd
@@ -36,11 +36,10 @@ func can_enter(node: Node2D) -> bool:
 
 
 func on_portal_enter(node: Node2D) -> AnimationPlayer:
-	var player: Player = node as Player
-	if not player:
-		return null
+	$PortalSFX.play()
+
+	var player_sprite = node.get_node_or_null("Sprite") if not node as Player else (node as Player).sprite
 
-	var player_sprite = player.sprite
 	if player_sprite as Sprite:
 		sprite.texture = player_sprite.texture
 		sprite.hframes = player_sprite.hframes
@@ -53,11 +52,11 @@ func on_portal_enter(node: Node2D) -> AnimationPlayer:
 		sprite.hframes = 1
 		sprite.vframes = 1
 		sprite.frame = 0
+	else:
+		return null
 
 	animationplayer.play("Swirl")
 
-	$PortalSFX.play()
-	EventBus.emit_signal("level_completed", {})
 	return animationplayer
 
 
diff --git a/scenes/ui/LevelCount.gd b/scenes/ui/LevelCount.gd
index dac77d6..ecb4c5b 100644
--- a/scenes/ui/LevelCount.gd
+++ b/scenes/ui/LevelCount.gd
@@ -1,21 +1,21 @@
 extends RichTextLabel
 
-const VisibleTime = 2
-var currentLevel := 1
+export(float) var visible_time: float = 2.0
+var current_level: int = 1
 
 
-func _ready():
+func _ready() -> void:
 	EventBus.connect("level_started", self, "_on_level_started")
 	EventBus.connect("level_completed", self, "_on_level_completed")
-	_on_level_started({})
+	_on_level_started("")
 
 
-func _on_level_started(_data):
+func _on_level_started(_name: String) -> void:
 	self.show()
-	bbcode_text = "\n" + (tr("[wave amp=50 freq=2]LEVEL:[rainbow freq=0.5 sat=1 val=20]%d[/rainbow] [/wave]") % currentLevel)
-	yield(get_tree().create_timer(VisibleTime), "timeout")
+	bbcode_text = "\n" + (tr("[wave amp=50 freq=2]LEVEL:[rainbow freq=0.5 sat=1 val=20]%d[/rainbow] [/wave]") % current_level)
+	yield(get_tree().create_timer(visible_time), "timeout")
 	self.hide()
 
 
-func _on_level_completed(_data):
-	currentLevel += 1
+func _on_level_completed(_data) -> void:
+	current_level += 1
diff --git a/scripts/Food.gd b/scripts/Food.gd
index a3a4788..e349f04 100644
--- a/scripts/Food.gd
+++ b/scripts/Food.gd
@@ -26,7 +26,6 @@ func _physics_process(delta: float):
 	$Sprite.rotation_degrees += vel_rotation
 	var collision = move_and_collide(Vector2(0, vel_y * delta))
 	if collision:
-		var collider = collision.collider as Object
 		if collision.collider as TileMap:
 			randomize_me()
 		elif collision.collider.name == "Player" and is_pie:
diff --git a/scripts/Main.gd b/scripts/Main.gd
index fe8546a..036206c 100644
--- a/scripts/Main.gd
+++ b/scripts/Main.gd
@@ -89,12 +89,15 @@ func _on_endportal_body_entered(body: Node2D, next_level: PackedScene, portal: E
 	for despawn in get_tree().get_nodes_in_group(PROJECTILES_GROUP):
 		despawn.queue_free()
 
-	var animation: AnimationPlayer = portal.on_portal_enter(body)
-	if animation == null:
-		return
 	body.get_parent().remove_child(body)
 
-	yield(animation, "animation_finished")
+	EventBus.emit_signal("level_completed", {})
+
+	var animation: AnimationPlayer = portal.on_portal_enter(body)
+	if animation == null:
+		yield(get_tree().create_timer(1.0), "timeout")
+	else:
+		yield(animation, "animation_finished")
 	call_deferred("_finish_level", next_level)
 
 
@@ -118,4 +121,4 @@ func _finish_level(next_level: PackedScene = null) -> void:
 
 	# Reset entering portal state
 	entering_portal = false
-	EventBus.emit_signal("level_started", {})
+	EventBus.emit_signal("level_started", "")
diff --git a/scripts/autoload/EventBus.gd b/scripts/autoload/EventBus.gd
index ddbc485..4693350 100644
--- a/scripts/autoload/EventBus.gd
+++ b/scripts/autoload/EventBus.gd
@@ -1,55 +1,59 @@
 #warning-ignore-all: UNUSED_SIGNAL
 extends Node
 
-signal jumping(data)
-signal coin_collected(data)
-signal star_collected(starname, again)
+# Player
+signal build_block(data)
 signal heart_changed(delta, total)
+signal jumping()
+signal player_died()
 signal shot()
-signal enemy_killed(data)
-signal build_block(data)
+
+# Inventory
+signal bus_collected(data)
+signal coin_collected(data)
+signal fire_flower_collected(data)
+signal star_collected(starname, again)
+
+# Enemies
+signal enemy_hit_coin()
+signal enemy_hit_fireball()
+signal enemy_killed()
+signal player_spotted(spotted_by, player)
+
+# Levels
 signal level_completed(data)
-signal level_started(data)
-signal player_died
+signal level_started(name)
 
-# The "stream" key will set the background music and
-# the "playing" will set whenever the music is playing
+# Audio
+# The "stream" key will set the background music and the "playing" will set whenever the music is playing
 signal bgm_changed(data)
 
+# UI
 # Expects a "visible" boolean key
 signal ui_visibility_changed(data)
+signal game_paused(pause_state)
 
-# Scene Transitions, expects "scene" key
+# Scenes
+# Expects a "scene" key
 signal change_scene(data)
+signal game_exit()
 
-# screenshakes
-signal small_screen_shake
-signal medium_screen_shake
-signal large_screen_shake
+# Screenshakes
+signal large_screen_shake()
+signal medium_screen_shake()
+signal small_screen_shake()
 
 # Camera follow
-signal cameraF_candidate_spawned
-signal cameraF_update_current_camera
-signal cameraF_reset_camera
-signal cameraF_set_following
-signal cameraF_move_camera_to(x,y)
+signal cameraF_candidate_spawned()
+signal cameraF_update_current_camera()
+signal cameraF_reset_camera()
+signal cameraF_set_following()
+signal cameraF_move_camera_to(x, y)
 signal cameraF_change_candidate(step_int)
 
-# pauses the game - used by PauseMenu.gd
-signal game_paused(data)
-# called once on game ends (or is restarted)
-signal game_exit
-
-## settings signals
+# Settings
 signal crt_filter_toggle(data)
 signal volume_changed(data)
 
-signal enemy_hit_coin
-signal enemy_hit_fireball
-
-signal fire_flower_collected(data)
-signal bus_collected(data)
-
-signal player_spotted(spotted_by, player)
-
+# Environmental
 signal big_button_pressed(button_id)
