diff --git a/godot/local_map/grid/GameBoard.gd b/godot/local_map/grid/GameBoard.gd
index b1d3b5d..0a4d44a 100644
--- a/godot/local_map/grid/GameBoard.gd
+++ b/godot/local_map/grid/GameBoard.gd
@@ -47,7 +47,7 @@ func find_path(start_world_position, end_world_position : Vector2) -> PoolVector
 	Returns an array of grid points that connect the start and end world position
 	"""
 	var end = world_to_map(end_world_position)
-	if get_cellv(end) == CELL_TYPES.OBSTACLE:
+	if get_cellv(end) != CELL_TYPES.EMPTY:
 		return PoolVector3Array()
 	var start = world_to_map(start_world_position)
 	return pathfinder.find_path(start, end)
