diff --git a/game/Scripts/Game/UI/Previews/ObjectPreview.gd b/game/Scripts/Game/UI/Previews/ObjectPreview.gd
index b044da8..647185b 100644
--- a/game/Scripts/Game/UI/Previews/ObjectPreview.gd
+++ b/game/Scripts/Game/UI/Previews/ObjectPreview.gd
@@ -221,7 +221,13 @@ func _remove_piece() -> void:
 	# case one escapes the _piece reference.
 	for child in _viewport.get_children():
 		if child is RigidBody:
-			_viewport.remove_child(child)
+			# Removing the piece from the scene tree early sometimes causes
+			# errors when the _physics_process notification fires, and it is
+			# still expecting the piece to be in the scene tree.
+			# TODO: Maybe stop removing children from the scene tree before
+			# calling queue_free() on them?
+			#_viewport.remove_child(child)
+			
 			ResourceManager.queue_free_object(child)
 	
 	_piece = null
diff --git a/game/Scripts/ResourceManager.gd b/game/Scripts/ResourceManager.gd
index 929c6a9..506c518 100644
--- a/game/Scripts/ResourceManager.gd
+++ b/game/Scripts/ResourceManager.gd
@@ -41,6 +41,11 @@ var _stex_refs = {}
 # path: The path of the resource to load.
 func load_res(path: String) -> Resource:
 	_res_mutex.lock()
+	
+	# Workaround for this issue:
+	# https://github.com/godotengine/godot/issues/55566
+	OS.delay_msec(1)
+	
 	var res: Resource = load(path)
 	
 	# On macOS, I came across a bug in GLES3 where a StreamTexture would be
