diff --git a/Scripts/Classes/Entities/Items/Coin.gd b/Scripts/Classes/Entities/Items/Coin.gd
index 2ea7840..4940485 100644
--- a/Scripts/Classes/Entities/Items/Coin.gd
+++ b/Scripts/Classes/Entities/Items/Coin.gd
@@ -24,8 +24,8 @@ func collect() -> void:
 		$Sprite.queue_free()
 	else:
 		queue_free()
-	if get_parent() is TileMapLayer:
-		get_parent().erase_cell(get_parent().local_to_map(position))
+		if get_parent() is TileMapLayer:
+			get_parent().erase_cell(get_parent().local_to_map(position))
 
 func summon_block_coin() -> void:
 	var node = spinning_coin_scene.instantiate()
