diff --git a/Scripts/Classes/Entities/Items/Fireball.gd b/Scripts/Classes/Entities/Items/Fireball.gd
index 528200b..4929251 100644
--- a/Scripts/Classes/Entities/Items/Fireball.gd
+++ b/Scripts/Classes/Entities/Items/Fireball.gd
@@ -17,8 +17,8 @@ func _physics_process(delta: float) -> void:
 	velocity.y += (15 / delta) * delta
 	velocity.y = clamp(velocity.y, -INF, 150)
 	if is_on_floor():
-		velocity.y = -150
-	if is_on_wall() or is_on_ceiling():
+		velocity.y = -125
+	if is_on_wall() or (abs(get_floor_normal().x) > 0 and is_on_ceiling()):
 		hit()
 	move_and_slide()
 
