diff --git a/scenes/enemies/common/ai/conditionals/is_in_combat/IsInCombat.gd b/scenes/enemies/common/ai/conditionals/is_in_combat/IsInCombat.gd
index 10e31ef..0eb0fd9 100644
--- a/scenes/enemies/common/ai/conditionals/is_in_combat/IsInCombat.gd
+++ b/scenes/enemies/common/ai/conditionals/is_in_combat/IsInCombat.gd
@@ -10,6 +10,8 @@ var target: JBody2D = null
 
 
 func tick(actor: Node, blackboard: Blackboard):
+	if blackboard.get_value("is_resetting"):
+		return FAILURE
 	if not reset_timer.timeout.is_connected(_on_reset_timer_timeout):
 		reset_timer.timeout.connect(_on_reset_timer_timeout)
 	if aggro_radius != null:
@@ -17,24 +19,35 @@ func tick(actor: Node, blackboard: Blackboard):
 			aggro_radius.body_entered.connect(_on_body_entered)
 	if not actor.synchronizer.got_hurt.is_connected(_on_got_hurt):
 		actor.synchronizer.got_hurt.connect(_on_got_hurt)
+	if not actor.synchronizer.died.is_connected(_on_died):
+		actor.synchronizer.died.connect(_on_died)
 	if start_combat:
 		in_combat = true
 		start_combat = false
 		reset_timer.start()
+		blackboard.set_value("reset_timer", reset_timer)
 		blackboard.set_value("aggro_target", target)
 		blackboard.set_value("leash_position", actor.global_position)
 		if aggro_radius != null:
 			aggro_radius.body_entered.disconnect(_on_body_entered)
 	if in_combat:
 		var leash_position: Vector2 = blackboard.get_value("leash_position")
-		if should_leash and leash_position.distance_to(actor.global_position) >= leash_distance:
+		if should_leash and (leash_position.distance_to(actor.global_position) >= leash_distance):
+			in_combat = false
 			return FAILURE
 		return SUCCESS
 	return FAILURE
 
 
-func _on_reset_timer_timeout():
+func _on_body_entered(body: Node2D):
+	if target == null:
+		start_combat = true
+		target = body
+
+
+func _on_died():
 	in_combat = false
+	start_combat = false
 	target = null
 
 
@@ -46,7 +59,6 @@ func _on_got_hurt(from: String, _hp: int, _max_hp: int, _damage: int):
 		target = get_node("/root/Root/World/JEntities/JPlayers/" + from)
 
 
-func _on_body_entered(body: Node2D):
-	if target == null:
-		start_combat = true
-		target = body
+func _on_reset_timer_timeout():
+	in_combat = false
+	target = null
diff --git a/scenes/enemies/common/ai/sequences/attack_sequence/Attack.gd b/scenes/enemies/common/ai/sequences/attack_sequence/Attack.gd
index e08f886..089982f 100644
--- a/scenes/enemies/common/ai/sequences/attack_sequence/Attack.gd
+++ b/scenes/enemies/common/ai/sequences/attack_sequence/Attack.gd
@@ -13,6 +13,8 @@ func tick(actor: Node, blackboard: Blackboard):
 		swing_timer.start()
 		ready_to_attack = false
 		var target: JBody2D = blackboard.get_value("aggro_target")
+		var reset_timer: Timer = blackboard.get_value("reset_timer")
+		reset_timer.start()
 		actor.attack(target)
 		return SUCCESS
 	return FAILURE
diff --git a/scenes/enemies/common/ai/sequences/leash_sequence/ShouldLeash.gd b/scenes/enemies/common/ai/sequences/leash_sequence/ShouldLeash.gd
index 796c40b..7ebf04a 100644
--- a/scenes/enemies/common/ai/sequences/leash_sequence/ShouldLeash.gd
+++ b/scenes/enemies/common/ai/sequences/leash_sequence/ShouldLeash.gd
@@ -1,10 +1,25 @@
 extends ConditionLeaf
 
+var is_resetting := false
 
-func tick(_actor: Node, blackboard: Blackboard):
+
+func tick(actor: Node, blackboard: Blackboard):
 	if blackboard.has_value("leash_position"):
+		if not actor.destination_reached.is_connected(_destination_reached):
+			actor.destination_reached.connect(_destination_reached)
+		J.logger.info("Should leash, erasing leash_position")
 		var leash_position: Vector2 = blackboard.get_value("leash_position")
 		blackboard.set_value("destination_global_position", leash_position)
 		blackboard.erase_value("leash_position")
+		is_resetting = true
+		blackboard.set_value("is_resetting", is_resetting)
+		return SUCCESS
+	if is_resetting:
 		return SUCCESS
+	if not is_resetting:
+		blackboard.erase_value("is_resetting")
 	return FAILURE
+
+
+func _destination_reached():
+	is_resetting = false
