diff --git a/scenes/levels/Stardepths/mega_arena.tscn b/scenes/levels/Stardepths/mega_arena.tscn
index 5b66489..2a5ac0a 100644
--- a/scenes/levels/Stardepths/mega_arena.tscn
+++ b/scenes/levels/Stardepths/mega_arena.tscn
@@ -1,4 +1,4 @@
-[gd_scene load_steps=9 format=2]
+[gd_scene load_steps=10 format=2]
 
 [ext_resource path="res://scripts/resources/stardepths_tileset.tres" type="TileSet" id=1]
 [ext_resource path="res://scenes/bosses/MegaBoss.tscn" type="PackedScene" id=2]
@@ -8,6 +8,7 @@
 [ext_resource path="res://scenes/bosses/MegaBossCut.tscn" type="PackedScene" id=6]
 [ext_resource path="res://scenes/EndPortal.tscn" type="PackedScene" id=7]
 [ext_resource path="res://scripts/objects/PortalManager.gd" type="Script" id=8]
+[ext_resource path="res://scenes/objects/CameraLimits.tscn" type="PackedScene" id=9]
 
 [node name="TileMap" type="TileMap"]
 tile_set = ExtResource( 1 )
@@ -92,6 +93,8 @@ script = ExtResource( 8 )
 visible = false
 position = Vector2( 512, 160 )
 
+[node name="CameraLimits" parent="." instance=ExtResource( 9 )]
+
 [connection signal="display_warning" from="BossNode2D/MegaBoss" to="Warnings/Warning" method="_on_display_warning"]
 [connection signal="display_warning" from="BossNode2D/MegaBoss" to="Warnings/Warning2" method="_on_display_warning"]
 [connection signal="display_warning" from="BossNode2D/MegaBoss" to="Warnings/Warning3" method="_on_display_warning"]
