diff --git a/.gitignore b/.gitignore
index ec708d6..ec8cffd 100644
--- a/.gitignore
+++ b/.gitignore
@@ -1,4 +1,5 @@
 # Godot 4+ specific ignores
 .godot/
 .env
-data/users.json
\ No newline at end of file
+data/users.json
+*csproj.old
diff --git a/Jdungeon.csproj b/Jdungeon.csproj
index 02709f4..d1efbb8 100644
--- a/Jdungeon.csproj
+++ b/Jdungeon.csproj
@@ -1,10 +1,10 @@
-<Project Sdk="Godot.NET.Sdk/4.1.1">
+<Project Sdk="Godot.NET.Sdk/4.1.2">
   <PropertyGroup>
     <TargetFramework>net6.0</TargetFramework>
     <EnableDynamicLoading>true</EnableDynamicLoading>
   </PropertyGroup>
   <ItemGroup>
-    <PackageReference Include="Npgsql" Version="7.0.6"/>
+    <PackageReference Include="Npgsql" Version="7.0.6" />
     <PackageReference Include="BCrypt.Net-Next" Version="4.0.3" />
   </ItemGroup>
 </Project>
\ No newline at end of file
diff --git a/scenes/enemies/Sheep/Sheep.gd b/scenes/enemies/Sheep/Sheep.gd
index 3016ebe..f7e1929 100644
--- a/scenes/enemies/Sheep/Sheep.gd
+++ b/scenes/enemies/Sheep/Sheep.gd
@@ -20,9 +20,7 @@ var movement_multiplier := 1.0
 
 func _init():
 	super()
-
 	enemy_class = "Sheep"
-
 	stats.movement_speed = 50
 	stats.max_hp = 20
 	stats.hp = stats.max_hp
@@ -32,12 +30,9 @@ func _ready():
 	# Make sure to connect to all signals before super is called
 	if not J.is_server() and J.client.player:
 		stats.synced.connect(_on_stats_synced)
-
 	super()
-
 	if J.is_server():
 		beehave_tree.enabled = true
-
 	synchronizer.loop_animation_changed.connect(_on_loop_animation_changed)
 	synchronizer.got_hurt.connect(_on_got_hurt)
 	synchronizer.died.connect(_on_died)
diff --git a/scenes/enemies/TreeTrunkGuy/TreeTrunkGuy.gd b/scenes/enemies/TreeTrunkGuy/TreeTrunkGuy.gd
index 8dda782..0bee975 100644
--- a/scenes/enemies/TreeTrunkGuy/TreeTrunkGuy.gd
+++ b/scenes/enemies/TreeTrunkGuy/TreeTrunkGuy.gd
@@ -21,8 +21,6 @@ var movement_multiplier := 1.0
 
 func _init():
 	super()
-	if J.is_server():
-		beehave_tree.enabled = true
 	enemy_class = "TreeTrunkGuy"
 	stats.movement_speed = 150
 	stats.experience_given = 50
@@ -32,9 +30,9 @@ func _ready():
 	# Make sure to connect to all signals before super is called
 	if not J.is_server() and J.client.player:
 		stats.synced.connect(_on_stats_synced)
-
 	super()
-
+	if J.is_server():
+		beehave_tree.enabled = true
 	stats.max_hp = 50
 	synchronizer.loop_animation_changed.connect(_on_loop_animation_changed)
 	synchronizer.got_hurt.connect(_on_got_hurt)
