diff --git a/game/Scripts/Game/Pieces/Stack.gd b/game/Scripts/Game/Pieces/Stack.gd
index f3999c9..83c317a 100644
--- a/game/Scripts/Game/Pieces/Stack.gd
+++ b/game/Scripts/Game/Pieces/Stack.gd
@@ -172,14 +172,14 @@ func add_piece(piece_entry: Dictionary, piece_transform: Transform,
 	
 	mass += piece_entry["mass"]
 	
+	_add_piece_at_pos(piece_entry, pos, flip_y)
+	
 	if is_card_stack():
 		if card_add_sounds != null:
 			play_effect(card_add_sounds.random_stream())
 	else:
 		if chip_add_sounds != null:
 			play_effect(chip_add_sounds.random_stream())
-	
-	_add_piece_at_pos(piece_entry, pos, flip_y)
 
 # Check if the stack is empty.
 # Returns: If the stack is empty.
diff --git a/game/Scripts/PieceBuilder.gd b/game/Scripts/PieceBuilder.gd
index 4784198..2aacc16 100644
--- a/game/Scripts/PieceBuilder.gd
+++ b/game/Scripts/PieceBuilder.gd
@@ -188,8 +188,11 @@ func fill_stack(stack: Stack, stack_entry: Dictionary) -> void:
 		# bit... dirty? There's got to be a better way of doing this :O
 		var asset = texture_path.get_file().get_basename()
 		var type_dir = texture_path.get_base_dir()
-		var type = type_dir.get_file()
-		var pack = type_dir.get_base_dir().get_file()
+		if type_dir.begins_with("user://assets/"):
+			type_dir = type_dir.substr(14)
+		var type_split = type_dir.split("/", false, 1)
+		var pack = type_split[0]
+		var type = type_split[1]
 		var entry_path = "%s/%s/%s" % [pack, type, asset]
 		
 		var piece_entry = AssetDB.search_path(entry_path)
