diff --git a/data/tileset/atlas.tres b/data/tileset/atlas.tres
index 59fa86b..fd00e61 100644
--- a/data/tileset/atlas.tres
+++ b/data/tileset/atlas.tres
@@ -12293,6 +12293,7 @@ texture_region_size = Vector2i(32, 32)
 4:29/0/y_sort_origin = -9
 4:29/0/physics_layer_0/linear_velocity = Vector2(0, 0)
 4:29/0/physics_layer_0/angular_velocity = 0.0
+
 [sub_resource type="TileSetAtlasSource" id="TileSetAtlasSource_5l4kn"]
 texture = ExtResource("6_my7h2")
 texture_region_size = Vector2i(32, 32)
diff --git a/scenes/terrain/areas/harbor/harbor.tscn b/scenes/terrain/areas/harbor/harbor.tscn
index 690367f..d17040e 100644
--- a/scenes/terrain/areas/harbor/harbor.tscn
+++ b/scenes/terrain/areas/harbor/harbor.tscn
@@ -3,8 +3,10 @@
 [ext_resource type="TileSet" uid="uid://pjsyexx3ov10" path="res://data/tileset/atlas.tres" id="1_kpsqp"]
 
 [node name="Harbor" type="Node2D"]
+y_sort_enabled = true
 
 [node name="Ground" type="TileMap" parent="."]
+z_index = -1
 scale = Vector2(2, 2)
 tile_set = ExtResource("1_kpsqp")
 format = 2
diff --git a/scenes/world/World.tscn b/scenes/world/World.tscn
index d41ee7a..e4e89df 100644
--- a/scenes/world/World.tscn
+++ b/scenes/world/World.tscn
@@ -1,5 +1,4 @@
-
-[gd_scene load_steps=11 format=3 uid="uid://beiq65may6bdw"]
+[gd_scene load_steps=12 format=3 uid="uid://beiq65may6bdw"]
 
 [ext_resource type="Script" path="res://scenes/world/World.gd" id="1_3w1ia"]
 [ext_resource type="PackedScene" uid="uid://bdgegw7cy4d1r" path="res://scenes/ui/loginpanel/LoginPanel.tscn" id="2_fjr74"]
@@ -9,7 +8,6 @@
 [ext_resource type="PackedScene" uid="uid://dh827s71bdr6o" path="res://scenes/ui/versioncheckpanel/VersionCheckPanel.tscn" id="4_5aijd"]
 [ext_resource type="PackedScene" uid="uid://dfq5lb87meq6q" path="res://scenes/enemies/Sheep/Sheep.tscn" id="4_uubdx"]
 [ext_resource type="PackedScene" uid="uid://2j64d7cd4644" path="res://scenes/terrain/areas/basecamp/BaseCamp.tscn" id="5_0b0pa"]
-[ext_resource type="PackedScene" uid="uid://drmaubupv57n1" path="res://scenes/enemies/MoldedDruvar/MoldedDruvar.tscn" id="6_17gjh"]
 [ext_resource type="PackedScene" uid="uid://bija15ujapx3h" path="res://scenes/terrain/areas/wakingforest/WakingForest.tscn" id="7_oodd1"]
 [ext_resource type="PackedScene" uid="uid://cpw0d00b7prm0" path="res://scenes/terrain/areas/harbor/harbor.tscn" id="9_70hw5"]
 [ext_resource type="PackedScene" uid="uid://cflqdnl6156j8" path="res://scenes/player/playersrepsawnlocation/PlayerRespawnLocation.tscn" id="12_7jhmb"]
@@ -39,98 +37,60 @@ visible = false
 
 [node name="TreeTrunkGuy" parent="Entities/Enemies" instance=ExtResource("3_fddk3")]
 position = Vector2(-1894, 844)
-should_respawn = true
-respawn_time = 20.0
 
 [node name="TreeTrunkGuy2" parent="Entities/Enemies" instance=ExtResource("3_fddk3")]
 position = Vector2(-2669, 42)
-should_respawn = true
-respawn_time = 20.0
 
 [node name="TreeTrunkGuy3" parent="Entities/Enemies" instance=ExtResource("3_fddk3")]
 position = Vector2(-3439, 870)
-should_respawn = true
-respawn_time = 20.0
 
 [node name="TreeTrunkGuy8" parent="Entities/Enemies" instance=ExtResource("3_fddk3")]
 position = Vector2(-2713, 1314)
-should_respawn = true
-respawn_time = 20.0
 
 [node name="TreeTrunkGuy9" parent="Entities/Enemies" instance=ExtResource("3_fddk3")]
 position = Vector2(-4428, 1212)
-should_respawn = true
-respawn_time = 20.0
 
 [node name="TreeTrunkGuy18" parent="Entities/Enemies" instance=ExtResource("3_fddk3")]
 position = Vector2(-4590, 394)
-should_respawn = true
-respawn_time = 20.0
 
 [node name="TreeTrunkGuy19" parent="Entities/Enemies" instance=ExtResource("3_fddk3")]
 position = Vector2(-4098, -1332)
-should_respawn = true
-respawn_time = 20.0
 
 [node name="TreeTrunkGuy20" parent="Entities/Enemies" instance=ExtResource("3_fddk3")]
 position = Vector2(-2870, -499)
-should_respawn = true
-respawn_time = 20.0
 
 [node name="TreeTrunkGuy10" parent="Entities/Enemies" instance=ExtResource("3_fddk3")]
 position = Vector2(-5580, 1135)
-should_respawn = true
-respawn_time = 20.0
 
 [node name="TreeTrunkGuy11" parent="Entities/Enemies" instance=ExtResource("3_fddk3")]
 position = Vector2(-5160, 229)
-should_respawn = true
-respawn_time = 20.0
 
 [node name="TreeTrunkGuy13" parent="Entities/Enemies" instance=ExtResource("3_fddk3")]
 position = Vector2(-5644, -639)
-should_respawn = true
-respawn_time = 20.0
 
 [node name="TreeTrunkGuy14" parent="Entities/Enemies" instance=ExtResource("3_fddk3")]
 position = Vector2(-5015, -1210)
-should_respawn = true
-respawn_time = 20.0
 
 [node name="TreeTrunkGuy15" parent="Entities/Enemies" instance=ExtResource("3_fddk3")]
 position = Vector2(-4900, -313)
-should_respawn = true
-respawn_time = 20.0
 
 [node name="TreeTrunkGuy16" parent="Entities/Enemies" instance=ExtResource("3_fddk3")]
 position = Vector2(-4382, -816)
-should_respawn = true
-respawn_time = 20.0
 
 [node name="TreeTrunkGuy17" parent="Entities/Enemies" instance=ExtResource("3_fddk3")]
 position = Vector2(-3405, -1229)
-should_respawn = true
-respawn_time = 20.0
 
 [node name="TreeTrunkGuy12" parent="Entities/Enemies" instance=ExtResource("3_fddk3")]
 position = Vector2(-5173, 768)
-should_respawn = true
-respawn_time = 20.0
 
 [node name="TreeTrunkGuy4" parent="Entities/Enemies" instance=ExtResource("3_fddk3")]
 position = Vector2(-2178, -701)
-should_respawn = true
-respawn_time = 20.0
 
 [node name="TreeTrunkGuy6" parent="Entities/Enemies" instance=ExtResource("3_fddk3")]
 position = Vector2(-1882, -1198)
-should_respawn = true
-respawn_time = 20.0
 
 [node name="TreeTrunkGuy7" parent="Entities/Enemies" instance=ExtResource("3_fddk3")]
 position = Vector2(-2438, -1397)
-should_respawn = true
-respawn_time = 20.0
 
 [node name="Sheep6" parent="Entities/Enemies" instance=ExtResource("4_uubdx")]
 position = Vector2(-1099, -418)
