diff --git a/scenes/Player.tscn b/scenes/Player.tscn
index 8f52335..f8dc524 100644
--- a/scenes/Player.tscn
+++ b/scenes/Player.tscn
@@ -136,3 +136,4 @@ flip_h = true
 visible = false
 position = Vector2( 6.75, 11.1875 )
 shape = SubResource( 13 )
+disabled = true
diff --git a/scripts/BallPlayer.gd b/scripts/BallPlayer.gd
index 12df1cc..34cd211 100644
--- a/scripts/BallPlayer.gd
+++ b/scripts/BallPlayer.gd
@@ -36,7 +36,7 @@ func _physics_process(_delta: float) -> void:
 		max_speed_modifier = 1.5
 		acceleration_modifier = 3
 		animationSpeed = 60
-	sprite.frames.set_animation_speed("run", animationSpeed)
+	#sprite.frames.set_animation_speed("run", animationSpeed)
 
 	if Input.is_action_pressed("right"):
 		motion.x += ACCEL * acceleration_modifier
@@ -54,7 +54,7 @@ func _physics_process(_delta: float) -> void:
 		
 		"""
 	else:
-		sprite.play("idle")
+		#sprite.play("idle")
 		motion.x = lerp(motion.x, 0, 0.05)
 
 	if Input.is_action_pressed("up"):
diff --git a/scripts/Player.gd b/scripts/Player.gd
index eccf511..6a70349 100644
--- a/scripts/Player.gd
+++ b/scripts/Player.gd
@@ -346,7 +346,7 @@ func _on_bus_collected(data):
 	if data.has("collected"):
 		isBus = data["collected"]
 		$BusSprite.visible = true
-		$BusCollision.visible = true
+		$BusCollision.set_deferred("disabled", false)
 		sprite.visible = false
-		$CollisionShape2D.visible = false
+		$CollisionShape2D.set_deferred("disabled", true)
 		trail.height = 15
