diff --git a/Scenes/Levels/SMBLL/World2/2-2.tscn b/Scenes/Levels/SMBLL/World2/2-2.tscn
index 7031f65..952cf42 100644
--- a/Scenes/Levels/SMBLL/World2/2-2.tscn
+++ b/Scenes/Levels/SMBLL/World2/2-2.tscn
@@ -268,85 +268,85 @@ position = Vector2(2712, -120)
 position = Vector2(2728, -72)
 
 [connection signal="collected" from="Tiles/Coin" to="Tiles/Coin/LevelPersistance" method="set_as_active"]
-[connection signal="collected" from="Tiles/@Node2D@92146" to="Tiles/@Node2D@92146/LevelPersistance" method="set_as_active"]
-[connection signal="collected" from="Tiles/@Node2D@92147" to="Tiles/@Node2D@92147/LevelPersistance" method="set_as_active"]
-[connection signal="collected" from="Tiles/@Node2D@92148" to="Tiles/@Node2D@92148/LevelPersistance" method="set_as_active"]
-[connection signal="collected" from="Tiles/@Node2D@92149" to="Tiles/@Node2D@92149/LevelPersistance" method="set_as_active"]
-[connection signal="collected" from="Tiles/@Node2D@92150" to="Tiles/@Node2D@92150/LevelPersistance" method="set_as_active"]
+[connection signal="collected" from="Tiles/@Node2D@47021" to="Tiles/@Node2D@47021/LevelPersistance" method="set_as_active"]
+[connection signal="collected" from="Tiles/@Node2D@47022" to="Tiles/@Node2D@47022/LevelPersistance" method="set_as_active"]
+[connection signal="collected" from="Tiles/@Node2D@47023" to="Tiles/@Node2D@47023/LevelPersistance" method="set_as_active"]
+[connection signal="collected" from="Tiles/@Node2D@47024" to="Tiles/@Node2D@47024/LevelPersistance" method="set_as_active"]
+[connection signal="collected" from="Tiles/@Node2D@47025" to="Tiles/@Node2D@47025/LevelPersistance" method="set_as_active"]
 [connection signal="block_destroyed" from="Tiles/BrickBlock" to="Tiles/BrickBlock/LevelPersistance" method="set_as_active_2"]
 [connection signal="block_emptied" from="Tiles/BrickBlock" to="Tiles/BrickBlock/LevelPersistance" method="set_as_active"]
 [connection signal="player_block_hit" from="Tiles/BrickBlock" to="Tiles/BrickBlock" method="on_block_hit"]
 [connection signal="player_block_hit" from="Tiles/BrickBlock" to="Tiles/BrickBlock/BlockAnimations" method="bounce_block" unbinds=1]
 [connection signal="shell_block_hit" from="Tiles/BrickBlock" to="Tiles/BrickBlock" method="on_shell_block_hit"]
 [connection signal="shell_block_hit" from="Tiles/BrickBlock" to="Tiles/BrickBlock/BlockAnimations" method="bounce_block" unbinds=1]
-[connection signal="collected" from="Tiles/@Node2D@92151" to="Tiles/@Node2D@92151/LevelPersistance" method="set_as_active"]
-[connection signal="collected" from="Tiles/@Node2D@92152" to="Tiles/@Node2D@92152/LevelPersistance" method="set_as_active"]
-[connection signal="block_destroyed" from="Tiles/@AnimatableBody2D@92153" to="Tiles/@AnimatableBody2D@92153/LevelPersistance" method="set_as_active_2"]
-[connection signal="block_emptied" from="Tiles/@AnimatableBody2D@92153" to="Tiles/@AnimatableBody2D@92153/LevelPersistance" method="set_as_active"]
-[connection signal="player_block_hit" from="Tiles/@AnimatableBody2D@92153" to="Tiles/@AnimatableBody2D@92153" method="on_block_hit"]
-[connection signal="player_block_hit" from="Tiles/@AnimatableBody2D@92153" to="Tiles/@AnimatableBody2D@92153/BlockAnimations" method="bounce_block" unbinds=1]
-[connection signal="shell_block_hit" from="Tiles/@AnimatableBody2D@92153" to="Tiles/@AnimatableBody2D@92153" method="on_shell_block_hit"]
-[connection signal="shell_block_hit" from="Tiles/@AnimatableBody2D@92153" to="Tiles/@AnimatableBody2D@92153/BlockAnimations" method="bounce_block" unbinds=1]
-[connection signal="collected" from="Tiles/@Node2D@92154" to="Tiles/@Node2D@92154/LevelPersistance" method="set_as_active"]
-[connection signal="collected" from="Tiles/@Node2D@92155" to="Tiles/@Node2D@92155/LevelPersistance" method="set_as_active"]
-[connection signal="collected" from="Tiles/@Node2D@92156" to="Tiles/@Node2D@92156/LevelPersistance" method="set_as_active"]
-[connection signal="collected" from="Tiles/@Node2D@92157" to="Tiles/@Node2D@92157/LevelPersistance" method="set_as_active"]
-[connection signal="collected" from="Tiles/@Node2D@92158" to="Tiles/@Node2D@92158/LevelPersistance" method="set_as_active"]
-[connection signal="collected" from="Tiles/@Node2D@92159" to="Tiles/@Node2D@92159/LevelPersistance" method="set_as_active"]
-[connection signal="collected" from="Tiles/@Node2D@92160" to="Tiles/@Node2D@92160/LevelPersistance" method="set_as_active"]
-[connection signal="collected" from="Tiles/@Node2D@92161" to="Tiles/@Node2D@92161/LevelPersistance" method="set_as_active"]
-[connection signal="collected" from="Tiles/@Node2D@92162" to="Tiles/@Node2D@92162/LevelPersistance" method="set_as_active"]
-[connection signal="collected" from="Tiles/@Node2D@92163" to="Tiles/@Node2D@92163/LevelPersistance" method="set_as_active"]
-[connection signal="collected" from="Tiles/@Node2D@92164" to="Tiles/@Node2D@92164/LevelPersistance" method="set_as_active"]
-[connection signal="collected" from="Tiles/@Node2D@92165" to="Tiles/@Node2D@92165/LevelPersistance" method="set_as_active"]
-[connection signal="collected" from="Tiles/@Node2D@92166" to="Tiles/@Node2D@92166/LevelPersistance" method="set_as_active"]
+[connection signal="collected" from="Tiles/@Node2D@47026" to="Tiles/@Node2D@47026/LevelPersistance" method="set_as_active"]
+[connection signal="collected" from="Tiles/@Node2D@47027" to="Tiles/@Node2D@47027/LevelPersistance" method="set_as_active"]
+[connection signal="block_destroyed" from="Tiles/@AnimatableBody2D@47028" to="Tiles/@AnimatableBody2D@47028/LevelPersistance" method="set_as_active_2"]
+[connection signal="block_emptied" from="Tiles/@AnimatableBody2D@47028" to="Tiles/@AnimatableBody2D@47028/LevelPersistance" method="set_as_active"]
+[connection signal="player_block_hit" from="Tiles/@AnimatableBody2D@47028" to="Tiles/@AnimatableBody2D@47028" method="on_block_hit"]
+[connection signal="player_block_hit" from="Tiles/@AnimatableBody2D@47028" to="Tiles/@AnimatableBody2D@47028/BlockAnimations" method="bounce_block" unbinds=1]
+[connection signal="shell_block_hit" from="Tiles/@AnimatableBody2D@47028" to="Tiles/@AnimatableBody2D@47028" method="on_shell_block_hit"]
+[connection signal="shell_block_hit" from="Tiles/@AnimatableBody2D@47028" to="Tiles/@AnimatableBody2D@47028/BlockAnimations" method="bounce_block" unbinds=1]
+[connection signal="collected" from="Tiles/@Node2D@47029" to="Tiles/@Node2D@47029/LevelPersistance" method="set_as_active"]
+[connection signal="collected" from="Tiles/@Node2D@47030" to="Tiles/@Node2D@47030/LevelPersistance" method="set_as_active"]
+[connection signal="collected" from="Tiles/@Node2D@47031" to="Tiles/@Node2D@47031/LevelPersistance" method="set_as_active"]
+[connection signal="collected" from="Tiles/@Node2D@47032" to="Tiles/@Node2D@47032/LevelPersistance" method="set_as_active"]
+[connection signal="collected" from="Tiles/@Node2D@47033" to="Tiles/@Node2D@47033/LevelPersistance" method="set_as_active"]
+[connection signal="collected" from="Tiles/@Node2D@47034" to="Tiles/@Node2D@47034/LevelPersistance" method="set_as_active"]
+[connection signal="collected" from="Tiles/@Node2D@47035" to="Tiles/@Node2D@47035/LevelPersistance" method="set_as_active"]
+[connection signal="collected" from="Tiles/@Node2D@47036" to="Tiles/@Node2D@47036/LevelPersistance" method="set_as_active"]
+[connection signal="collected" from="Tiles/@Node2D@47037" to="Tiles/@Node2D@47037/LevelPersistance" method="set_as_active"]
+[connection signal="collected" from="Tiles/@Node2D@47038" to="Tiles/@Node2D@47038/LevelPersistance" method="set_as_active"]
+[connection signal="collected" from="Tiles/@Node2D@47039" to="Tiles/@Node2D@47039/LevelPersistance" method="set_as_active"]
+[connection signal="collected" from="Tiles/@Node2D@47040" to="Tiles/@Node2D@47040/LevelPersistance" method="set_as_active"]
+[connection signal="collected" from="Tiles/@Node2D@47041" to="Tiles/@Node2D@47041/LevelPersistance" method="set_as_active"]
 [connection signal="area_entered" from="Tiles/DeathPit" to="Tiles/DeathPit" method="area_entered"]
-[connection signal="area_entered" from="Tiles/@Area2D@92167" to="Tiles/@Area2D@92167" method="area_entered"]
-[connection signal="area_entered" from="Tiles/@Area2D@92168" to="Tiles/@Area2D@92168" method="area_entered"]
-[connection signal="area_entered" from="Tiles/@Area2D@92169" to="Tiles/@Area2D@92169" method="area_entered"]
-[connection signal="area_entered" from="Tiles/@Area2D@92170" to="Tiles/@Area2D@92170" method="area_entered"]
-[connection signal="area_entered" from="Tiles/@Area2D@92171" to="Tiles/@Area2D@92171" method="area_entered"]
-[connection signal="area_entered" from="Tiles/@Area2D@92172" to="Tiles/@Area2D@92172" method="area_entered"]
-[connection signal="area_entered" from="Tiles/@Area2D@92173" to="Tiles/@Area2D@92173" method="area_entered"]
-[connection signal="area_entered" from="Tiles/@Area2D@92174" to="Tiles/@Area2D@92174" method="area_entered"]
-[connection signal="area_entered" from="Tiles/@Area2D@92175" to="Tiles/@Area2D@92175" method="area_entered"]
-[connection signal="area_entered" from="Tiles/@Area2D@92176" to="Tiles/@Area2D@92176" method="area_entered"]
-[connection signal="area_entered" from="Tiles/@Area2D@92177" to="Tiles/@Area2D@92177" method="area_entered"]
-[connection signal="area_entered" from="Tiles/@Area2D@92178" to="Tiles/@Area2D@92178" method="area_entered"]
-[connection signal="area_entered" from="Tiles/@Area2D@92179" to="Tiles/@Area2D@92179" method="area_entered"]
-[connection signal="area_entered" from="Tiles/@Area2D@92180" to="Tiles/@Area2D@92180" method="area_entered"]
-[connection signal="area_entered" from="Tiles/@Area2D@92181" to="Tiles/@Area2D@92181" method="area_entered"]
-[connection signal="area_entered" from="Tiles/@Area2D@92182" to="Tiles/@Area2D@92182" method="area_entered"]
-[connection signal="area_entered" from="Tiles/@Area2D@92183" to="Tiles/@Area2D@92183" method="area_entered"]
-[connection signal="area_entered" from="Tiles/@Area2D@92184" to="Tiles/@Area2D@92184" method="area_entered"]
-[connection signal="area_entered" from="Tiles/@Area2D@92185" to="Tiles/@Area2D@92185" method="area_entered"]
-[connection signal="area_entered" from="Tiles/@Area2D@92186" to="Tiles/@Area2D@92186" method="area_entered"]
-[connection signal="area_entered" from="Tiles/@Area2D@92187" to="Tiles/@Area2D@92187" method="area_entered"]
-[connection signal="area_entered" from="Tiles/@Area2D@92188" to="Tiles/@Area2D@92188" method="area_entered"]
-[connection signal="area_entered" from="Tiles/@Area2D@92189" to="Tiles/@Area2D@92189" method="area_entered"]
-[connection signal="area_entered" from="Tiles/@Area2D@92190" to="Tiles/@Area2D@92190" method="area_entered"]
-[connection signal="area_entered" from="Tiles/@Area2D@92191" to="Tiles/@Area2D@92191" method="area_entered"]
-[connection signal="area_entered" from="Tiles/@Area2D@92192" to="Tiles/@Area2D@92192" method="area_entered"]
-[connection signal="area_entered" from="Tiles/@Area2D@92193" to="Tiles/@Area2D@92193" method="area_entered"]
-[connection signal="area_entered" from="Tiles/@Area2D@92194" to="Tiles/@Area2D@92194" method="area_entered"]
-[connection signal="area_entered" from="Tiles/@Area2D@92195" to="Tiles/@Area2D@92195" method="area_entered"]
-[connection signal="area_entered" from="Tiles/@Area2D@92196" to="Tiles/@Area2D@92196" method="area_entered"]
-[connection signal="area_entered" from="Tiles/@Area2D@92197" to="Tiles/@Area2D@92197" method="area_entered"]
-[connection signal="area_entered" from="Tiles/@Area2D@92198" to="Tiles/@Area2D@92198" method="area_entered"]
-[connection signal="area_entered" from="Tiles/@Area2D@92199" to="Tiles/@Area2D@92199" method="area_entered"]
-[connection signal="area_entered" from="Tiles/@Area2D@92200" to="Tiles/@Area2D@92200" method="area_entered"]
-[connection signal="area_entered" from="Tiles/@Area2D@92201" to="Tiles/@Area2D@92201" method="area_entered"]
-[connection signal="area_entered" from="Tiles/@Area2D@92202" to="Tiles/@Area2D@92202" method="area_entered"]
-[connection signal="area_entered" from="Tiles/@Area2D@92203" to="Tiles/@Area2D@92203" method="area_entered"]
-[connection signal="area_entered" from="Tiles/@Area2D@92204" to="Tiles/@Area2D@92204" method="area_entered"]
-[connection signal="area_entered" from="Tiles/@Area2D@92205" to="Tiles/@Area2D@92205" method="area_entered"]
-[connection signal="area_entered" from="Tiles/@Area2D@92206" to="Tiles/@Area2D@92206" method="area_entered"]
-[connection signal="area_entered" from="Tiles/@Area2D@92207" to="Tiles/@Area2D@92207" method="area_entered"]
-[connection signal="area_entered" from="Tiles/@Area2D@92208" to="Tiles/@Area2D@92208" method="area_entered"]
-[connection signal="area_entered" from="Tiles/@Area2D@92209" to="Tiles/@Area2D@92209" method="area_entered"]
-[connection signal="area_entered" from="Tiles/@Area2D@92210" to="Tiles/@Area2D@92210" method="area_entered"]
-[connection signal="area_entered" from="Tiles/@Area2D@92211" to="Tiles/@Area2D@92211" method="area_entered"]
-[connection signal="area_entered" from="Tiles/@Area2D@92212" to="Tiles/@Area2D@92212" method="area_entered"]
-[connection signal="area_entered" from="Tiles/@Area2D@92213" to="Tiles/@Area2D@92213" method="area_entered"]
-[connection signal="area_entered" from="Tiles/@Area2D@92214" to="Tiles/@Area2D@92214" method="area_entered"]
-[connection signal="area_entered" from="Tiles/@Area2D@92215" to="Tiles/@Area2D@92215" method="area_entered"]
+[connection signal="area_entered" from="Tiles/@Area2D@47042" to="Tiles/@Area2D@47042" method="area_entered"]
+[connection signal="area_entered" from="Tiles/@Area2D@47043" to="Tiles/@Area2D@47043" method="area_entered"]
+[connection signal="area_entered" from="Tiles/@Area2D@47044" to="Tiles/@Area2D@47044" method="area_entered"]
+[connection signal="area_entered" from="Tiles/@Area2D@47045" to="Tiles/@Area2D@47045" method="area_entered"]
+[connection signal="area_entered" from="Tiles/@Area2D@47046" to="Tiles/@Area2D@47046" method="area_entered"]
+[connection signal="area_entered" from="Tiles/@Area2D@47047" to="Tiles/@Area2D@47047" method="area_entered"]
+[connection signal="area_entered" from="Tiles/@Area2D@47048" to="Tiles/@Area2D@47048" method="area_entered"]
+[connection signal="area_entered" from="Tiles/@Area2D@47049" to="Tiles/@Area2D@47049" method="area_entered"]
+[connection signal="area_entered" from="Tiles/@Area2D@47050" to="Tiles/@Area2D@47050" method="area_entered"]
+[connection signal="area_entered" from="Tiles/@Area2D@47051" to="Tiles/@Area2D@47051" method="area_entered"]
+[connection signal="area_entered" from="Tiles/@Area2D@47052" to="Tiles/@Area2D@47052" method="area_entered"]
+[connection signal="area_entered" from="Tiles/@Area2D@47053" to="Tiles/@Area2D@47053" method="area_entered"]
+[connection signal="area_entered" from="Tiles/@Area2D@47054" to="Tiles/@Area2D@47054" method="area_entered"]
+[connection signal="area_entered" from="Tiles/@Area2D@47055" to="Tiles/@Area2D@47055" method="area_entered"]
+[connection signal="area_entered" from="Tiles/@Area2D@47056" to="Tiles/@Area2D@47056" method="area_entered"]
+[connection signal="area_entered" from="Tiles/@Area2D@47057" to="Tiles/@Area2D@47057" method="area_entered"]
+[connection signal="area_entered" from="Tiles/@Area2D@47058" to="Tiles/@Area2D@47058" method="area_entered"]
+[connection signal="area_entered" from="Tiles/@Area2D@47059" to="Tiles/@Area2D@47059" method="area_entered"]
+[connection signal="area_entered" from="Tiles/@Area2D@47060" to="Tiles/@Area2D@47060" method="area_entered"]
+[connection signal="area_entered" from="Tiles/@Area2D@47061" to="Tiles/@Area2D@47061" method="area_entered"]
+[connection signal="area_entered" from="Tiles/@Area2D@47062" to="Tiles/@Area2D@47062" method="area_entered"]
+[connection signal="area_entered" from="Tiles/@Area2D@47063" to="Tiles/@Area2D@47063" method="area_entered"]
+[connection signal="area_entered" from="Tiles/@Area2D@47064" to="Tiles/@Area2D@47064" method="area_entered"]
+[connection signal="area_entered" from="Tiles/@Area2D@47065" to="Tiles/@Area2D@47065" method="area_entered"]
+[connection signal="area_entered" from="Tiles/@Area2D@47066" to="Tiles/@Area2D@47066" method="area_entered"]
+[connection signal="area_entered" from="Tiles/@Area2D@47067" to="Tiles/@Area2D@47067" method="area_entered"]
+[connection signal="area_entered" from="Tiles/@Area2D@47068" to="Tiles/@Area2D@47068" method="area_entered"]
+[connection signal="area_entered" from="Tiles/@Area2D@47069" to="Tiles/@Area2D@47069" method="area_entered"]
+[connection signal="area_entered" from="Tiles/@Area2D@47070" to="Tiles/@Area2D@47070" method="area_entered"]
+[connection signal="area_entered" from="Tiles/@Area2D@47071" to="Tiles/@Area2D@47071" method="area_entered"]
+[connection signal="area_entered" from="Tiles/@Area2D@47072" to="Tiles/@Area2D@47072" method="area_entered"]
+[connection signal="area_entered" from="Tiles/@Area2D@47073" to="Tiles/@Area2D@47073" method="area_entered"]
+[connection signal="area_entered" from="Tiles/@Area2D@47074" to="Tiles/@Area2D@47074" method="area_entered"]
+[connection signal="area_entered" from="Tiles/@Area2D@47075" to="Tiles/@Area2D@47075" method="area_entered"]
+[connection signal="area_entered" from="Tiles/@Area2D@47076" to="Tiles/@Area2D@47076" method="area_entered"]
+[connection signal="area_entered" from="Tiles/@Area2D@47077" to="Tiles/@Area2D@47077" method="area_entered"]
+[connection signal="area_entered" from="Tiles/@Area2D@47078" to="Tiles/@Area2D@47078" method="area_entered"]
+[connection signal="area_entered" from="Tiles/@Area2D@47079" to="Tiles/@Area2D@47079" method="area_entered"]
+[connection signal="area_entered" from="Tiles/@Area2D@47080" to="Tiles/@Area2D@47080" method="area_entered"]
+[connection signal="area_entered" from="Tiles/@Area2D@47081" to="Tiles/@Area2D@47081" method="area_entered"]
+[connection signal="area_entered" from="Tiles/@Area2D@47082" to="Tiles/@Area2D@47082" method="area_entered"]
+[connection signal="area_entered" from="Tiles/@Area2D@47083" to="Tiles/@Area2D@47083" method="area_entered"]
+[connection signal="area_entered" from="Tiles/@Area2D@47084" to="Tiles/@Area2D@47084" method="area_entered"]
+[connection signal="area_entered" from="Tiles/@Area2D@47085" to="Tiles/@Area2D@47085" method="area_entered"]
+[connection signal="area_entered" from="Tiles/@Area2D@47086" to="Tiles/@Area2D@47086" method="area_entered"]
+[connection signal="area_entered" from="Tiles/@Area2D@47087" to="Tiles/@Area2D@47087" method="area_entered"]
+[connection signal="area_entered" from="Tiles/@Area2D@47088" to="Tiles/@Area2D@47088" method="area_entered"]
+[connection signal="area_entered" from="Tiles/@Area2D@47089" to="Tiles/@Area2D@47089" method="area_entered"]
+[connection signal="area_entered" from="Tiles/@Area2D@47090" to="Tiles/@Area2D@47090" method="area_entered"]
diff --git a/Scenes/Prefabs/Blocks/BrickBlocks/BrickBlock.tscn b/Scenes/Prefabs/Blocks/BrickBlocks/BrickBlock.tscn
index 7bb6ddf..da5c705 100644
--- a/Scenes/Prefabs/Blocks/BrickBlocks/BrickBlock.tscn
+++ b/Scenes/Prefabs/Blocks/BrickBlocks/BrickBlock.tscn
@@ -67,6 +67,7 @@ metadata/_custom_type_script = "uid://maqpreddu5kg"
 
 [connection signal="block_destroyed" from="." to="LevelPersistance" method="set_as_active_2"]
 [connection signal="block_emptied" from="." to="LevelPersistance" method="set_as_active"]
+[connection signal="item_changed" from="." to="." method="check_brick_empty"]
 [connection signal="player_block_hit" from="." to="." method="on_block_hit"]
 [connection signal="player_block_hit" from="." to="BlockAnimations" method="bounce_block" unbinds=1]
 [connection signal="shell_block_hit" from="." to="." method="on_shell_block_hit"]
diff --git a/Scripts/Classes/Blocks/BlockClass.gd b/Scripts/Classes/Blocks/BlockClass.gd
index 5df0731..15efcfe 100644
--- a/Scripts/Classes/Blocks/BlockClass.gd
+++ b/Scripts/Classes/Blocks/BlockClass.gd
@@ -6,7 +6,10 @@ signal shell_block_hit(shell: Shell)
 
 @export var visuals: Node = null
 const EMPTY_BLOCK = ("res://Scenes/Prefabs/Blocks/EmptyBlock.tscn")
-@export var item: PackedScene = null
+@export var item: PackedScene = null:
+	set(value):
+		item = value
+		item_changed.emit()
 @export var destructable := true
 @export var destruction_particle_scene: PackedScene = null
 @export_range(1, 99) var item_amount := 1
@@ -19,7 +22,7 @@ var bouncing := false
 const NO_SFX_ITEMS := ["res://Scenes/Prefabs/Entities/Items/SpinningRedCoin.tscn","res://Scenes/Prefabs/Entities/Items/SpinningCoin.tscn", "res://Scenes/Prefabs/Entities/Items/Vine.tscn" ]
 
 @export var start_z := -1
-
+signal item_changed
 signal block_emptied
 signal block_destroyed
 
diff --git a/Scripts/Classes/Blocks/BrickBlock.gd b/Scripts/Classes/Blocks/BrickBlock.gd
index 9daf0c6..1ef429e 100644
--- a/Scripts/Classes/Blocks/BrickBlock.gd
+++ b/Scripts/Classes/Blocks/BrickBlock.gd
@@ -4,10 +4,12 @@ extends Block
 var ticking_down := false
 
 func _ready() -> void:
-	$PSwitcher.enabled = item == null
 	if item_amount == 10 and item.resource_path == "res://Scenes/Prefabs/Entities/Items/SpinningCoin.tscn" and is_instance_valid(Global.level_editor) == false:
 		Global.log_warning("Coin Brick Block is wrong! please report!: " + name)
 
+func check_brick_empty() -> void:
+	$PSwitcher.enabled = item == null
+
 func on_block_hit(player: Player) -> void:
 	if player.power_state.hitbox_size == "Big":
 		if item == null:
