diff --git a/project.godot b/project.godot
index 9f8fe00..6ddc852 100644
--- a/project.godot
+++ b/project.godot
@@ -618,7 +618,9 @@ show_notes={
 2d_physics/layer_7="Player"
 2d_physics/layer_9="Player Projectile"
 3d_physics/layer_9="Marwing - Player Ship"
+2d_physics/layer_10="Obstacles"
 3d_physics/layer_10="Marwing - Enemy Ship"
+2d_physics/layer_11="Coins"
 3d_physics/layer_11="Marwing - Player Projectile"
 3d_physics/layer_12="Marwing - Enemy Projectile"
 3d_physics/layer_13="Marwing - Obstacle"
diff --git a/scenes/levels/Flappy/Characters/Obstacles/Pipes.tscn b/scenes/levels/Flappy/Characters/Obstacles/Pipes.tscn
index 7973647..4d8481d 100644
--- a/scenes/levels/Flappy/Characters/Obstacles/Pipes.tscn
+++ b/scenes/levels/Flappy/Characters/Obstacles/Pipes.tscn
@@ -12,9 +12,13 @@ position = Vector2( 500, 300 )
 script = ExtResource( 2 )
 
 [node name="Coin" parent="." instance=ExtResource( 3 )]
+collision_layer = 1024
+collision_mask = 0
 
 [node name="pipe_top" type="Area2D" parent="."]
 position = Vector2( 0, -65 )
+collision_layer = 512
+collision_mask = 0
 
 [node name="pipe_end" type="Sprite" parent="pipe_top"]
 rotation = -3.14159
@@ -106,6 +110,8 @@ shape = SubResource( 1 )
 [node name="pipe_down" type="Area2D" parent="."]
 position = Vector2( 0, 65 )
 rotation = 3.14159
+collision_layer = 512
+collision_mask = 0
 
 [node name="pipe_end" type="Sprite" parent="pipe_down"]
 rotation = -3.14159
diff --git a/scenes/levels/Flappy/Characters/Player/PlayerFlying.tscn b/scenes/levels/Flappy/Characters/Player/PlayerFlying.tscn
index 5805c55..914e483 100644
--- a/scenes/levels/Flappy/Characters/Player/PlayerFlying.tscn
+++ b/scenes/levels/Flappy/Characters/Player/PlayerFlying.tscn
@@ -10,6 +10,8 @@
 radius = 15.0
 
 [node name="Player_RigidBody" type="RigidBody2D"]
+collision_layer = 64
+collision_mask = 1536
 gravity_scale = 5.0
 script = ExtResource( 2 )
 
diff --git a/scenes/levels/Flappy/Flappy01.tscn b/scenes/levels/Flappy/Flappy01.tscn
index e4c44c4..d0fc2bc 100644
--- a/scenes/levels/Flappy/Flappy01.tscn
+++ b/scenes/levels/Flappy/Flappy01.tscn
@@ -102,6 +102,8 @@ fit_content_height = true
 script = ExtResource( 6 )
 
 [node name="deathBounds" type="Area2D" parent="."]
+collision_layer = 512
+collision_mask = 0
 script = ExtResource( 5 )
 
 [node name="CollisionShape2D" type="CollisionShape2D" parent="deathBounds"]
