diff --git a/Scenes/Levels/SMBS/World7/7-4.tscn b/Scenes/Levels/SMBS/World7/7-4.tscn
index da823ef..46999bc 100644
--- a/Scenes/Levels/SMBS/World7/7-4.tscn
+++ b/Scenes/Levels/SMBS/World7/7-4.tscn
@@ -126,13 +126,13 @@ item = ExtResource("17_ekraa")
 [connection signal="player_block_hit" from="Tiles/QuestionBlock" to="Tiles/QuestionBlock/BlockAnimations" method="bounce_block" unbinds=1]
 [connection signal="shell_block_hit" from="Tiles/QuestionBlock" to="Tiles/QuestionBlock" method="dispense_item" unbinds=1]
 [connection signal="shell_block_hit" from="Tiles/QuestionBlock" to="Tiles/QuestionBlock/BlockAnimations" method="bounce_block" unbinds=1]
-[connection signal="block_destroyed" from="Tiles/@AnimatableBody2D@287613" to="Tiles/@AnimatableBody2D@287613/LevelPersistance" method="set_as_active_2"]
-[connection signal="block_emptied" from="Tiles/@AnimatableBody2D@287613" to="Tiles/@AnimatableBody2D@287613/LevelPersistance" method="set_as_active"]
-[connection signal="player_block_hit" from="Tiles/@AnimatableBody2D@287613" to="Tiles/@AnimatableBody2D@287613" method="player_mushroom_check"]
-[connection signal="player_block_hit" from="Tiles/@AnimatableBody2D@287613" to="Tiles/@AnimatableBody2D@287613" method="dispense_item" unbinds=1]
-[connection signal="player_block_hit" from="Tiles/@AnimatableBody2D@287613" to="Tiles/@AnimatableBody2D@287613/BlockAnimations" method="bounce_block" unbinds=1]
-[connection signal="shell_block_hit" from="Tiles/@AnimatableBody2D@287613" to="Tiles/@AnimatableBody2D@287613" method="dispense_item" unbinds=1]
-[connection signal="shell_block_hit" from="Tiles/@AnimatableBody2D@287613" to="Tiles/@AnimatableBody2D@287613/BlockAnimations" method="bounce_block" unbinds=1]
+[connection signal="block_destroyed" from="Tiles/@AnimatableBody2D@26079" to="Tiles/@AnimatableBody2D@26079/LevelPersistance" method="set_as_active_2"]
+[connection signal="block_emptied" from="Tiles/@AnimatableBody2D@26079" to="Tiles/@AnimatableBody2D@26079/LevelPersistance" method="set_as_active"]
+[connection signal="player_block_hit" from="Tiles/@AnimatableBody2D@26079" to="Tiles/@AnimatableBody2D@26079" method="player_mushroom_check"]
+[connection signal="player_block_hit" from="Tiles/@AnimatableBody2D@26079" to="Tiles/@AnimatableBody2D@26079" method="dispense_item" unbinds=1]
+[connection signal="player_block_hit" from="Tiles/@AnimatableBody2D@26079" to="Tiles/@AnimatableBody2D@26079/BlockAnimations" method="bounce_block" unbinds=1]
+[connection signal="shell_block_hit" from="Tiles/@AnimatableBody2D@26079" to="Tiles/@AnimatableBody2D@26079" method="dispense_item" unbinds=1]
+[connection signal="shell_block_hit" from="Tiles/@AnimatableBody2D@26079" to="Tiles/@AnimatableBody2D@26079/BlockAnimations" method="bounce_block" unbinds=1]
 [connection signal="block_destroyed" from="Tiles/InvisibleQuestionBlock" to="Tiles/InvisibleQuestionBlock/LevelPersistance" method="set_as_active_2"]
 [connection signal="block_emptied" from="Tiles/InvisibleQuestionBlock" to="Tiles/InvisibleQuestionBlock/LevelPersistance" method="set_as_active"]
 [connection signal="player_block_hit" from="Tiles/InvisibleQuestionBlock" to="Tiles/InvisibleQuestionBlock" method="player_mushroom_check"]
@@ -140,24 +140,24 @@ item = ExtResource("17_ekraa")
 [connection signal="player_block_hit" from="Tiles/InvisibleQuestionBlock" to="Tiles/InvisibleQuestionBlock/BlockAnimations" method="bounce_block" unbinds=1]
 [connection signal="shell_block_hit" from="Tiles/InvisibleQuestionBlock" to="Tiles/InvisibleQuestionBlock" method="dispense_item" unbinds=1]
 [connection signal="shell_block_hit" from="Tiles/InvisibleQuestionBlock" to="Tiles/InvisibleQuestionBlock/BlockAnimations" method="bounce_block" unbinds=1]
-[connection signal="block_destroyed" from="Tiles/@AnimatableBody2D@287614" to="Tiles/@AnimatableBody2D@287614/LevelPersistance" method="set_as_active_2"]
-[connection signal="block_emptied" from="Tiles/@AnimatableBody2D@287614" to="Tiles/@AnimatableBody2D@287614/LevelPersistance" method="set_as_active"]
-[connection signal="player_block_hit" from="Tiles/@AnimatableBody2D@287614" to="Tiles/@AnimatableBody2D@287614" method="player_mushroom_check"]
-[connection signal="player_block_hit" from="Tiles/@AnimatableBody2D@287614" to="Tiles/@AnimatableBody2D@287614" method="dispense_item" unbinds=1]
-[connection signal="player_block_hit" from="Tiles/@AnimatableBody2D@287614" to="Tiles/@AnimatableBody2D@287614/BlockAnimations" method="bounce_block" unbinds=1]
-[connection signal="shell_block_hit" from="Tiles/@AnimatableBody2D@287614" to="Tiles/@AnimatableBody2D@287614" method="dispense_item" unbinds=1]
-[connection signal="shell_block_hit" from="Tiles/@AnimatableBody2D@287614" to="Tiles/@AnimatableBody2D@287614/BlockAnimations" method="bounce_block" unbinds=1]
-[connection signal="block_destroyed" from="Tiles/@AnimatableBody2D@287615" to="Tiles/@AnimatableBody2D@287615/LevelPersistance" method="set_as_active_2"]
-[connection signal="block_emptied" from="Tiles/@AnimatableBody2D@287615" to="Tiles/@AnimatableBody2D@287615/LevelPersistance" method="set_as_active"]
-[connection signal="player_block_hit" from="Tiles/@AnimatableBody2D@287615" to="Tiles/@AnimatableBody2D@287615" method="player_mushroom_check"]
-[connection signal="player_block_hit" from="Tiles/@AnimatableBody2D@287615" to="Tiles/@AnimatableBody2D@287615" method="dispense_item" unbinds=1]
-[connection signal="player_block_hit" from="Tiles/@AnimatableBody2D@287615" to="Tiles/@AnimatableBody2D@287615/BlockAnimations" method="bounce_block" unbinds=1]
-[connection signal="shell_block_hit" from="Tiles/@AnimatableBody2D@287615" to="Tiles/@AnimatableBody2D@287615" method="dispense_item" unbinds=1]
-[connection signal="shell_block_hit" from="Tiles/@AnimatableBody2D@287615" to="Tiles/@AnimatableBody2D@287615/BlockAnimations" method="bounce_block" unbinds=1]
-[connection signal="block_destroyed" from="Tiles/@AnimatableBody2D@287616" to="Tiles/@AnimatableBody2D@287616/LevelPersistance" method="set_as_active_2"]
-[connection signal="block_emptied" from="Tiles/@AnimatableBody2D@287616" to="Tiles/@AnimatableBody2D@287616/LevelPersistance" method="set_as_active"]
-[connection signal="player_block_hit" from="Tiles/@AnimatableBody2D@287616" to="Tiles/@AnimatableBody2D@287616" method="player_mushroom_check"]
-[connection signal="player_block_hit" from="Tiles/@AnimatableBody2D@287616" to="Tiles/@AnimatableBody2D@287616" method="dispense_item" unbinds=1]
-[connection signal="player_block_hit" from="Tiles/@AnimatableBody2D@287616" to="Tiles/@AnimatableBody2D@287616/BlockAnimations" method="bounce_block" unbinds=1]
-[connection signal="shell_block_hit" from="Tiles/@AnimatableBody2D@287616" to="Tiles/@AnimatableBody2D@287616" method="dispense_item" unbinds=1]
-[connection signal="shell_block_hit" from="Tiles/@AnimatableBody2D@287616" to="Tiles/@AnimatableBody2D@287616/BlockAnimations" method="bounce_block" unbinds=1]
+[connection signal="block_destroyed" from="Tiles/@AnimatableBody2D@26080" to="Tiles/@AnimatableBody2D@26080/LevelPersistance" method="set_as_active_2"]
+[connection signal="block_emptied" from="Tiles/@AnimatableBody2D@26080" to="Tiles/@AnimatableBody2D@26080/LevelPersistance" method="set_as_active"]
+[connection signal="player_block_hit" from="Tiles/@AnimatableBody2D@26080" to="Tiles/@AnimatableBody2D@26080" method="player_mushroom_check"]
+[connection signal="player_block_hit" from="Tiles/@AnimatableBody2D@26080" to="Tiles/@AnimatableBody2D@26080" method="dispense_item" unbinds=1]
+[connection signal="player_block_hit" from="Tiles/@AnimatableBody2D@26080" to="Tiles/@AnimatableBody2D@26080/BlockAnimations" method="bounce_block" unbinds=1]
+[connection signal="shell_block_hit" from="Tiles/@AnimatableBody2D@26080" to="Tiles/@AnimatableBody2D@26080" method="dispense_item" unbinds=1]
+[connection signal="shell_block_hit" from="Tiles/@AnimatableBody2D@26080" to="Tiles/@AnimatableBody2D@26080/BlockAnimations" method="bounce_block" unbinds=1]
+[connection signal="block_destroyed" from="Tiles/@AnimatableBody2D@26081" to="Tiles/@AnimatableBody2D@26081/LevelPersistance" method="set_as_active_2"]
+[connection signal="block_emptied" from="Tiles/@AnimatableBody2D@26081" to="Tiles/@AnimatableBody2D@26081/LevelPersistance" method="set_as_active"]
+[connection signal="player_block_hit" from="Tiles/@AnimatableBody2D@26081" to="Tiles/@AnimatableBody2D@26081" method="player_mushroom_check"]
+[connection signal="player_block_hit" from="Tiles/@AnimatableBody2D@26081" to="Tiles/@AnimatableBody2D@26081" method="dispense_item" unbinds=1]
+[connection signal="player_block_hit" from="Tiles/@AnimatableBody2D@26081" to="Tiles/@AnimatableBody2D@26081/BlockAnimations" method="bounce_block" unbinds=1]
+[connection signal="shell_block_hit" from="Tiles/@AnimatableBody2D@26081" to="Tiles/@AnimatableBody2D@26081" method="dispense_item" unbinds=1]
+[connection signal="shell_block_hit" from="Tiles/@AnimatableBody2D@26081" to="Tiles/@AnimatableBody2D@26081/BlockAnimations" method="bounce_block" unbinds=1]
+[connection signal="block_destroyed" from="Tiles/@AnimatableBody2D@26082" to="Tiles/@AnimatableBody2D@26082/LevelPersistance" method="set_as_active_2"]
+[connection signal="block_emptied" from="Tiles/@AnimatableBody2D@26082" to="Tiles/@AnimatableBody2D@26082/LevelPersistance" method="set_as_active"]
+[connection signal="player_block_hit" from="Tiles/@AnimatableBody2D@26082" to="Tiles/@AnimatableBody2D@26082" method="player_mushroom_check"]
+[connection signal="player_block_hit" from="Tiles/@AnimatableBody2D@26082" to="Tiles/@AnimatableBody2D@26082" method="dispense_item" unbinds=1]
+[connection signal="player_block_hit" from="Tiles/@AnimatableBody2D@26082" to="Tiles/@AnimatableBody2D@26082/BlockAnimations" method="bounce_block" unbinds=1]
+[connection signal="shell_block_hit" from="Tiles/@AnimatableBody2D@26082" to="Tiles/@AnimatableBody2D@26082" method="dispense_item" unbinds=1]
+[connection signal="shell_block_hit" from="Tiles/@AnimatableBody2D@26082" to="Tiles/@AnimatableBody2D@26082/BlockAnimations" method="bounce_block" unbinds=1]
diff --git a/Scripts/Classes/Entities/PipeArea.gd b/Scripts/Classes/Entities/PipeArea.gd
index f738c2a..9481cd1 100644
--- a/Scripts/Classes/Entities/PipeArea.gd
+++ b/Scripts/Classes/Entities/PipeArea.gd
@@ -64,6 +64,7 @@ func update_visuals() -> void:
 		hide()
 
 func exit_pipe() -> void:
+	can_enter = false
 	pipe_exited.emit()
 	for i in get_tree().get_nodes_in_group("Players"):
 		i.go_to_exit_pipe(self)
@@ -71,6 +72,7 @@ func exit_pipe() -> void:
 		await get_tree().create_timer(0.5, false).timeout
 		await i.exit_pipe(self)
 	exiting_pipe_id = -1
+	can_enter = true
 
 func get_vector(direction := 0) -> Vector2:
 	match direction:
