diff --git a/scenes/levels/blargle/Angry.tscn b/scenes/levels/blargle/Angry.tscn
index 208e420..2c1b194 100644
--- a/scenes/levels/blargle/Angry.tscn
+++ b/scenes/levels/blargle/Angry.tscn
@@ -1,6 +1,6 @@
 [gd_scene load_steps=6 format=2]
 
-[ext_resource path="res://scenes/levels/Default/Level01.tscn" type="PackedScene" id=1]
+[ext_resource path="res://scenes/levels/blargle/Disappearing.tscn" type="PackedScene" id=1]
 [ext_resource path="res://scenes/levels/TileSet.tres" type="TileSet" id=2]
 [ext_resource path="res://scenes/EndPortal.tscn" type="PackedScene" id=3]
 [ext_resource path="res://scenes/SpawnPoint.tscn" type="PackedScene" id=6]
diff --git a/scenes/levels/blargle/Disappearing.tscn b/scenes/levels/blargle/Disappearing.tscn
index 7a19390..1e90819 100644
--- a/scenes/levels/blargle/Disappearing.tscn
+++ b/scenes/levels/blargle/Disappearing.tscn
@@ -1,12 +1,12 @@
 [gd_scene load_steps=8 format=2]
 
-[ext_resource path="res://scenes/levels/Default/Level01.tscn" type="PackedScene" id=1]
 [ext_resource path="res://scenes/levels/TileSet.tres" type="TileSet" id=2]
 [ext_resource path="res://scenes/EndPortal.tscn" type="PackedScene" id=3]
 [ext_resource path="res://sprites/Coin_Gems/Coin/Coin.tscn" type="PackedScene" id=4]
 [ext_resource path="res://scenes/boxes/CoinBox.tscn" type="PackedScene" id=5]
 [ext_resource path="res://scenes/SpawnPoint.tscn" type="PackedScene" id=6]
 [ext_resource path="res://scenes/boxes/DisappearingPlatform.tscn" type="PackedScene" id=7]
+[ext_resource path="res://scenes/CameraFollow.tscn" type="PackedScene" id=8]
 
 [node name="TileMap2" type="TileMap"]
 tile_set = ExtResource( 2 )
@@ -16,7 +16,6 @@ tile_data = PoolIntArray( -1179645, 0, 0, -1179644, 0, 0, -1179643, 0, 0, -11796
 
 [node name="EndPortal" parent="." instance=ExtResource( 3 )]
 position = Vector2( 54, -690 )
-next_level = ExtResource( 1 )
 
 [node name="Coin" parent="." instance=ExtResource( 4 )]
 position = Vector2( 736, 345 )
@@ -137,3 +136,6 @@ disappearDelay = 0.1
 
 [node name="SpawnPoint" parent="." instance=ExtResource( 6 )]
 position = Vector2( 139, 422 )
+
+[node name="CameraFollow" parent="." instance=ExtResource( 8 )]
+follow_vertically = true
