diff --git a/Scripts/Parts/Checkpoint.gd b/Scripts/Parts/Checkpoint.gd
index be42506..3e0052c 100644
--- a/Scripts/Parts/Checkpoint.gd
+++ b/Scripts/Parts/Checkpoint.gd
@@ -35,7 +35,7 @@ func _ready() -> void:
 		if Settings.file.difficulty.checkpoint_style != 0:
 			queue_free()
 			return
-	elif Settings.file.difficulty.checkpoint_style == 0:
+	elif Settings.file.difficulty.checkpoint_style == 0 and [Global.GameMode.CUSTOM_LEVEL, Global.GameMode.LEVEL_EDITOR].has(Global.current_game_mode) == false:
 		queue_free()
 		return
 	if passed and PipeArea.exiting_pipe_id == -1 and Global.current_game_mode != Global.GameMode.LEVEL_EDITOR and Level.vine_return_level == "" and passed_checkpoints[passed_checkpoints.size() - 1] == id:
