diff --git a/scripts/classes/JBody2D.gd b/scripts/classes/JBody2D.gd
index 8789c1d..66925be 100644
--- a/scripts/classes/JBody2D.gd
+++ b/scripts/classes/JBody2D.gd
@@ -24,10 +24,11 @@ func _init():
 	stats.name = "Stats"
 	stats.parent = self
 	add_child(stats)
-	
+
 	#Call this at the last moment, prevents possible errors from non-ready collisions and allows overriden _init methods to run first.
 	_init_collisions.call_deferred()
 
+
 func _init_collisions():
 	#Each creature only exists on their own layer, they collide with the world thanks to their collision_mask
 	match entity_type:
@@ -37,28 +38,28 @@ func _init_collisions():
 			collision_layer = J.PHYSICS_LAYER_ENEMIES
 		J.ENTITY_TYPE.NPC:
 			collision_layer = J.PHYSICS_LAYER_NPCS
-			
+
 	if J.is_server():
 		#By default, only collide with the world. No other entities.
 		collision_mask = J.PHYSICS_LAYER_WORLD
-		
+
 		#Set what it will collide with
 		match entity_type:
 			#The player cannot walk past NPCs and enemies. But other players cannot block their path.
 			J.ENTITY_TYPE.PLAYER:
 				collision_mask += J.PHYSICS_LAYER_ENEMIES + J.PHYSICS_LAYER_NPCS
-			
+
 			#Enemies can be blocked by NPCs and players.
 			J.ENTITY_TYPE.ENEMY:
 				collision_mask += J.PHYSICS_LAYER_PLAYERS + J.PHYSICS_LAYER_NPCS
-				
+
 			#NPCs cannot be stopped by any entity.
 			J.ENTITY_TYPE.NPC:
 				collision_mask = 0
 	else:
 		# Don't handle collision on client side
 		collision_mask = 0
-	
+
 
 func attack(target: CharacterBody2D):
 	var damage = randi_range(stats.attack_power_min, stats.attack_power_max)
