diff --git a/Scripts/Classes/Entities/Enemies/Lakitu.gd b/Scripts/Classes/Entities/Enemies/Lakitu.gd
index aef3665..e250b4f 100644
--- a/Scripts/Classes/Entities/Enemies/Lakitu.gd
+++ b/Scripts/Classes/Entities/Enemies/Lakitu.gd
@@ -12,8 +12,6 @@ var lakitu_point := Vector2.ZERO
 
 const BLOCK_DISTANCE := 64
 
-static var fixed_throw := true
-
 var player: Player = null
 
 var retreat := false
@@ -28,7 +26,6 @@ func _ready() -> void:
 	can_enter = false
 	$ThrowTimer.start()
 	lakitu_point = to_local(global_position)
-	fixed_throw = Settings.file.difficulty.lakitu_style == 1
 	get_parent().move_child(self, 0)
 
 func _process(_delta: float) -> void:
@@ -80,7 +77,7 @@ func throw_spiny() -> void:
 	node.set("in_egg", true)
 	node.global_position = $Cloud/Sprite.global_position
 	node.velocity = Vector2(0, -150)
-	if fixed_throw:
+	if Settings.file.difficulty.lakitu_style == 1:
 		node.velocity.x = 50 * (sign(player.global_position.x - global_position.x))
 		node.set("direction", sign(node.velocity.x))
 	add_sibling(node)
diff --git a/Scripts/Classes/Entities/Enemies/Spiny.gd b/Scripts/Classes/Entities/Enemies/Spiny.gd
index 78aec40..ce65ae3 100755
--- a/Scripts/Classes/Entities/Enemies/Spiny.gd
+++ b/Scripts/Classes/Entities/Enemies/Spiny.gd
@@ -9,9 +9,13 @@ func _physics_process(delta: float) -> void:
 
 func handle_movement(_delta: float) -> void:
 	if in_egg:
+		$BasicEnemyMovement.move_speed = 0
+		$BasicEnemyMovement.second_quest_speed = 0
 		if is_on_floor():
 			var player = get_tree().get_first_node_in_group("Players")
 			direction = sign(player.global_position.x - global_position.x)
+			$BasicEnemyMovement.move_speed = 32
+			$BasicEnemyMovement.second_quest_speed = 36
 			in_egg = false
 		$Sprite.play("Egg")
 	else:
