diff --git a/Scenes/Prefabs/UI/MarathonResults.tscn b/Scenes/Prefabs/UI/MarathonResults.tscn
index 1b8c64c..63b556a 100644
--- a/Scenes/Prefabs/UI/MarathonResults.tscn
+++ b/Scenes/Prefabs/UI/MarathonResults.tscn
@@ -175,6 +175,7 @@ offset_right = 90.0
 offset_bottom = 32.5
 grow_horizontal = 2
 grow_vertical = 2
+has_speedrun_stuff = true
 
 [node name="LevelSelect" parent="." instance=ExtResource("8_orie5")]
 visible = false
diff --git a/Scripts/Classes/Singletons/SpeedrunHandler.gd b/Scripts/Classes/Singletons/SpeedrunHandler.gd
index 9b7d129..24342d9 100644
--- a/Scripts/Classes/Singletons/SpeedrunHandler.gd
+++ b/Scripts/Classes/Singletons/SpeedrunHandler.gd
@@ -303,7 +303,7 @@ func run_finished() -> void:
 	if Global.current_game_mode == Global.GameMode.BOO_RACE:
 		pass
 	else:
-		var best := -1
+		var best: float = -1
 		if Global.current_game_mode == Global.GameMode.MARATHON_PRACTICE:
 			if is_warp_run:
 				best = best_level_any_times.get(str(Global.world_num) + "-" + str(Global.level_num), -1)
@@ -314,7 +314,6 @@ func run_finished() -> void:
 				best = marathon_best_any_time
 			else:
 				best = marathon_best_warpless_time
-		
 		if best <= 0 or best > timer:
 			if Global.current_game_mode == Global.GameMode.MARATHON_PRACTICE:
 				save_recording()
