diff --git a/scenes/root/Root.gd b/scenes/root/Root.gd
index 76d6811..798bc12 100644
--- a/scenes/root/Root.gd
+++ b/scenes/root/Root.gd
@@ -96,7 +96,7 @@ func _on_run_as_client_pressed():
 	$SelectRunMode.queue_free()
 
 	if not J.init_client():
-		J.logger.err("Could not initialize the client, quitting")
+		J.logger.error("Could not initialize the client, quitting")
 		get_tree().quit()
 		return
 
