diff --git a/scripts/components/networking/statssynchronizercomponent/StatsSynchronizerComponent.gd b/scripts/components/networking/statssynchronizercomponent/StatsSynchronizerComponent.gd
index b2e5909..65fa70f 100644
--- a/scripts/components/networking/statssynchronizercomponent/StatsSynchronizerComponent.gd
+++ b/scripts/components/networking/statssynchronizercomponent/StatsSynchronizerComponent.gd
@@ -643,6 +643,6 @@ func sync_energy_recovery(timestamp: float, from: String, current_energy: int, r
 func _on_energy_regen_timer_timeout():
 	if not ready_done:
 		return
-	
+
 	if energy != energy_max:
 		energy_recovery(target_node.get_name(), energy_regen)
