diff --git a/source/match/players/human/UnitActionsController.gd b/source/match/players/human/UnitActionsController.gd
index fda0d2a..7fb52c5 100644
--- a/source/match/players/human/UnitActionsController.gd
+++ b/source/match/players/human/UnitActionsController.gd
@@ -55,11 +55,14 @@ func _navigate_selected_units_towards_unit(target_unit):
 		elif Actions.Constructing.is_applicable(unit, target_unit):
 			unit.action = Actions.Constructing.new(target_unit)
 		elif (
-			target_unit.is_in_group("adversary_units")
-			or target_unit.is_in_group("controlled_units")
+			(
+				target_unit.is_in_group("adversary_units")
+				or target_unit.is_in_group("controlled_units")
+			)
+			and Actions.Following.is_applicable(unit)
 		):
 			unit.action = Actions.Following.new(target_unit)
-		else:
+		elif Actions.MovingToUnit.is_applicable(unit):
 			unit.action = Actions.MovingToUnit.new(target_unit)
 
 
diff --git a/source/match/units/actions/Following.gd b/source/match/units/actions/Following.gd
index 7fe5106..4574990 100644
--- a/source/match/units/actions/Following.gd
+++ b/source/match/units/actions/Following.gd
@@ -12,6 +12,10 @@ var _sub_action = null
 @onready var _unit = Utils.NodeEx.find_parent_with_group(self, "units")
 
 
+static func is_applicable(source_unit):
+	return MovingToUnit.is_applicable(source_unit)
+
+
 func _init(target_unit):
 	_target_unit = target_unit
 
