diff --git a/scripts/components/statssynchronizercomponent/StatsSynchronizerComponent.gd b/scripts/components/statssynchronizercomponent/StatsSynchronizerComponent.gd
index 536be9d..353a8cf 100644
--- a/scripts/components/statssynchronizercomponent/StatsSynchronizerComponent.gd
+++ b/scripts/components/statssynchronizercomponent/StatsSynchronizerComponent.gd
@@ -170,6 +170,9 @@ func _ready():
 	# Physics only needed on client side
 	if G.is_server():
 		set_physics_process(false)
+
+		#Uses a setter to automatically call server_periodic_update() when true
+		server_periodic_update_enabled = true
 	else:
 		#Wait until the connection is ready to synchronize stats
 		if not multiplayer.has_multiplayer_peer():
@@ -184,9 +187,6 @@ func _ready():
 
 		G.sync_rpc.statssynchronizer_sync_stats.rpc_id(1, target_node.name)
 
-		#Uses a setter to automatically call server_periodic_update() when true
-		server_periodic_update_enabled = true
-
 	# Make sure this line is called on server and client's side
 	ready_done = true
 
