diff --git a/godot/combat/battlers/Battler.gd b/godot/combat/battlers/Battler.gd
index 0310fa7..bebceca 100644
--- a/godot/combat/battlers/Battler.gd
+++ b/godot/combat/battlers/Battler.gd
@@ -17,7 +17,7 @@ var selectable : bool = false
 export var party_member = false
 
 func initialize():
-	var direction : Vector2 = Vector2(-1.0, 0.0) if is_in_group('party') else Vector2(1.0, 0.0)
+	var direction : Vector2 = Vector2(-1.0, 0.0) if party_member else Vector2(1.0, 0.0)
 	target_global_position = $TargetAnchor.global_position + direction * TARGET_OFFSET_DISTANCE
 	
 	stats.connect("health_depleted", self, "_on_health_depleted")
diff --git a/godot/combat/interface/SelectArrow.gd b/godot/combat/interface/SelectArrow.gd
index 8fca2ea..cc55a98 100644
--- a/godot/combat/interface/SelectArrow.gd
+++ b/godot/combat/interface/SelectArrow.gd
@@ -17,7 +17,7 @@ func select_target(battlers : Array) -> Battler:
 	visible = true
 	targets = battlers
 	target_active = targets[0]
-	scale.x = -1.0 if target_active.is_in_group('monster') else 1.0
+	scale.x = 1.0 if target_active.party_member else -1.0
 	global_position = target_active.target_global_position
 	anim_player.play("wiggle")
 	var selected_target : Battler = yield(self, "target_selected")
