diff --git a/game/Scenes/Game/CameraController.tscn b/game/Scenes/Game/CameraController.tscn
index 175bda9..20580ea 100644
--- a/game/Scenes/Game/CameraController.tscn
+++ b/game/Scenes/Game/CameraController.tscn
@@ -777,6 +777,8 @@ autowrap = true
 [connection signal="pressed" from="PieceContextMenu/SpeakerMenu/VBoxContainer/HBoxContainer/SpeakerSelectTrackButton" to="." method="_on_SpeakerSelectTrackButton_pressed"]
 [connection signal="pressed" from="PieceContextMenu/SpeakerMenu/VBoxContainer/HBoxContainer/SpeakerPlayStopButton" to="." method="_on_SpeakerPlayStopButton_pressed"]
 [connection signal="pressed" from="PieceContextMenu/SpeakerMenu/VBoxContainer/HBoxContainer/SpeakerPauseButton" to="." method="_on_SpeakerPauseButton_pressed"]
+[connection signal="mouse_entered" from="PieceContextMenu/SpeakerMenu/VBoxContainer/SpeakerVolumeSlider" to="." method="_on_SpeakerVolumeSlider_mouse_entered"]
+[connection signal="mouse_exited" from="PieceContextMenu/SpeakerMenu/VBoxContainer/SpeakerVolumeSlider" to="." method="_on_SpeakerVolumeSlider_mouse_exited"]
 [connection signal="value_changed" from="PieceContextMenu/SpeakerMenu/VBoxContainer/SpeakerVolumeSlider" to="." method="_on_SpeakerVolumeSlider_value_changed"]
 [connection signal="pressed" from="PieceContextMenu/TakeOffTopMenu/TakeOffTopSpinBoxButton" to="." method="_on_TakeOffTopSpinBoxButton_pressed"]
 [connection signal="pressed" from="PieceContextMenu/TakeOutMenu/TakeOutSpinBoxButton" to="." method="_on_TakeOutSpinBoxButton_pressed"]
diff --git a/game/Scripts/Game/CameraController.gd b/game/Scripts/Game/CameraController.gd
index 5f920f9..e4922fc 100644
--- a/game/Scripts/Game/CameraController.gd
+++ b/game/Scripts/Game/CameraController.gd
@@ -213,7 +213,7 @@ var _send_erase_position = false
 var _send_paint_position = false
 var _spawn_point_position = Vector3()
 var _speaker_connected: SpeakerPiece = null
-var _speaker_volume_awaiting_update = false
+var _speaker_volume_mouse_over = false
 var _target_zoom = 0.0
 var _timer_connected: TimerPiece = null
 var _timer_last_time_update = 0
@@ -928,10 +928,8 @@ func _on_speaker_track_resumed() -> void:
 	_set_speaker_controls()
 
 func _on_speaker_unit_size_changed(_unit_size: float) -> void:
-	if not _speaker_volume_awaiting_update:
+	if not _speaker_volume_mouse_over:
 		_set_speaker_controls()
-	
-	_speaker_volume_awaiting_update = false
 
 func _on_timer_mode_changed(_new_mode: int) -> void:
 	_set_timer_controls()
@@ -2198,9 +2196,14 @@ func _on_SpeakerPlayStopButton_pressed():
 func _on_SpeakerSelectTrackButton_pressed():
 	_track_dialog.popup_centered()
 
+func _on_SpeakerVolumeSlider_mouse_entered():
+	_speaker_volume_mouse_over = true
+
+func _on_SpeakerVolumeSlider_mouse_exited():
+	_speaker_volume_mouse_over = false
+
 func _on_SpeakerVolumeSlider_value_changed(value: float):
 	if _speaker_connected:
-		_speaker_volume_awaiting_update = true
 		_speaker_connected.rpc_id(1, "request_set_unit_size", value)
 
 func _on_StartStopCountdownButton_pressed():
