diff --git a/scripts/singletons/jdungeon.gd b/scripts/singletons/jdungeon.gd
index 61e1c1b..12aec5f 100644
--- a/scripts/singletons/jdungeon.gd
+++ b/scripts/singletons/jdungeon.gd
@@ -31,7 +31,7 @@ var audio: SoundManager = SoundManager.new()
 func _ready():
 	uuid_util = load("res://scripts/uuid/uuid.gd").new()
 	
-	audio.parent = self
+	audio.player_parent = self
 	audio.main_instance = audio
 
 
