diff --git a/godot/local_map/LocalMap.tscn b/godot/local_map/LocalMap.tscn
index f43a79b..aef974c 100644
--- a/godot/local_map/LocalMap.tscn
+++ b/godot/local_map/LocalMap.tscn
@@ -40,7 +40,7 @@ tile_data = PoolIntArray( -65536, 1, 0, -65535, 1, 0, -65534, 1, 0, -65533, 1, 0
 script = ExtResource( 4 )
 
 [node name="SpawningPoint" parent="GameBoard" instance=ExtResource( 5 )]
-position = Vector2( 520, 360 )
+position = Vector2( 920, 600 )
 
 [node name="Pawns" type="YSort" parent="GameBoard"]
 script = ExtResource( 6 )
@@ -49,10 +49,10 @@ party_scene = ExtResource( 7 )
 [node name="EncounterPawn" parent="GameBoard/Pawns" instance=ExtResource( 8 )]
 position = Vector2( 440, 600 )
 facing = {
-"down": true,
-"left": true,
-"right": true,
-"up": true
+"down": false,
+"left": false,
+"right": false,
+"up": false
 }
 
 [node name="CollisionShape2D" parent="GameBoard/Pawns/EncounterPawn" index="2"]
diff --git a/godot/local_map/pawns/InteractivePawn.gd b/godot/local_map/pawns/InteractivePawn.gd
index c205fef..08ea690 100644
--- a/godot/local_map/pawns/InteractivePawn.gd
+++ b/godot/local_map/pawns/InteractivePawn.gd
@@ -14,16 +14,24 @@ export var facing = {
 	"down": true
 }
 
+var active_raycasts := []
+
 func _ready() -> void:
-	connect('body_entered', self, '_on_body_entered')
-	connect('body_exited', self, '_on_body_exited')
-	
-	var enable_process = 0
+	# Initializes raycast nodes, deactivates the area if using raycasts
+	# for player detection
+	var use_area = true
 	for raycast in raycasts.get_children():
-		raycast.enabled = facing[raycast.name.to_lower()]
-		raycast.cast_to *= sight_distance
-		enable_process += int(raycast.enabled)
-	set_physics_process(enable_process > 0)
+		if not facing[raycast.name.to_lower()]:
+			continue
+		raycast.enabled = true
+		raycast.cast_to = raycast.cast_to.normalized() * sight_distance
+		active_raycasts.append(raycast)
+		use_area = false
+	
+	if use_area:
+		connect('body_entered', self, '_on_body_entered')
+		connect('body_exited', self, '_on_body_exited')
+		set_physics_process(false)
 
 func _unhandled_input(event: InputEvent) -> void:
 	if event.is_action_pressed("ui_accept") and dialogue_balloon.visible:
@@ -31,16 +39,21 @@ func _unhandled_input(event: InputEvent) -> void:
 		get_tree().set_input_as_handled()
 
 func _physics_process(delta : float) -> void:
-	for raycast in raycasts.get_children():
-		if not raycast.is_colliding():
-			return
-		if AUTO_START_INTERACTION:
-			start_interaction()
-		elif dialogue_balloon.visible:
-			return
-		dialogue_balloon.show()
-	if dialogue_balloon.visible:
-		dialogue_balloon.hide()
+	# Only runs if using raycasts/specific directions for player detection
+	if not dialogue_balloon.visible:
+		for raycast in active_raycasts:
+			if not raycast.is_colliding():
+				continue
+			if AUTO_START_INTERACTION:
+				start_interaction()
+			dialogue_balloon.show()
+	else:
+		var inactive_count : int = 0
+		for raycast in active_raycasts:
+			if not raycast.is_colliding():
+				inactive_count += 1
+		if inactive_count == active_raycasts.size():
+			dialogue_balloon.visible = false
 
 func _on_body_entered(body : PhysicsBody2D) -> void:
 	if AUTO_START_INTERACTION:
diff --git a/godot/local_map/pawns/InteractivePawn.tscn b/godot/local_map/pawns/InteractivePawn.tscn
index f0588b9..23c92d1 100644
--- a/godot/local_map/pawns/InteractivePawn.tscn
+++ b/godot/local_map/pawns/InteractivePawn.tscn
@@ -37,19 +37,18 @@ shape = SubResource( 1 )
 [node name="Raycasts" type="Node2D" parent="."]
 
 [node name="Up" type="RayCast2D" parent="Raycasts"]
-cast_to = Vector2( 0, -1 )
+cast_to = Vector2( 0, -50 )
 collision_mask = 2
 
 [node name="Down" type="RayCast2D" parent="Raycasts"]
-cast_to = Vector2( 0, 1 )
 collision_mask = 2
 
 [node name="Left" type="RayCast2D" parent="Raycasts"]
-cast_to = Vector2( -1, 0 )
+cast_to = Vector2( -50, 0 )
 collision_mask = 2
 
 [node name="Right" type="RayCast2D" parent="Raycasts"]
-cast_to = Vector2( 1, 0 )
+cast_to = Vector2( 50, 0 )
 collision_mask = 2
 
 [node name="Actions" type="Node" parent="."]
