diff --git a/scripts/components/statssynchronizercomponent/StatsSynchronizerComponent.gd b/scripts/components/statssynchronizercomponent/StatsSynchronizerComponent.gd
index f981d48..4d4fef0 100644
--- a/scripts/components/statssynchronizercomponent/StatsSynchronizerComponent.gd
+++ b/scripts/components/statssynchronizercomponent/StatsSynchronizerComponent.gd
@@ -87,7 +87,7 @@ signal respawned
 
 @export var energy_regen: int = 10:
 	set(val):
-		energy = val
+		energy_regen = val
 		stats_changed.emit(TYPE.ENERGY_REGEN)
 
 @export var attack_power_min: int = 0:
@@ -472,16 +472,7 @@ func _sync_float_change(stat_type: TYPE, value: float):
 		)
 
 
-func sync_stats():
-	if not G.is_server():
-		return
-
-	var id = multiplayer.get_remote_sender_id()
-
-	# Only allow logged in players
-	if not G.is_user_logged_in(id):
-		return
-
+func sync_stats(id: int):
 	G.sync_rpc.statssynchronizer_sync_response.rpc_id(id, target_node.name, to_json(true))
 
 
diff --git a/scripts/network/rpcs/gameserver/sync.gd b/scripts/network/rpcs/gameserver/sync.gd
index 953ffa1..c3265db 100644
--- a/scripts/network/rpcs/gameserver/sync.gd
+++ b/scripts/network/rpcs/gameserver/sync.gd
@@ -85,7 +85,7 @@ func positionsynchronizer_sync(n: String, t: float, p: Vector2, v: Vector2):
 		return
 
 	if entity.component_list.has("stats_synchronizer"):
-		entity.component_list["stats_synchronizer"].sync_stats()
+		entity.component_list["stats_synchronizer"].sync_stats(id)
 
 
 @rpc("call_remote", "authority", "reliable")
