diff --git a/scenes/title/TitleScreen.tscn b/scenes/title/TitleScreen.tscn
index 103a6c5..be2cb57 100644
--- a/scenes/title/TitleScreen.tscn
+++ b/scenes/title/TitleScreen.tscn
@@ -1,4 +1,4 @@
-[gd_scene load_steps=20 format=2]
+[gd_scene load_steps=21 format=2]
 
 [ext_resource path="res://fonts/ALittleNameCalle.ttf" type="DynamicFontData" id=1]
 [ext_resource path="res://sprites/bg-tile.png" type="Texture" id=2]
@@ -6,6 +6,7 @@
 [ext_resource path="res://scenes/title/Background.gd" type="Script" id=4]
 [ext_resource path="res://audio/sfx/meow.wav" type="AudioStream" id=5]
 [ext_resource path="res://scenes/title/ContributorsLabel.gd" type="Script" id=6]
+[ext_resource path="res://scripts/Audio.gd" type="Script" id=7]
 [ext_resource path="res://scenes/title/CreditsContainer.gd" type="Script" id=10]
 [ext_resource path="res://audio/music/dramaticgamemusic.mp3" type="AudioStream" id=12]
 [ext_resource path="res://scenes/title/TitleScreen.gd" type="Script" id=15]
@@ -133,6 +134,9 @@ anchor_bottom = 1.0
 mouse_filter = 2
 theme = ExtResource( 3 )
 script = ExtResource( 15 )
+__meta__ = {
+"_edit_use_anchors_": false
+}
 
 [node name="Background" type="Sprite" parent="."]
 modulate = Color( 0.7, 0.7, 0.7, 1 )
@@ -255,7 +259,6 @@ margin_bottom = 52.0
 text = "MARWING"
 redirect_scene = "res://scenes/marwing/levels/1-0.tscn"
 
-
 [node name="OptionsButton" parent="VBoxContainer/VBoxContainer" instance=ExtResource( 16 )]
 margin_left = 324.0
 margin_top = 112.0
@@ -264,7 +267,6 @@ margin_bottom = 164.0
 disabled = true
 text = "OPTIONS"
 
-
 [node name="CreditsButton" parent="VBoxContainer/VBoxContainer" instance=ExtResource( 16 )]
 margin_left = 324.0
 margin_top = 168.0
@@ -321,7 +323,10 @@ script = ExtResource( 6 )
 wait_time = 3.0
 autostart = true
 
-[node name="AudioStreamPlayer" type="AudioStreamPlayer" parent="."]
+[node name="Audio" type="Node" parent="."]
+script = ExtResource( 7 )
+
+[node name="BGM" type="AudioStreamPlayer" parent="Audio"]
 pause_mode = 2
 stream = ExtResource( 12 )
 volume_db = -20.0
