diff --git a/Scenes/Prefabs/UI/BasePauseMenu.tscn b/Scenes/Prefabs/UI/BasePauseMenu.tscn
index 32e95ae..c3bf8a1 100644
--- a/Scenes/Prefabs/UI/BasePauseMenu.tscn
+++ b/Scenes/Prefabs/UI/BasePauseMenu.tscn
@@ -99,17 +99,5 @@ metadata/_custom_type_script = "uid://co6tjg3w6qpd8"
 layout_mode = 1
 disabled_containers = [NodePath("PanelContainer/MarginContainer/VBoxContainer/Difficulty"), NodePath("PanelContainer/MarginContainer/VBoxContainer/Data")]
 
-[node name="Label" parent="SettingsMenu/PanelContainer/MarginContainer/VBoxContainer/ResourcePacks" index="0"]
-visible = true
-
-[node name="LabelFontChanger" parent="SettingsMenu" index="1" node_paths=PackedStringArray("labels")]
-labels = [NodePath("../PanelContainer/MarginContainer/VBoxContainer/Title"), NodePath("../PanelContainer/MarginContainer/VBoxContainer/CategorySelect/HBoxContainer/Category"), null, null, null, null, null, null, null, NodePath("../PanelContainer/MarginContainer/VBoxContainer/Audio/SelectableOptionNode/Value"), null, NodePath("../PanelContainer/MarginContainer/VBoxContainer/Audio/SelectableOptionNode2/Value"), null, NodePath("../PanelContainer/MarginContainer/VBoxContainer/Audio/SelectableOptionNode3/Value"), NodePath("../PanelContainer/MarginContainer/VBoxContainer/Difficulty/Description/AutoScrollContainer/MarginContainer/Desc")]
-
 [connection signal="closed" from="." to="SettingsMenu" method="close"]
 [connection signal="option_3_selected" from="." to="." method="open_settings"]
-
-[editable path="SettingsMenu"]
-[editable path="SettingsMenu/PanelContainer/MarginContainer/VBoxContainer/Video/Language"]
-[editable path="SettingsMenu/PanelContainer/MarginContainer/VBoxContainer/Audio/SelectableOptionNode"]
-[editable path="SettingsMenu/PanelContainer/MarginContainer/VBoxContainer/Audio/SelectableOptionNode2"]
-[editable path="SettingsMenu/PanelContainer/MarginContainer/VBoxContainer/Audio/SelectableOptionNode3"]
diff --git a/Scenes/Prefabs/UI/BooRacePauseMenu.tscn b/Scenes/Prefabs/UI/BooRacePauseMenu.tscn
index 1391a82..b543185 100644
--- a/Scenes/Prefabs/UI/BooRacePauseMenu.tscn
+++ b/Scenes/Prefabs/UI/BooRacePauseMenu.tscn
@@ -38,9 +38,3 @@ script = SubResource("GDScript_3lwy2")
 [connection signal="option_2_selected" from="." to="Handler" method="restart_race"]
 [connection signal="option_4_selected" from="." to="." method="close"]
 [connection signal="option_4_selected" from="." to="Handler" method="quit_to_menu"]
-
-[editable path="SettingsMenu"]
-[editable path="SettingsMenu/PanelContainer/MarginContainer/VBoxContainer/Video/Language"]
-[editable path="SettingsMenu/PanelContainer/MarginContainer/VBoxContainer/Audio/SelectableOptionNode"]
-[editable path="SettingsMenu/PanelContainer/MarginContainer/VBoxContainer/Audio/SelectableOptionNode2"]
-[editable path="SettingsMenu/PanelContainer/MarginContainer/VBoxContainer/Audio/SelectableOptionNode3"]
diff --git a/Scenes/Prefabs/UI/MarathonPause.tscn b/Scenes/Prefabs/UI/MarathonPause.tscn
index 1e429fb..56383df 100644
--- a/Scenes/Prefabs/UI/MarathonPause.tscn
+++ b/Scenes/Prefabs/UI/MarathonPause.tscn
@@ -45,9 +45,3 @@ script = SubResource("GDScript_k4rkm")
 [connection signal="option_2_selected" from="." to="Handler" method="restart_run"]
 [connection signal="option_4_selected" from="." to="." method="close"]
 [connection signal="option_4_selected" from="." to="Handler" method="quit_to_menu"]
-
-[editable path="SettingsMenu"]
-[editable path="SettingsMenu/PanelContainer/MarginContainer/VBoxContainer/Video/Language"]
-[editable path="SettingsMenu/PanelContainer/MarginContainer/VBoxContainer/Audio/SelectableOptionNode"]
-[editable path="SettingsMenu/PanelContainer/MarginContainer/VBoxContainer/Audio/SelectableOptionNode2"]
-[editable path="SettingsMenu/PanelContainer/MarginContainer/VBoxContainer/Audio/SelectableOptionNode3"]
diff --git a/Scenes/Prefabs/UI/MarathonPracticePause.tscn b/Scenes/Prefabs/UI/MarathonPracticePause.tscn
index 4cd34ff..02986d5 100644
--- a/Scenes/Prefabs/UI/MarathonPracticePause.tscn
+++ b/Scenes/Prefabs/UI/MarathonPracticePause.tscn
@@ -47,9 +47,3 @@ script = SubResource("GDScript_k4rkm")
 
 [connection signal="option_2_selected" from="." to="Handler" method="restart_run"]
 [connection signal="option_3_selected" from="." to="Handler" method="quit_to_menu"]
-
-[editable path="SettingsMenu"]
-[editable path="SettingsMenu/PanelContainer/MarginContainer/VBoxContainer/Video/Language"]
-[editable path="SettingsMenu/PanelContainer/MarginContainer/VBoxContainer/Audio/SelectableOptionNode"]
-[editable path="SettingsMenu/PanelContainer/MarginContainer/VBoxContainer/Audio/SelectableOptionNode2"]
-[editable path="SettingsMenu/PanelContainer/MarginContainer/VBoxContainer/Audio/SelectableOptionNode3"]
diff --git a/Scenes/Prefabs/UI/StoryPause.tscn b/Scenes/Prefabs/UI/StoryPause.tscn
index f6c71b4..ba27b15 100644
--- a/Scenes/Prefabs/UI/StoryPause.tscn
+++ b/Scenes/Prefabs/UI/StoryPause.tscn
@@ -45,15 +45,6 @@ texture = SubResource("AtlasTexture_eud56")
 [node name="Handler" type="Node" parent="." index="2"]
 script = SubResource("GDScript_r6hlk")
 
-[node name="LabelFontChanger" parent="SettingsMenu" index="1" node_paths=PackedStringArray("labels")]
-labels = [NodePath("../PanelContainer/MarginContainer/VBoxContainer/Title"), NodePath("../PanelContainer/MarginContainer/VBoxContainer/CategorySelect/HBoxContainer/Category"), null, null, null, null, null, null, null, NodePath("../PanelContainer/MarginContainer/VBoxContainer/Audio/SelectableOptionNode/Value"), null, NodePath("../PanelContainer/MarginContainer/VBoxContainer/Audio/SelectableOptionNode2/Value"), null, NodePath("../PanelContainer/MarginContainer/VBoxContainer/Audio/SelectableOptionNode3/Value"), NodePath("../PanelContainer/MarginContainer/VBoxContainer/Difficulty/Description/AutoScrollContainer/MarginContainer/Desc")]
-
 [connection signal="option_1_selected" from="." to="." method="close"]
 [connection signal="option_2_selected" from="." to="Handler" method="restart_level"]
 [connection signal="option_4_selected" from="." to="Handler" method="quit_to_menu"]
-
-[editable path="SettingsMenu"]
-[editable path="SettingsMenu/PanelContainer/MarginContainer/VBoxContainer/Video/Language"]
-[editable path="SettingsMenu/PanelContainer/MarginContainer/VBoxContainer/Audio/SelectableOptionNode"]
-[editable path="SettingsMenu/PanelContainer/MarginContainer/VBoxContainer/Audio/SelectableOptionNode2"]
-[editable path="SettingsMenu/PanelContainer/MarginContainer/VBoxContainer/Audio/SelectableOptionNode3"]
