diff --git a/scenes/levels/CameraFollow.gd b/scenes/levels/CameraFollow.gd
index ce651a3..69e88fc 100644
--- a/scenes/levels/CameraFollow.gd
+++ b/scenes/levels/CameraFollow.gd
@@ -7,8 +7,7 @@ export(float,0,1) var follow_speed: float = 0.1
 
 func _process(_delta):
 	
-	if(not player_reference):
-		print("not player")
+	if(not player_reference or not player_reference.is_inside_tree()):
 		player_reference = $"../Player"
 
 	if(camera_reference and player_reference):
