diff --git a/game/Scripts/AssetDB.gd b/game/Scripts/AssetDB.gd
index 5a62d46..a1b5442 100644
--- a/game/Scripts/AssetDB.gd
+++ b/game/Scripts/AssetDB.gd
@@ -529,17 +529,23 @@ func _calculate_bounding_box_recursive(scene: Spatial, transform: Transform) ->
 	
 	if scene is MeshInstance:
 		var mesh  = scene.mesh
-		var shape = mesh.create_convex_shape()
-		for point in shape.points:
-			var adj_point = new_transform * point
-			
-			bounding_box[0].x = min(bounding_box[0].x, adj_point.x)
-			bounding_box[0].y = min(bounding_box[0].y, adj_point.y)
-			bounding_box[0].z = min(bounding_box[0].z, adj_point.z)
-			
-			bounding_box[1].x = max(bounding_box[1].x, adj_point.x)
-			bounding_box[1].y = max(bounding_box[1].y, adj_point.y)
-			bounding_box[1].z = max(bounding_box[1].z, adj_point.z)
+		# Check that the mesh actually has any vertices in it.
+		var num_verts = 0
+		for surface_id in range(mesh.get_surface_count()):
+			num_verts += mesh.surface_get_arrays(surface_id)[0].size()
+		
+		if num_verts > 0:
+			var shape = mesh.create_convex_shape()
+			for point in shape.points:
+				var adj_point = new_transform * point
+				
+				bounding_box[0].x = min(bounding_box[0].x, adj_point.x)
+				bounding_box[0].y = min(bounding_box[0].y, adj_point.y)
+				bounding_box[0].z = min(bounding_box[0].z, adj_point.z)
+				
+				bounding_box[1].x = max(bounding_box[1].x, adj_point.x)
+				bounding_box[1].y = max(bounding_box[1].y, adj_point.y)
+				bounding_box[1].z = max(bounding_box[1].z, adj_point.z)
 	
 	for child in scene.get_children():
 		if child is Spatial:
diff --git a/game/Scripts/PieceBuilder.gd b/game/Scripts/PieceBuilder.gd
index 7191174..90afedf 100644
--- a/game/Scripts/PieceBuilder.gd
+++ b/game/Scripts/PieceBuilder.gd
@@ -291,7 +291,13 @@ func _extract_and_shape_mesh_instances(add_to: Node, from: Node,
 		if num_materials < num_surfaces:
 			push_warning("Mesh '%s' has %d surfaces, but only %d materials!" %
 				[from.mesh.resource_path, num_surfaces, num_materials])
+		
+		var num_verts = 0
 		for surface in range(num_surfaces):
+			# A number of arrays make up the surface - the first of them being
+			# the vertex array (which is guaranteed to be there).
+			num_verts += from.mesh.surface_get_arrays(surface)[0].size()
+			
 			# NOTE: We always assume that imported meshes put their materials
 			# into the mesh itself, not the mesh instance.
 			var material = from.mesh.surface_get_material(surface)
@@ -309,24 +315,23 @@ func _extract_and_shape_mesh_instances(add_to: Node, from: Node,
 						omnilight.light_energy = material.emission_energy
 						from.add_child(omnilight)
 		
-		var collision_shape = CollisionShape.new()
-		# Setting clean to false here, because I've noticed it makes little to
-		# no difference in the collision shapes themselves, but it speeds up
-		# the process significantly.
-		collision_shape.shape = from.mesh.create_convex_shape(false, false)
-		
-		# The collision shape's transform needs to match up with the mesh
-		# instance's, but they can't both use the same transform, otherwise
-		# the transform of the mesh instance will be wrong.
-		var collision_transform = transform
-		collision_transform.basis = from.transform.basis * collision_transform.basis
-		collision_transform.origin = from.transform.origin + collision_transform.origin
-		
-		collision_shape.transform = collision_transform
-		from.transform = Transform.IDENTITY
-		
-		collision_shape.add_child(from)
-		add_to.add_child(collision_shape)
+		# Don't bother making a collision shape if there's no vertices.
+		if num_verts > 0:
+			var collision_shape = CollisionShape.new()
+			collision_shape.shape = from.mesh.create_convex_shape()
+			
+			# The collision shape's transform needs to match up with the mesh
+			# instance's, but they can't both use the same transform, otherwise
+			# the transform of the mesh instance will be wrong.
+			var collision_transform = transform
+			collision_transform.basis = from.transform.basis * collision_transform.basis
+			collision_transform.origin = from.transform.origin + collision_transform.origin
+			
+			collision_shape.transform = collision_transform
+			from.transform = Transform.IDENTITY
+			
+			collision_shape.add_child(from)
+			add_to.add_child(collision_shape)
 
 # Free an object in a thread-safe manner.
 # object: The object to free.
