diff --git a/source/match/handlers/BuildingPlacementHandler.gd b/source/match/handlers/BuildingPlacementHandler.gd
index 01b152c..20c741d 100644
--- a/source/match/handlers/BuildingPlacementHandler.gd
+++ b/source/match/handlers/BuildingPlacementHandler.gd
@@ -99,7 +99,7 @@ func _calculate_blueprint_position_validity():
 	var placement_validity = Utils.Match.Unit.Placement.validate_agent_placement_position(
 		_active_blueprint_node.global_position,
 		_pending_building_radius + Constants.Match.Units.EMPTY_SPACE_RADIUS_SURROUNDING_STRUCTURE_M,
-		get_tree().get_nodes_in_group("units"),
+		get_tree().get_nodes_in_group("units") + get_tree().get_nodes_in_group("resource_units"),
 		_pending_building_navmap_rid
 	)
 	if placement_validity == Utils.Match.Unit.Placement.COLLIDES_WITH_AGENT:
diff --git a/source/match/utils/UnitPlacementUtils.gd b/source/match/utils/UnitPlacementUtils.gd
index af90efe..fe2756d 100644
--- a/source/match/utils/UnitPlacementUtils.gd
+++ b/source/match/utils/UnitPlacementUtils.gd
@@ -20,7 +20,9 @@ static func find_valid_position_radially_yet_skip_starting_radius(
 	scene_tree
 ):
 	var starting_position_yless = starting_position * Vector3(1, 0, 1)
-	var units = scene_tree.get_nodes_in_group("units")
+	var units = (
+		scene_tree.get_nodes_in_group("units") + scene_tree.get_nodes_in_group("resource_units")
+	)
 	var starting_distance = (
 		0 if is_zero_approx(starting_radius) else starting_radius + radius + spacing
 	)
