diff --git a/godot/quest_system/quests/Quest.gd b/godot/quest_system/quests/Quest.gd
index e570f33..e2bfea8 100644
--- a/godot/quest_system/quests/Quest.gd
+++ b/godot/quest_system/quests/Quest.gd
@@ -14,11 +14,14 @@ export var title : String
 export var description : String
 
 export var reward_on_delivery : bool = false
-export var reward_experience : int
+export var _reward_experience : int
+onready var _reward_items : Node = $ItemRewards
 
 func _start():
 	for objective in get_objectives():
-		objective.connect("objective_finished", self, "_on_Objective_finished")
+		objective.connect("completed", self, "_on_Objective_completed")
+	for objective in get_objectives():
+		objective.finish()
 
 func get_objectives():
 	return objectives.get_children()
@@ -31,7 +34,7 @@ func get_completed_objectives():
 		completed.append(objective)
 	return completed
 
-func _on_Objective_finished(objective) -> void:
+func _on_Objective_completed(objective) -> void:
 	if get_completed_objectives().size() == get_objectives().size():
 		emit_signal("completed", self)
 
@@ -41,9 +44,18 @@ func notify_slay_objectives() -> void:
 			continue
 		(objective as QuestSlayObjective).connect_signals()
 
+func get_rewards() -> Dictionary:
+	"""
+	Returns the rewards from the quest as a dictionary of the form:
+	"""
+	return {
+		'experience' : _reward_experience, # int
+		'items': _reward_items.get_children() # Array of Item
+	}
+
 func get_rewards_as_text() -> Array:
 	var text : = []
-	text.append(" - Experience: %s" % str(reward_experience))
-	for item in $ItemRewards.get_children():
+	text.append(" - Experience: %s" % str(_reward_experience))
+	for item in _reward_items.get_items():
 		text.append(" - [%s] x (%s)\n" % [item.item.name, str(item.amount)])
 	return text
diff --git a/godot/quest_system/quests/TestQuest.tscn b/godot/quest_system/quests/TestQuest.tscn
index 84d6b1d..598ef3b 100644
--- a/godot/quest_system/quests/TestQuest.tscn
+++ b/godot/quest_system/quests/TestQuest.tscn
@@ -10,8 +10,6 @@
 [node name="TestQuest" index="0" instance=ExtResource( 1 )]
 title = "Test Quest"
 description = "Lorem ipsum dolor sit amet, consectetur adipiscing elit."
-exp_reward = 5
-has_to_be_delivered = true
 
 [node name="QuestItemReward" parent="ItemRewards" index="0" instance=ExtResource( 2 )]
 item = ExtResource( 3 )
