diff --git a/scripts/components/networking/statuseffectcomponent/StatusEffectResource.gd b/scripts/components/networking/statuseffectcomponent/StatusEffectResource.gd
index f8d6b5a..f719d84 100644
--- a/scripts/components/networking/statuseffectcomponent/StatusEffectResource.gd
+++ b/scripts/components/networking/statuseffectcomponent/StatusEffectResource.gd
@@ -15,7 +15,7 @@ enum Properties {
 @export var status_class: String
 @export var displayed_name: String = "Unnamed Status"
 @export_multiline var description: String
-@export_file var icon_path: String
+@export_file var icon_path: String = "res://icon.svg"
 
 @export_group("General Behaviour")
 ## How many stacks are removed per timeout (flat amount)
diff --git a/scripts/components/networking/statuseffectcomponent/StatusEffects/Poison.tres b/scripts/components/networking/statuseffectcomponent/StatusEffects/Poison.tres
index 12734e1..0df9e1e 100644
--- a/scripts/components/networking/statuseffectcomponent/StatusEffects/Poison.tres
+++ b/scripts/components/networking/statuseffectcomponent/StatusEffects/Poison.tres
@@ -5,7 +5,8 @@ script/source = "extends StatusEffectResource
 
 func _effect_timeout(target: Node, json_data: Dictionary):
 	var targetStats: StatsSynchronizerComponent = get_entity_stats_component(target)
-	targetStats.hurt(json_data[\"owner\"], json_data[\"stacks\"])
+	var entity: Node = G.world.get_entity_by_name(json_data[\"applier\"])
+	targetStats.hurt(entity, json_data[\"stacks\"])
 	
 "
 
@@ -14,7 +15,7 @@ script = SubResource("GDScript_cqumv")
 status_class = "Poison"
 displayed_name = "Poison"
 description = "Deals damage every time it's duration ends equal to it's stack, then looses 20% of it's stacks. "
-icon_path = ""
+icon_path = "res://icon.svg"
 stack_consumption_flat = 0
 stack_consumption_percent = 0.2
 active_stat_modifiers = {}
diff --git a/scripts/components/networking/statuseffectcomponent/statuseffectcomponent.gd b/scripts/components/networking/statuseffectcomponent/statuseffectcomponent.gd
index 03527e3..dd83dcf 100644
--- a/scripts/components/networking/statuseffectcomponent/statuseffectcomponent.gd
+++ b/scripts/components/networking/statuseffectcomponent/statuseffectcomponent.gd
@@ -90,12 +90,12 @@ func process_statuses():
 			set_stacks(status, (startStacks - (startStacks * startResource.stack_consumption_percent)) - startResource.stack_consumption_flat )
 			startResource.effect_timeout(user, get_status_effect_data(status))
 		
-		#If any stacks remain, reset the duration. Otherwise remove this status.
-		if get_stacks(status) > 0:
-			set_duration(status, startDuration)
-		else:
-			remove_status(status)
 			
+			#If any stacks remain, reset the duration. Otherwise remove this status.	
+			if get_stacks(status) > 0:
+				set_duration(status, startResource.default_duration)
+			else:
+				remove_status(status)
 		sync_effect(user.peer_id, status)
 	
 
@@ -105,12 +105,14 @@ func add_status_effect(status_class: String, applier: Node, stack_override: int
 	var currDuration: float
 	if duration_override >= 0:
 		currDuration = duration_override
+		newStatus.default_duration = duration_override
 	else:
 		currDuration = newStatus.default_duration
 	
 	var currStacks: int
 	if stack_override >= 0:
 		currStacks = stack_override
+		newStatus.default_stacks = stack_override
 	else:
 		currStacks = newStatus.default_stacks
 	
@@ -146,6 +148,7 @@ func remove_status(status_class: String):
 
 func set_duration(status_class: String, dur: float):
 	if not status_effects_active.has(status_class):
+		GodotLogger.error("This status is not present in this component.")
 		return
 	
 	status_effects_active[status_class]["duration"] = dur
@@ -159,6 +162,7 @@ func get_duration(status_class: String) -> float:
 
 func set_stacks(status_class: String, sta: int):
 	if not status_effects_active.has(status_class):
+		GodotLogger.error("This status is not present in this component.")
 		return
 		
 	status_effects_active[status_class]["stacks"] = sta
@@ -170,6 +174,7 @@ func get_stacks(status_class: String) -> int:
 
 func set_resource(status_class: String, resource: StatusEffectResource):
 	if not status_effects_active.has(status_class):
+		GodotLogger.error("This status is not present in this component.")
 		return
 		
 	status_effects_active[status_class]["resource"] = resource
@@ -181,6 +186,7 @@ func get_resource(status_class: String) -> StatusEffectResource:
 
 func set_applier(status_class: String, applier_name: String):
 	if not status_effects_active.has(status_class):
+		GodotLogger.error("This status is not present in this component.")
 		return
 		
 	status_effects_active[status_class]["applier"] = applier_name
diff --git a/scripts/components/player/charclasscomponent/CharClassSelection.gd b/scripts/components/player/charclasscomponent/CharClassSelection.gd
index f18292b..934d92b 100644
--- a/scripts/components/player/charclasscomponent/CharClassSelection.gd
+++ b/scripts/components/player/charclasscomponent/CharClassSelection.gd
@@ -39,15 +39,6 @@ func _ready() -> void:
 	add_child(syncTimer)
 	syncTimer.start(1)
 
-	if not class_component:
-		(
-			GodotLogger
-			. warn(
-				"No target has been selected. Make sure to run select_target before anything is attempted."
-			)
-		)
-		return
-
 
 func select_target(class_comp: CharacterClassComponent):
 	if not is_node_ready():
diff --git a/scripts/components/player/skillcomponent/skillcomponent.gd b/scripts/components/player/skillcomponent/skillcomponent.gd
index 16a8b26..5ac6a71 100644
--- a/scripts/components/player/skillcomponent/skillcomponent.gd
+++ b/scripts/components/player/skillcomponent/skillcomponent.gd
@@ -133,7 +133,7 @@ func _ready() -> void:
 		assert(skills[0] is SkillComponentResource)
 		# TEMP ends
 
-		skills_changed.connect(sync_skills)
+		skills_changed.connect(sync_skills.bind(_target_node.peer_id))
 
 	# Client-side logic
 	else:
