diff --git a/scripts/components/networking/projectilesynchronizercomponent/Projectile2D.gd b/scripts/components/networking/projectilesynchronizercomponent/Projectile2D.gd
index d5df0ea..d9f661f 100644
--- a/scripts/components/networking/projectilesynchronizercomponent/Projectile2D.gd
+++ b/scripts/components/networking/projectilesynchronizercomponent/Projectile2D.gd
@@ -74,7 +74,7 @@ func process_collisions(motion: Vector2):
 			hit_object.emit(coll.get("collider"))
 			collision_count += 1
 
-			if collision_count > max_collisions:
+			if collision_count >= max_collisions:
 				collision_mask = 0
 				collision_layer = 0
 				queue_free.call_deferred()
diff --git a/scripts/components/networking/projectilesynchronizercomponent/projectilesynchronizercomponent.gd b/scripts/components/networking/projectilesynchronizercomponent/projectilesynchronizercomponent.gd
index b5dfeb4..27d122d 100644
--- a/scripts/components/networking/projectilesynchronizercomponent/projectilesynchronizercomponent.gd
+++ b/scripts/components/networking/projectilesynchronizercomponent/projectilesynchronizercomponent.gd
@@ -34,18 +34,24 @@ func _ready():
 
 func launch_projectile(target_global_pos: Vector2, projectile_class: String):
 	var projectile: Projectile2D = create_projectile_node(projectile_class)
-	var projectiles_in_group: Array[Node] = get_tree().get_nodes_in_group(
-		get_projectile_group(
+	
+	# Handle instance limit
+	var projectile_group_name: String = 		get_projectile_group(
 			target_node.get_name(), target_node.get("entity_type"), projectile_class
-		)
-	)
-	if projectiles_in_group.size() > projectile.instance_limit:
+			)
+	var projectiles_in_group: Array[Node] = get_tree().get_nodes_in_group(
+				projectile_group_name
+			)
+	if projectiles_in_group.size() >= projectile.instance_limit:
 		var projectile_arr: Array[Projectile2D] = []
 		projectile_arr.assign(projectiles_in_group)
 		var oldest_projectile: Projectile2D = get_oldest_projectile(projectile_arr)
 
 		if is_instance_valid(oldest_projectile):
+			oldest_projectile.remove_from_group(projectile_group_name)
 			oldest_projectile.queue_free()
+	
+	projectile.add_to_group(projectile_group_name)
 
 	# Do not hit owner
 	projectile.add_collision_exception_with(target_node)
