diff --git a/game/scenes/loginpanel/LoginPanel.tscn b/game/scenes/loginpanel/LoginPanel.tscn
index 18bec87..1343895 100644
--- a/game/scenes/loginpanel/LoginPanel.tscn
+++ b/game/scenes/loginpanel/LoginPanel.tscn
@@ -484,7 +484,6 @@ custom_minimum_size = Vector2(350, 0)
 layout_mode = 2
 size_flags_horizontal = 4
 size_flags_vertical = 4
-focus_mode = 1
 placeholder_text = "Enter Your Password"
 alignment = 1
 secret = true
@@ -496,7 +495,6 @@ custom_minimum_size = Vector2(350, 0)
 layout_mode = 2
 size_flags_horizontal = 4
 size_flags_vertical = 4
-focus_mode = 1
 placeholder_text = "Re-Enter Your Password"
 alignment = 1
 secret = true
