diff --git a/game/Scenes/Game/3D/PaintPlane.tscn b/game/Scenes/Game/3D/PaintPlane.tscn
index 817288a..822c2f6 100644
--- a/game/Scenes/Game/3D/PaintPlane.tscn
+++ b/game/Scenes/Game/3D/PaintPlane.tscn
@@ -11,6 +11,7 @@ viewport_path = NodePath("Viewport")
 
 [sub_resource type="ShaderMaterial" id=3]
 resource_local_to_scene = true
+render_priority = -1
 shader = ExtResource( 3 )
 shader_param/FilteringEnabled = false
 shader_param/TexturePixelSize = Vector2( 1, 1 )
@@ -25,6 +26,9 @@ shader_param/BrushColor = null
 shader_param/BrushEnabled = false
 shader_param/BrushPosition = Vector2( 0, 0 )
 shader_param/BrushSize = 0.0
+shader_param/InverseQuadCol1 = null
+shader_param/InverseQuadCol2 = null
+shader_param/QuadCorner = null
 
 [node name="PaintPlane" type="Spatial"]
 script = ExtResource( 1 )
